---
title: "Long script"
output: html_document
date: "2024-07-29"
---

# -1. Objectives
The objectives of this code are:
*to create objects usable for the creation of a metanetwork of organisms on DIAMS
*to make figures and analyses for a first article with functional assignations of organisms on DIAMS

Inspired by the phyloseq object structure, 3 main objects will be created:
*tidy_data
*tax_table
*fun_table
They will be combined in phyloseq objects.

From raw data to the final combined object, each community will pass through several steps.
For all communities except microorganisms:
*step 1 = raw data collection
*step 2a = taxonomic assignation (taxize::gnr_resolve and small manual corrections)
*step 2b = taxonomic assignation (taxize::tax_name)
*step 2 = taxonomic assignation (manual additions)
*step 3 = functional assignation

For microorganisms :
*step 1 = raw data collection
*step 2 = taxonomic cleaning and filtering
*step 3 = functional assignation

Note that step number don't correspond exactly to the objects and exported files (for the moment).

# 0. Load packages, frame and import data

```{r warning=FALSE}
library(readr)
library(dplyr)
library(tidyr)
library(readxl)
library(janitor)
library(stringr)
library(rebus)
library(taxize)
library(rgbif)
library(GratinNavigatoR)
library(ggplot2)
library(phyloseq)
library(microbiome)
library(metagMisc)
library(vegan)
library(cowplot)
library(microViz)
library(MiscMetabar)
library(svglite)
library(RColorBrewer)
```

```{r}
frame_simple <- theme_bw() +
  theme(axis.text = element_text(size=8),
        axis.title = element_text(size=9,face="bold"),
        title = element_text(size=10, face="bold"))

order_level = c("kingdom", "phylum", "class", "order", "family", "genus", "species")

order_community = c("macrofauna_surface", "macrofauna_aerial", "macrofauna_foliar", "nematodes", "micro_arthropodes", "bacteria", "fungi", "protists")

order_uh = c("MC_C", "AF_C", "AF_T", "TP_T")

order_date = c('14.03.2023','17.04.2023','02.05.2023','25.05.2023')

order_spatial = c('homo', 'hetero')

brewer.pal(n = 12, name = "Dark2") # + Set2
order_color = c(
"Araneae"="#1B9E77",
"Coleoptera"="#D95F02",
"Diptera"="#7570B3",
"Hemiptera"="#E7298A",
"Hymenoptera"="#66A61E",
"Lepidoptera"="#E6AB02",
"Orthoptera"="#A6761D",
"Thysanoptera"="#E5C494",
"unassigned"="#666666",
"Isopoda"="#89C5DA",
"Geophilomorpha"="#CE50CA",
"Embioptera"="#3F4921",
"Zygentoma"="#C0717C",
"Blattodea"="#CBD588",
"Opiliones"="#5F7FC7",
"Pseudoscorpiones"="#D3D93E",
"Ixodida"="#CD9BCD",
"Trichoptera"="#6DDE88",
"Enchytraeida"="#652926",
"Julida"="#7FDCC0",
"Psocoptera"="#C84248",
"Stylommatophora"="#5E738F",
"Dermaptera"="#D1A33D")
fill_manual_order <- scale_fill_manual(values = order_color)

brewer.pal(n = 12, name = "Paired")
family_color = c("Elateridae"="#A6CEE3","Hesperiidae"="#FB9A99","Theridiidae"="#B2DF8A","unassigned"="#666666")
fill_manual_family <- scale_fill_manual(values = family_color)
```

suffix "_0" = raw data (as imported)

```{r message=FALSE, warning=FALSE}
macrofauna_surface_0 <- read_delim("data/raw_data/1. barber_taxo2023-09-26.csv", show_col_types = FALSE)[,-1]

macrofauna_aerial_0 <- read_csv("data/raw_data/1. pantrap_taxo2023-09-08.csv", show_col_types = FALSE)[,-1]

macrofauna_foliar_0 <- read_csv("data/raw_data/clean_data_robinier_2023-09-16.csv", show_col_types =  FALSE)[,-1]

micro_arthropodes_all_0 <- read_excel("data/raw_data/NH_Tri_mesofaune - INRAE MONTPELLER_AM_MH.xlsx")[,-c(11,12)]
micro_arthropodes_col_0 <- read_excel("data/raw_data/BDD Collemboles - INRAE MONTPELLIER.xlsx")

nematodes_0 <- read_excel("data/raw_data/raw data march april 2023.xlsx")

bacteria_0 <- read_delim("data/raw_data/OTU_Table_Bacterie.tsv")[,c(0:72)]

fungi_0 <- read_delim("data/raw_data/OTU_Table_Champignon.tsv")[,c(0:72)]

protists_0 <- read_delim("data/raw_data/OTU_Table_Cercozoa.tsv")[,c(0:72)]

metadata_mo_0 <- read_delim("data/raw_data/mapping_brut.csv")
```

# 1. Tidy data

Files generated in this section:
prefix "tidy_"
suffix "_1" = homogenized data (at the end of the chunk)
suffix "_2" = data with minimal information (almost ready to export)
suffix "_ALL" = all data combined (all communities)

Minimal information we want in the final tidy data files: abundance, finest_name (of individual), code_uhbr
code_uhbr = short code including land use, habitat type, block and replicate abbreviations
Additional information: date, method, community

## macrofauna_surface

```{r}
tidy_macrofauna_surface_1 <- macrofauna_surface_0

# rename columns
tidy_macrofauna_surface_1 <- tidy_macrofauna_surface_1 %>%  rename(block = bloc, replicate = rep)

# add column method, date and community
tidy_macrofauna_surface_1 <- tidy_macrofauna_surface_1 %>% mutate(method = "barber_trap")
tidy_macrofauna_surface_1 <- tidy_macrofauna_surface_1 %>% mutate(community = "macrofauna_surface")
tidy_macrofauna_surface_1 <- tidy_macrofauna_surface_1 %>%
  mutate(date = case_when(
      grepl(number_range(229, 288), id_sample) ~ "14/03/2023",
      grepl(number_range(349, 408), id_sample) ~ "17/04/2023",
      grepl(number_range(679, 738), id_sample) ~ "02/05/2023"))
#id_sample ANI-23-0229 to ANI-23-0288 = id_campaign 189 = date 14/03/23
#id_sample ANI-23-0349 to ANI-23-0408 = id_campaign 190 = date 17/04/23
#id_sample ANI-23-0679 to ANI-23-0738 = id_campaign 191 = date 02/05/23

# homogeneise code
tidy_macrofauna_surface_1$usage[tidy_macrofauna_surface_1$usage == 'C'] <- 'MC'
tidy_macrofauna_surface_1$usage[tidy_macrofauna_surface_1$usage == 'F'] <- 'TP'

tidy_macrofauna_surface_1$block[tidy_macrofauna_surface_1$block == '1'] <- 'B1'
tidy_macrofauna_surface_1$block[tidy_macrofauna_surface_1$block == '2'] <- 'B2'
tidy_macrofauna_surface_1$block[tidy_macrofauna_surface_1$block == '3'] <- 'B3'

tidy_macrofauna_surface_1$habitat[tidy_macrofauna_surface_1$habitat == 'Cult' | tidy_macrofauna_surface_1$habitat == 'cult' ] <- 'C'
tidy_macrofauna_surface_1$habitat[tidy_macrofauna_surface_1$habitat == 'Forest' | tidy_macrofauna_surface_1$habitat == 'LSA' ] <- 'T'

tidy_macrofauna_surface_1$replicate[tidy_macrofauna_surface_1$replicate == '1'] <- 'R1'
tidy_macrofauna_surface_1$replicate[tidy_macrofauna_surface_1$replicate == '2'] <- 'R2'
tidy_macrofauna_surface_1$replicate[tidy_macrofauna_surface_1$replicate == '3'] <- 'R3'
tidy_macrofauna_surface_1$replicate[tidy_macrofauna_surface_1$replicate == '4'] <- 'R4'
tidy_macrofauna_surface_1$replicate[tidy_macrofauna_surface_1$replicate == '5'] <- 'R5'

# create column code_uhbr
tidy_macrofauna_surface_1$code_uhbr <- paste(tidy_macrofauna_surface_1$usage, tidy_macrofauna_surface_1$habitat, tidy_macrofauna_surface_1$block, tidy_macrofauna_surface_1$replicate, sep = "_")

# rename name to finest_name
tidy_macrofauna_surface_1 = rename(tidy_macrofauna_surface_1, finest_name = name)
```

## macrofauna_aerial

```{r }
tidy_macrofauna_aerial_1 <- macrofauna_aerial_0

# rename columns
tidy_macrofauna_aerial_1 <- tidy_macrofauna_aerial_1 %>%  rename(block = bloc, replicate = rep)

# add column method, date and community
tidy_macrofauna_aerial_1 <- tidy_macrofauna_aerial_1 %>% mutate(method = "pan_trap")
tidy_macrofauna_aerial_1 <- tidy_macrofauna_aerial_1 %>%
  mutate(date = case_when(
    grepl('03/23', month) ~ "14/03/2023",
    grepl('04/23', month) ~ "17/04/2023"))
# samples collected in May (id_campaign 191 and id_sample from ANI-23-0739 to ANI-23-0798 are not in this table because they are not identified yet)
tidy_macrofauna_aerial_1 <- tidy_macrofauna_aerial_1 %>% mutate(community = "macrofauna_aerial")

# homogeneise code
tidy_macrofauna_aerial_1$usage[tidy_macrofauna_aerial_1$usage == 'C'] <- 'MC'
tidy_macrofauna_aerial_1$usage[tidy_macrofauna_aerial_1$usage == 'F'] <- 'TP'

tidy_macrofauna_aerial_1$block[tidy_macrofauna_aerial_1$block == '1'] <- 'B1'
tidy_macrofauna_aerial_1$block[tidy_macrofauna_aerial_1$block == '2'] <- 'B2'
tidy_macrofauna_aerial_1$block[tidy_macrofauna_aerial_1$block == '3'] <- 'B3'

tidy_macrofauna_aerial_1$habitat[tidy_macrofauna_aerial_1$habitat == 'Cult' | tidy_macrofauna_aerial_1$habitat == 'cult' ] <- 'C'
tidy_macrofauna_aerial_1$habitat[tidy_macrofauna_aerial_1$habitat == 'Forest' | tidy_macrofauna_aerial_1$habitat == 'LSA' ] <- 'T'

tidy_macrofauna_aerial_1$replicate[tidy_macrofauna_aerial_1$replicate == '1'] <- 'R1'
tidy_macrofauna_aerial_1$replicate[tidy_macrofauna_aerial_1$replicate == '2'] <- 'R2'
tidy_macrofauna_aerial_1$replicate[tidy_macrofauna_aerial_1$replicate == '3'] <- 'R3'
tidy_macrofauna_aerial_1$replicate[tidy_macrofauna_aerial_1$replicate == '4'] <- 'R4'
tidy_macrofauna_aerial_1$replicate[tidy_macrofauna_aerial_1$replicate == '5'] <- 'R5'

# create column code_uhbr
tidy_macrofauna_aerial_1$code_uhbr <- paste(tidy_macrofauna_aerial_1$usage, tidy_macrofauna_aerial_1$habitat, tidy_macrofauna_aerial_1$block, tidy_macrofauna_aerial_1$replicate, sep = "_")

# rename name to finest_name
tidy_macrofauna_aerial_1 = rename(tidy_macrofauna_aerial_1, finest_name = name)
```

## macrofauna_foliar

```{r}
tidy_macrofauna_foliar_1 <- macrofauna_foliar_0

# modify column method (two organs were collected for macrofauna_foliar)
tidy_macrofauna_foliar_1$method[macrofauna_foliar_0$method == 'rameau'] <- 'twig'
tidy_macrofauna_foliar_1$method[tidy_macrofauna_foliar_1$method == 'dissection inflo'] <- 'inflorescence'

# add column habitat, date and community
tidy_macrofauna_foliar_1 <- tidy_macrofauna_foliar_1 %>% mutate(habitat = "T")
tidy_macrofauna_foliar_1 <- tidy_macrofauna_foliar_1 %>% mutate(community = "macrofauna_foliar")
tidy_macrofauna_foliar_1 <- tidy_macrofauna_foliar_1 %>%
  mutate(date = case_when(
      grepl(number_range(409, 498), id_sample) ~ "17/04/2023",
      grepl(number_range(499, 588), id_sample) ~ "02/05/2023",
      grepl(number_range(589, 678), id_sample) ~ "25/05/2023"))
#id_sample ANI-23-0409 to ANI-23-0498 = id_campaign 190 = date 17/04/23
#id_sample ANI-23-0499 to ANI-23-0588 = id_campaign 191 = date 02/05/23
#id_sample ANI-23-0589 to ANI-23-0678 = id_campaign 192 = date 26/05/23
#samples collected on leaf pods in June (id_campaign 192 and id_sample from ANI-23-0799 to ANI-23-0887) are not in this table

# homogeneise code
tidy_macrofauna_foliar_1 <- tidy_macrofauna_foliar_1 %>% 
   separate(col = usage_bloc , into = c("usage","block","rang","arbre"), sep = "_")

tidy_macrofauna_foliar_1 <- tidy_macrofauna_foliar_1 %>% mutate(across('rang', str_replace, 'arbre', ''))

tidy_macrofauna_foliar_1$usage <- gsub('^F','TP', tidy_macrofauna_foliar_1$usage)

# OPTION 1 (keeping rang and arbre info in code_uhbr)
#macrofauna_foliar_1 <- macrofauna_foliar_1 %>% mutate(rang = gsub("rang", "R", rang)) 
#macrofauna_foliar_1 <- macrofauna_foliar_1 %>% mutate(arbre = gsub("arbre", "", arbre))
#macrofauna_foliar_1 <- macrofauna_foliar_1 %>% mutate(replicate = paste(rang, arbre ,sep="."))

# OPTION 2 (simplified: rang number becomes replicate number and are transformed to range from R1 to R5)
tidy_macrofauna_foliar_1 <- tidy_macrofauna_foliar_1 %>% mutate(replicate = gsub("rang", "R", rang))

tidy_macrofauna_foliar_1$replicate <- paste0("R", (as.numeric(sub("R", "", tidy_macrofauna_foliar_1$replicate)) %% 5) + 1)

tidy_macrofauna_foliar_1$block[tidy_macrofauna_foliar_1$block == '1'] <- 'B1'
tidy_macrofauna_foliar_1$block[tidy_macrofauna_foliar_1$block == '2'] <- 'B2'
tidy_macrofauna_foliar_1$block[tidy_macrofauna_foliar_1$block == '3'] <- 'B3'

# create column code_uhbr
tidy_macrofauna_foliar_1$code_uhbr <- paste(tidy_macrofauna_foliar_1$usage, tidy_macrofauna_foliar_1$habitat, tidy_macrofauna_foliar_1$block, tidy_macrofauna_foliar_1$replicate, sep = "_")

# rename name to finest_name
tidy_macrofauna_foliar_1 = rename(tidy_macrofauna_foliar_1, finest_name = name)
```

## nematodes

### march

```{r}
nematodes_m_0 <- nematodes_0[15:27,2:26]
nematodes_m_0 <- nematodes_m_0 %>% row_to_names(row_number = 1)
nematodes_m_0 <- nematodes_m_0 %>% mutate_at(-c(1,2), as.numeric) %>% mutate_if(is.numeric, round, digits=3)

# add columns replicate, date, method and community
nematodes_m_0 <- nematodes_m_0 %>% mutate(replicate = "R1")
nematodes_m_0 <- nematodes_m_0 %>% mutate(date = "14/03/2023")
nematodes_m_0 <- nematodes_m_0 %>% mutate(method = "baermann")
nematodes_m_0 <- nematodes_m_0 %>% mutate(community = "nematodes")

# homogeneise code
nematodes_m_0 <- nematodes_m_0 %>%
  mutate(
    usage = case_when(
      Taxa == "Tree plantation" ~ "TP",
      Taxa == "Annual crop" ~ "MC",
      Taxa == "Under tree in AF" |  Taxa == "Under annual crop in AF" ~ "AF"),
    habitat = case_when(
      Taxa == "Tree plantation" | Taxa == "Under tree in AF" ~ "T",
      Taxa == "Annual crop" | Taxa == "Under annual crop in AF" ~ "C"),
    block = paste0("B", parse_number(Block))
    )

# create column code_uhbr
nematodes_m_0$code_uhbr <- paste(nematodes_m_0$usage, nematodes_m_0$habitat, nematodes_m_0$block, nematodes_m_0$replicate, sep="_")
```

### april

```{r}
nematodes_a_0 <- nematodes_0[1:13,2:32]
nematodes_a_0 <- nematodes_a_0 %>% row_to_names(row_number = 1)
nematodes_a_0 <- nematodes_a_0 %>% mutate_at(-c(1,2), as.numeric) %>% mutate_if(is.numeric, round, digits=3)

# add columns replicate, date, method and community
nematodes_a_0 <- nematodes_a_0 %>% mutate(replicate = "R1")
nematodes_a_0 <- nematodes_a_0 %>% mutate(date = "17/04/2023")
nematodes_a_0 <- nematodes_a_0 %>% mutate(method = "baermann")
nematodes_a_0 <- nematodes_a_0 %>% mutate(community = "nematodes")

# homogeneise code
nematodes_a_0 <- nematodes_a_0 %>%
  mutate(
    usage = case_when(
      Taxa == "Tree plantation" ~ "TP",
      Taxa == "Annual crop" ~ "MC",
      Taxa == "Under tree in AF" |  Taxa == "Under annual crop in AF" ~ "AF"),
    habitat = case_when(
      Taxa == "Tree plantation" | Taxa == "Under tree in AF" ~ "T",
      Taxa == "Annual crop" | Taxa == "Under annual crop in AF" ~ "C"),
    block = paste0("B", parse_number(Block))
    )

# create column code_uhbr
nematodes_a_0$code_uhbr <- paste(nematodes_a_0$usage, nematodes_a_0$habitat, nematodes_a_0$block, nematodes_a_0$replicate, sep="_")
```

### all

```{r}
# bind and pivot data
tidy_nematodes_1 <- bind_rows(nematodes_a_0[,-c(1,2)], nematodes_m_0[,-c(1,2)])
tidy_nematodes_1 <- tidy_nematodes_1 %>% replace(is.na(.), 0)
tidy_nematodes_1 <- tidy_nematodes_1 %>% select(code_uhbr, usage, habitat, block, replicate, date, method, community, everything())
tidy_nematodes_1 <- pivot_longer(tidy_nematodes_1, cols = c(9:38), names_to = "taxa", values_to = "abundance")
tidy_nematodes_1 <- tidy_nematodes_1 %>% rename(finest_name=taxa)
tidy_nematodes_1$abundance <- lapply(tidy_nematodes_1$abundance, round)
```

## micro_arthropodes

```{r}
micro_arthropodes_all_1 <- micro_arthropodes_all_0

# delete 3 last rows (they are not samples)
micro_arthropodes_all_1 <- micro_arthropodes_all_1 %>% slice(-c(346:348))

# remplace NA by 0 for abundance
micro_arthropodes_all_1 <- micro_arthropodes_all_1 %>%
  mutate(Nbr_identifiés = ifelse(is.na(Nbr_identifiés), 0, Nbr_identifiés),
         Nbr_ind_tri = ifelse(is.na(Nbr_ind_tri), 0, Nbr_ind_tri))

# add column Nbr_individus_final = Nbr_ind_tri - Nbr_identifiés
## Nbr_ind_tri is the total number of individuals
## Nbr_identifiés is the number of collembola more finely identified
## Nbr_individus_final is needed because not all collembola have been identified
micro_arthropodes_all_1$Nbr_ind_tri <- as.numeric(micro_arthropodes_all_1$Nbr_ind_tri)
micro_arthropodes_all_1$Nbr_identifiés <- as.numeric(micro_arthropodes_all_1$Nbr_identifiés) 

micro_arthropodes_all_1 <- micro_arthropodes_all_1 %>% mutate(Nbr_individus_final = Nbr_ind_tri - Nbr_identifiés)

# add collembola identified
micro_arthropodes_all_1 <- micro_arthropodes_all_1 %>% 
  rename(Sites=Num_prélevement)

tidy_micro_arthropodes_1 <- bind_rows(micro_arthropodes_all_1, micro_arthropodes_col_0)

# add a unique column abundance
tidy_micro_arthropodes_1 <- tidy_micro_arthropodes_1 %>% mutate(abundance = coalesce(Nbr_individus_final, Nbr_individus))
tidy_micro_arthropodes_1 <- tidy_micro_arthropodes_1 %>% mutate(finest_name = coalesce(Noms_espèces, best_ID))

# add columns date, method and community
tidy_micro_arthropodes_1 <- tidy_micro_arthropodes_1 %>% mutate(date = "17/04/2023")
tidy_micro_arthropodes_1 <- tidy_micro_arthropodes_1 %>% mutate(method = "macfayden")
tidy_micro_arthropodes_1 <- tidy_micro_arthropodes_1 %>% mutate(community = "micro_arthropodes")

# homogeneise code
tidy_micro_arthropodes_1 <- tidy_micro_arthropodes_1 %>%
 mutate(usage = case_when(
    grepl('C[[:digit:]]_', Sites) ~ "MC",
    grepl('AF[[:digit:]]_', Sites) ~ "AF",
    grepl('F[[:digit:]]_', Sites) ~ "TP"),
        habitat = case_when(
    grepl('C[[:digit:]]_|AF[[:digit:]]_C', Sites) ~ "C",
    grepl('F[[:digit:]]_|AF[[:digit:]]_LSA', Sites) ~ "T"),
        block = case_when(
    grepl('1_', Sites) ~ "B1",
    grepl('2_', Sites) ~ "B2",
    grepl('3_', Sites) ~ "B3"),
        replicate = case_when(
    grepl('_C1|_LSA1', Sites) ~ "R1",
    grepl('_C2|_LSA2', Sites) ~ "R2",
    grepl('_C3|_LSA3', Sites) ~ "R3",
    grepl('_C4|_LSA4', Sites) ~ "R4",
    grepl('_C5|_LSA5', Sites) ~ "R5"))

# create column code_uhbr
tidy_micro_arthropodes_1$code_uhbr <- paste(tidy_micro_arthropodes_1$usage, tidy_micro_arthropodes_1$habitat, tidy_micro_arthropodes_1$block, tidy_micro_arthropodes_1$replicate, sep="_")
```

## microorganisms

```{r}
metadata_mo_1 <- metadata_mo_0

# rename columns
metadata_mo_1 <- metadata_mo_1[c(1:60),] %>%  rename(land_use = usage, block = Bloc, replicate = REP, habitat = HABITAT)

# add columns date and method
metadata_mo_1 <- metadata_mo_1 %>% mutate(date = "17/04/2023")
metadata_mo_1 <- metadata_mo_1 %>% mutate(method = "metabarcoding")

# homogeneise code
metadata_mo_1$land_use[metadata_mo_1$land_use == 'C'] <- 'MC'
metadata_mo_1$land_use[metadata_mo_1$land_use == 'F'] <- 'TP'

metadata_mo_1$block <- paste0("B", parse_number(metadata_mo_1$block))

metadata_mo_1$habitat[metadata_mo_1$habitat == 'Tree_line' | metadata_mo_1$habitat == 'Tree_plantation' ] <- 'T'
metadata_mo_1$habitat[metadata_mo_1$habitat == 'Crop' ] <- 'C'

metadata_mo_1$replicate <- paste0("R", parse_number(metadata_mo_1$replicate))

# create column code_uhbr
metadata_mo_1$code_uhbr <- paste(metadata_mo_1$land_use, metadata_mo_1$habitat, metadata_mo_1$block, metadata_mo_1$replicate, sep = "_")
metadata_mo_1$code_uh <- paste(metadata_mo_1$land_use, metadata_mo_1$habitat, sep = "_")
metadata_mo_1 <- metadata_mo_1[,-c(4,7,8)]
metadata_mo_1 <- metadata_mo_1[,c(1,2,4,3,5,6,7,8,9)]
```

### bacteria

```{r}
# Tidy metadata
tidy_bacteria_1 <- bacteria_0

tidy_bacteria_1 <- pivot_longer(tidy_bacteria_1, cols = c(13:72), names_to =  "Reference_ADNid", values_to = "abundance", names_repair = "minimal")

tidy_bacteria_1$Reference_ADNid <- str_replace(tidy_bacteria_1$Reference_ADNid, '_.+_.+', "")

tidy_bacteria_1 <- left_join(tidy_bacteria_1, metadata_mo_1, by = "Reference_ADNid")

tidy_bacteria_1 <- tidy_bacteria_1 %>% mutate(community = "bacteria")
```

```{r}
# Check raw data
test <- tidy_bacteria_1 # We have 3557700 rows
test_OTU <- unique(test[c("observation_name")]) # that correspond to 59295 OTUs (number of rows * 60 samples)
test_blast <- unique(test[c("blast_taxonomy")]) # that correspond to 3715 blast_taxonomy assignations
test_rdp <- unique(test[c("rdp_tax_and_bootstrap")]) # that correspond to 32968 rdp_tax_and_bootstrap assignations
test_abund <- sum(test$abundance) # that make a total abundance of 3858868

tracking_raw_data <- tibble(community='bacteria', step='raw data', observation_number=nrow(test), OTU_number=nrow(test_OTU), blast_number=nrow(test_blast), rdp_number=nrow(test_rdp), total_abundance=sum(test$abundance))

# cleaning step 1
## Some OTUs have no blast taxonomy results ("no data") supposedly because the match between their sequence and the RDP database is not strong enough.
test <- subset(tidy_bacteria_1, blast_taxonomy == "no data")

tracking_no_blast_results <- tibble(community='bacteria', step='no blast data', observation_number=nrow(test), OTU_number=nrow(unique(test[c("observation_name")])), blast_number=nrow(unique(test[c("blast_taxonomy")])), rdp_number=nrow(unique(test[c("rdp_tax_and_bootstrap")])), total_abundance=sum(test$abundance))

tidy_bacteria_1a <- tidy_bacteria_1 %>% mutate(full_name = blast_taxonomy)
tidy_bacteria_1a$full_name <- str_replace(tidy_bacteria_1a$full_name, "no data", "unassigned")
test_final <- subset(tidy_bacteria_1a, full_name == "no data")  # OK (no more "no data")

test <- tidy_bacteria_1a
tracking_1a <- tibble(community='bacteria', step='blast results', observation_number=nrow(test), OTU_number=nrow(unique(test[c("observation_name")])), blast_number=nrow(unique(test[c("blast_taxonomy")])), rdp_number=nrow(unique(test[c("rdp_tax_and_bootstrap")])), total_abundance=sum(test$abundance))

# cleaning step 2
## full_names have to be separated and cleaned to keep only assigned taxa full_names (remove "Multi-affiliation", "unidentified" in species, "unknown class/order/family/genus/species" and "*metagenome")

tidy_bacteria_2a <- tidy_bacteria_1a

tidy_bacteria_2a <- tidy_bacteria_2a %>% 
   separate(col = full_name, into = c("kingdom","phylum","class","order", "family", "genus", "species"), sep = ";")

tidy_bacteria_2a <- tidy_bacteria_2a %>% replace(is.na(.), "unassigned")

tidy_bacteria_2a <- tidy_bacteria_2a %>% mutate(across(c(everything()), ~ sub("Multi-affiliation|^unidentified.*|^unknown.*|metagenome", "unassigned", .)))

tidy_bacteria_2a <- tidy_bacteria_2a %>%  mutate(species = str_replace_all(species, ".*bacterium$|.*sp.$|.*Micromonosporaceae.*|.*soil unassigned.*", "unassigned"))

tidy_bacteria_2a$full_name_cleaned <- paste(tidy_bacteria_2a$kingdom, tidy_bacteria_2a$phylum, tidy_bacteria_2a$class, tidy_bacteria_2a$order, tidy_bacteria_2a$family, tidy_bacteria_2a$genus, tidy_bacteria_2a$species, sep = ";")

test_final <- filter(tidy_bacteria_2a, grepl("Multi-affiliation|unknown|unidentified|metagenome", full_name_cleaned)) # OK (no remaining uncleaned taxa)

tidy_bacteria_2a$abundance <- as.numeric(tidy_bacteria_2a$abundance)
test <- tidy_bacteria_2a
tracking_2a <- tibble(community='bacteria', step='blast results_cleaned', observation_number=nrow(test), OTU_number=nrow(unique(test[c("observation_name")])), blast_number=nrow(unique(test[c("blast_taxonomy")])), rdp_number=nrow(unique(test[c("rdp_tax_and_bootstrap")])), total_abundance=sum(test$abundance))

# cleaning step 3 = filtering

tidy_bacteria_3a <- tidy_bacteria_2a 

### Filter 1.1 = Bacteria
tidy_bacteria_3a.1.1 <- tidy_bacteria_3a[!(tidy_bacteria_3a$kingdom != 'Bacteria'),] # to keep only Bacteria

diff <- anti_join(tidy_bacteria_3a, tidy_bacteria_3a.1.1) 
test_final <- unique(diff[c("full_name_cleaned")]) # They correspond  to Archeae and 1 totally unassigned

test <- tidy_bacteria_3a.1.1
tracking_3a.1.1 <- tibble(community='bacteria', step='blast results_cleaned_filter 1.1 (Bacteria)', observation_number=nrow(test), OTU_number=nrow(unique(test[c("observation_name")])), blast_number=nrow(unique(test[c("blast_taxonomy")])), rdp_number=nrow(unique(test[c("rdp_tax_and_bootstrap")])), total_abundance=sum(test$abundance))

### Filter 1.2 = Chloroplasts
tidy_bacteria_3a.1.2 <- tidy_bacteria_3a.1.1[(tidy_bacteria_3a.1.1$order != 'Chloroplast'),] # to remove Chloroplasts

diff <- anti_join(tidy_bacteria_3a.1.1, tidy_bacteria_3a.1.2) 
test_final <- unique(diff[c("full_name_cleaned")]) # Many correspond to plants

test <- tidy_bacteria_3a.1.2
tracking_3a.1.2 <- tibble(community='bacteria', step='blast results_cleaned_filter 1.2 (Chloroplasts)', observation_number=nrow(test), OTU_number=nrow(unique(test[c("observation_name")])), blast_number=nrow(unique(test[c("blast_taxonomy")])), rdp_number=nrow(unique(test[c("rdp_tax_and_bootstrap")])), total_abundance=sum(test$abundance))

### Filter 2 = low abundant (cf Ogier et al., 2019)
total_abundance_per_sample <- aggregate(abundance ~ code_uhbr, data = tidy_bacteria_3a.1.2, FUN = sum)
colnames(total_abundance_per_sample)[2] <- "total_abondance"
tidy_bacteria_3a.2 <- merge(tidy_bacteria_3a.1.2, total_abundance_per_sample, by = "code_uhbr")
tidy_bacteria_3a.2$threshold <- tidy_bacteria_3a.2$total_abondance * 0.001
tidy_bacteria_3a.2 <- tidy_bacteria_3a.2[tidy_bacteria_3a.2$abundance >= tidy_bacteria_3a.2$threshold, ]

diff <- anti_join(tidy_bacteria_3a.1.2, tidy_bacteria_3a.2) 
test_final <- unique(diff[c("full_name_cleaned")])

test <- tidy_bacteria_3a.2
tracking_3a.2 <- tibble(community='bacteria', step='blast results_cleaned_filter 2 (low abundant)', observation_number=nrow(test), OTU_number=nrow(unique(test[c("observation_name")])), blast_number=nrow(unique(test[c("blast_taxonomy")])), rdp_number=nrow(unique(test[c("rdp_tax_and_bootstrap")])), total_abundance=sum(test$abundance))

# cleaning step 4
## The last assigned level will be kept as "finest_name"

tidy_bacteria_4a <- tidy_bacteria_3a.2

tidy_bacteria_4a$finest_name <- apply(tidy_bacteria_4a[,c("species", "genus", "family", "order", "class", "phylum", "kingdom")], 1, function(x) {
  finest_name <- x[x != "unassigned"]
  if (length(finest_name) > 0) {
    return(finest_name[1])
  } else {
    return("unassigned")
  }
})

test <- tidy_bacteria_4a
tracking_4a <- tibble(community='bacteria', step='blast results_cleaned_filtered_finest name', observation_number=nrow(test), OTU_number=nrow(unique(test[c("observation_name")])), blast_number=nrow(unique(test[c("blast_taxonomy")])), rdp_number=nrow(unique(test[c("rdp_tax_and_bootstrap")])), total_abundance=sum(test$abundance))

# Exporting tracking and taxonomy (cf 2. Taxonomy table)
tracking_bacteria <- rbind(tracking_raw_data, tracking_no_blast_results, tracking_1a, tracking_2a, tracking_3a.1.1, tracking_3a.1.2, tracking_3a.2, tracking_4a)
#write.csv(tracking_bacteria, "analyses/tracking/bacteria_2024-09-13.csv", row.names = FALSE)

tidy_bacteria_4a$finest_name <- paste(tidy_bacteria_4a$finest_name, tidy_bacteria_4a$observation_name, sep = "::")
tidy_bacteria_5 <- tidy_bacteria_4a %>% 
  select(code_uhbr, finest_name, abundance, date, method, community)
#write.csv(tidy_bacteria_5, "data/derived_data/tidy_data/tidy_bacteria_blast_cleaned_filtered_2024-09-13.csv", row.names = FALSE)
#saveRDS(tidy_bacteria_5, "data/derived_data/tidy_data/tidy_bacteria_blast_cleaned_filtered_2024-09-13.rds")

tidy_bacteria <- tidy_bacteria_4a # blast_taxonomy results
```

### fungi

```{r}
# Tidy metadata
tidy_fungi_1 <- fungi_0

tidy_fungi_1 <- pivot_longer(tidy_fungi_1, cols = c(13:72), names_to =  "Reference_ADNid", values_to = "abundance", names_repair = "minimal")

tidy_fungi_1$Reference_ADNid <- str_replace(tidy_fungi_1$Reference_ADNid, '_.+_.+', "")

tidy_fungi_1 <- left_join(tidy_fungi_1, metadata_mo_1, by = "Reference_ADNid")

tidy_fungi_1 <- tidy_fungi_1 %>% mutate(community = "fungi")
```

```{r}
# Check raw data
test <- tidy_fungi_1 # We have 202500 rows
test_OTU <- unique(test[c("observation_name")]) # that correspond to 3375 OTUs (number of rows * 60 samples)
test_blast <- unique(test[c("blast_taxonomy")]) # that correspond to 960 blast_taxonomy assignations
test_rdp <- unique(test[c("rdp_tax_and_bootstrap")]) # that correspond to 2833 rdp_tax_and_bootstrap assignations
test_abund <- sum(test$abundance) # that make a total abundance of 1263924

tracking_raw_data <- tibble(community='fungi', step='raw data', observation_number=nrow(test), OTU_number=nrow(test_OTU), blast_number=nrow(test_blast), rdp_number=nrow(test_rdp), total_abundance=sum(test$abundance))

# cleaning step 1
## Some OTUs have no blast taxonomy results ("no data") supposedly because the match between their sequence and the RDP database is not strong enough.
test <- subset(tidy_fungi_1, blast_taxonomy == "no data")

tracking_no_blast_results <- tibble(community='fungi', step='no blast data', observation_number=nrow(test), OTU_number=nrow(unique(test[c("observation_name")])), blast_number=nrow(unique(test[c("blast_taxonomy")])), rdp_number=nrow(unique(test[c("rdp_tax_and_bootstrap")])), total_abundance=sum(test$abundance))

tidy_fungi_1a <- tidy_fungi_1 %>% mutate(full_name = blast_taxonomy)
tidy_fungi_1a$full_name <- str_replace(tidy_fungi_1a$full_name, "no data", "unassigned")
test_final <- subset(tidy_fungi_1a, full_name == "no data")  # OK (no more "no data")

test <- tidy_fungi_1a
tracking_1a <- tibble(community='fungi', step='blast results', observation_number=nrow(test), OTU_number=nrow(unique(test[c("observation_name")])), blast_number=nrow(unique(test[c("blast_taxonomy")])), rdp_number=nrow(unique(test[c("rdp_tax_and_bootstrap")])), total_abundance=sum(test$abundance))

# cleaning step 2
## full_names have to be separated and cleaned to keep only assigned taxa full_names (remove "Multi-affiliation", "unidentified" in species, "unknown class/order/family/genus/species" and "*metagenome")

tidy_fungi_2a <- tidy_fungi_1a

tidy_fungi_2a <- tidy_fungi_2a %>% 
   separate(col = full_name, into = c("kingdom","phylum","class","order", "family", "genus", "species"), sep = ";")

tidy_fungi_2a[] <- lapply(tidy_fungi_2a, gsub, pattern='.__', replacement='')

tidy_fungi_2a <- tidy_fungi_2a %>% replace(is.na(.), "unassigned")

tidy_fungi_2a <- tidy_fungi_2a %>% mutate(across(c(everything()), ~ sub("Multi-affiliation|^unidentified.*|^unknown.*|metagenome|.*_sp$", "unassigned", .)))

tidy_fungi_2a <- tidy_fungi_2a %>% mutate(across(24:30, ~ sub("_", " ", .)))

tidy_fungi_2a$full_name_cleaned <- paste(tidy_fungi_2a$kingdom, tidy_fungi_2a$phylum, tidy_fungi_2a$class, tidy_fungi_2a$order, tidy_fungi_2a$family, tidy_fungi_2a$genus, tidy_fungi_2a$species, sep = ";")

test_final <- filter(tidy_fungi_2a, grepl("Multi-affiliation|unknown|unidentified|metagenome", full_name_cleaned)) # OK (no remaining uncleaned taxa)

tidy_fungi_2a$abundance <- as.numeric(tidy_fungi_2a$abundance)
test <- tidy_fungi_2a
tracking_2a <- tibble(community='fungi', step='blast results_cleaned', observation_number=nrow(test), OTU_number=nrow(unique(test[c("observation_name")])), blast_number=nrow(unique(test[c("blast_taxonomy")])), rdp_number=nrow(unique(test[c("rdp_tax_and_bootstrap")])), total_abundance=sum(test$abundance))

# cleaning step 3 = filtering

tidy_fungi_3a <- tidy_fungi_2a 

### Filter 1 = Fungi
tidy_fungi_3a.1 <- tidy_fungi_3a[!(tidy_fungi_3a$kingdom != 'Fungi'),] # to keep only Fungi

diff <- anti_join(tidy_fungi_3a, tidy_fungi_3a.1) 
test_final <- unique(diff[c("full_name_cleaned")]) # They correspond to totally unassigned

tidy_fungi_3a.1$abundance <- as.numeric(tidy_fungi_3a.1$abundance)
test <- tidy_fungi_3a.1
tracking_3a.1 <- tibble(community='fungi', step='blast results_cleaned_filter 1 (Fungi)', observation_number=nrow(test), OTU_number=nrow(unique(test[c("observation_name")])), blast_number=nrow(unique(test[c("blast_taxonomy")])), rdp_number=nrow(unique(test[c("rdp_tax_and_bootstrap")])), total_abundance=sum(test$abundance))

### Filter 2 = low abundant (cf Ogier et al., 2019)
total_abundance_per_sample <- aggregate(abundance ~ code_uhbr, data = tidy_fungi_3a.1, FUN = sum)
colnames(total_abundance_per_sample)[2] <- "total_abondance"
tidy_fungi_3a.2 <- merge(tidy_fungi_3a.1, total_abundance_per_sample, by = "code_uhbr")
tidy_fungi_3a.2$threshold <- tidy_fungi_3a.2$total_abondance * 0.001
tidy_fungi_3a.2 <- tidy_fungi_3a.2[tidy_fungi_3a.2$abundance >= tidy_fungi_3a.2$threshold, ]

diff <- anti_join(tidy_fungi_3a.1, tidy_fungi_3a.2) 
test_final <- unique(diff[c("full_name_cleaned")]) # They correspond mainly to Ascomycota

test <- tidy_fungi_3a.2
tracking_3a.2 <- tibble(community='fungi', step='blast results_cleaned_filter 2 (low abundant)', observation_number=nrow(test), OTU_number=nrow(unique(test[c("observation_name")])), blast_number=nrow(unique(test[c("blast_taxonomy")])), rdp_number=nrow(unique(test[c("rdp_tax_and_bootstrap")])), total_abundance=sum(test$abundance))

# cleaning step 4
## The last assigned level will be kept as "finest_name"

tidy_fungi_4a <- tidy_fungi_3a.2

tidy_fungi_4a$finest_name <- apply(tidy_fungi_4a[,c("species", "genus", "family", "order", "class", "phylum", "kingdom")], 1, function(x) {
  finest_name <- x[x != "unassigned"]
  if (length(finest_name) > 0) {
    return(finest_name[1])
  } else {
    return("unassigned")
  }
})

test <- tidy_fungi_4a
tracking_4a <- tibble(community='fungi', step='blast results_cleaned_filtered_finest name', observation_number=nrow(test), OTU_number=nrow(unique(test[c("observation_name")])), blast_number=nrow(unique(test[c("blast_taxonomy")])), rdp_number=nrow(unique(test[c("rdp_tax_and_bootstrap")])), total_abundance=sum(test$abundance))

# Exporting tracking and taxonomy (cf 2. Taxonomy table)
tracking_fungi <- rbind(tracking_raw_data, tracking_no_blast_results, tracking_1a, tracking_2a, tracking_3a.1, tracking_3a.2, tracking_4a)
#write.csv(tracking_fungi, "analyses/tracking/fungi_2024-09-13.csv", row.names = FALSE)

tidy_fungi_4a$finest_name <- paste(tidy_fungi_4a$finest_name, tidy_fungi_4a$observation_name, sep = "::")
tidy_fungi_5 <- tidy_fungi_4a %>% 
  select(code_uhbr, finest_name, abundance, date, method, community)
#write.csv(tidy_fungi_5, "data/derived_data/tidy_data/tidy_fungi_blast_cleaned_filtered_2024-09-13.csv", row.names = FALSE)
#saveRDS(tidy_fungi_5, "data/derived_data/tidy_data/tidy_fungi_blast_cleaned_filtered_2024-09-13.rds")

tidy_fungi <- tidy_fungi_4a # blast_taxonomy results
```

### protists

```{r}
# Tidy metadata
tidy_protists_1 <- protists_0

tidy_protists_1 <- pivot_longer(tidy_protists_1, cols = c(13:72), names_to =  "Reference_ADNid", values_to = "abundance", names_repair = "minimal")

tidy_protists_1$Reference_ADNid <- str_replace(tidy_protists_1$Reference_ADNid, '_.+_.+', "")

tidy_protists_1 <- left_join(tidy_protists_1, metadata_mo_1, by = "Reference_ADNid")

tidy_protists_1 <- tidy_protists_1 %>% mutate(community = "protists")
```

```{r}
# Check raw data
test <- tidy_protists_1 # We have 581820 rows
test_OTU <- unique(test[c("observation_name")]) # that correspond to 9697 OTUs (number of rows * 60 samples)
test_blast <- unique(test[c("blast_taxonomy")]) # that correspond to 252 blast_taxonomy assignations
test_rdp <- unique(test[c("rdp_tax_and_bootstrap")]) # that correspond to 6221 rdp_tax_and_bootstrap assignations
test_abund <- sum(test$abundance) # that make a total abundance of 1383974

tracking_raw_data <- tibble(community='protists', step='raw data', observation_number=nrow(test), OTU_number=nrow(test_OTU), blast_number=nrow(test_blast), rdp_number=nrow(test_rdp), total_abundance=sum(test$abundance))

# cleaning step 1
## Some OTUs have no blast taxonomy results ("no data") supposedly because the match between their sequence and the RDP database is not strong enough.
test <- subset(tidy_protists_1, blast_taxonomy == "no data")

tracking_no_blast_results <- tibble(community='protists', step='no blast data', observation_number=nrow(test), OTU_number=nrow(unique(test[c("observation_name")])), blast_number=nrow(unique(test[c("blast_taxonomy")])), rdp_number=nrow(unique(test[c("rdp_tax_and_bootstrap")])), total_abundance=sum(test$abundance))

tidy_protists_1a <- tidy_protists_1 %>% mutate(full_name = blast_taxonomy)
tidy_protists_1a$full_name <- str_replace(tidy_protists_1a$full_name, "no data", "unassigned")
test_final <- subset(tidy_protists_1a, full_name == "no data")  # OK (no more "no data")

test <- tidy_protists_1a
tracking_1a <- tibble(community='protists', step='blast results', observation_number=nrow(test), OTU_number=nrow(unique(test[c("observation_name")])), blast_number=nrow(unique(test[c("blast_taxonomy")])), rdp_number=nrow(unique(test[c("rdp_tax_and_bootstrap")])), total_abundance=sum(test$abundance))

# cleaning step 2
## full_names have to be separated and cleaned to keep only assigned taxa full_names (remove "Multi-affiliation", "unidentified" in species, "unknown class/order/family/genus/species" and "*metagenome")

tidy_protists_2a <- tidy_protists_1a

tidy_protists_2a <- tidy_protists_2a %>% 
   separate(col = full_name, into = c("kingdom","phylum","class","order", "family", "genus", "species"), sep = ";")

tidy_protists_2a <- tidy_protists_2a %>% replace(is.na(.), "unassigned")

tidy_protists_2a <- tidy_protists_2a %>% mutate(across(c(everything()), ~ sub("Multi-affiliation|^unidentified.*|^unknown.*|metagenome", "unassigned", .)))

tidy_protists_2a <- tidy_protists_2a %>%  mutate(species = str_replace_all(species, "Cercomonas sp.|Cercozoa sp.|Eocercomonas sp.|Heteromita sp.|Paracercomonas sp.|Thaumatomastix sp.", "unassigned")) # These species names with sp. are replaced because they already have the info in genus column. They others, that don't have it, are kept.

tidy_protists_2a$full_name_cleaned <- paste(tidy_protists_2a$kingdom, tidy_protists_2a$phylum, tidy_protists_2a$class, tidy_protists_2a$order, tidy_protists_2a$family, tidy_protists_2a$genus, tidy_protists_2a$species, sep = ";")

test_final <- filter(tidy_protists_2a, grepl("Multi-affiliation|unknown|unidentified|metagenome", full_name_cleaned)) # OK (no remaining uncleaned taxa)

tidy_protists_2a$abundance <- as.numeric(tidy_protists_2a$abundance)
test <- tidy_protists_2a
tracking_2a <- tibble(community='protists', step='blast results_cleaned', observation_number=nrow(test), OTU_number=nrow(unique(test[c("observation_name")])), blast_number=nrow(unique(test[c("blast_taxonomy")])), rdp_number=nrow(unique(test[c("rdp_tax_and_bootstrap")])), total_abundance=sum(test$abundance))

# cleaning step 3 = filtering

tidy_protists_3a <- tidy_protists_2a 

### Filter 1 = Cercozoa
tidy_protists_3a.1 <- tidy_protists_3a[!(tidy_protists_3a$phylum != 'Cercozoa'),] # to keep only Cercozoa

diff <- anti_join(tidy_protists_3a, tidy_protists_3a.1) 
test_final <- unique(diff[c("full_name_cleaned")]) # They correspond mainly to Nematozoa

test <- tidy_protists_3a.1
tracking_3a.1 <- tibble(community='protists', step='blast results_cleaned_filter 1 (Cercozoa)', observation_number=nrow(test), OTU_number=nrow(unique(test[c("observation_name")])), blast_number=nrow(unique(test[c("blast_taxonomy")])), rdp_number=nrow(unique(test[c("rdp_tax_and_bootstrap")])), total_abundance=sum(test$abundance))

### Filter 2 = low abundant (cf Ogier et al., 2019)
total_abundance_per_sample <- aggregate(abundance ~ code_uhbr, data = tidy_protists_3a.1, FUN = sum)
colnames(total_abundance_per_sample)[2] <- "total_abondance"
tidy_protists_3a.2 <- merge(tidy_protists_3a.1, total_abundance_per_sample, by = "code_uhbr")
tidy_protists_3a.2$threshold <- tidy_protists_3a.2$total_abondance * 0.001
tidy_protists_3a.2 <- tidy_protists_3a.2[tidy_protists_3a.2$abundance >= tidy_protists_3a.2$threshold, ]

diff <- anti_join(tidy_protists_3a.1, tidy_protists_3a.2) 
test_final <- unique(diff[c("full_name_cleaned")])

test <- tidy_protists_3a.2
tracking_3a.2 <- tibble(community='protists', step='blast results_cleaned_filter 2 (low abundant)', observation_number=nrow(test), OTU_number=nrow(unique(test[c("observation_name")])), blast_number=nrow(unique(test[c("blast_taxonomy")])), rdp_number=nrow(unique(test[c("rdp_tax_and_bootstrap")])), total_abundance=sum(test$abundance))

# cleaning step 4
## The last assigned level will be kept as "finest_name"

tidy_protists_4a <- tidy_protists_3a.2

tidy_protists_4a$finest_name <- apply(tidy_protists_4a[,c("species", "genus", "family", "order", "class", "phylum", "kingdom")], 1, function(x) {
  finest_name <- x[x != "unassigned"]
  if (length(finest_name) > 0) {
    return(finest_name[1])
  } else {
    return("unassigned")
  }
})

test <- tidy_protists_4a
tracking_4a <- tibble(community='protists', step='blast results_cleaned_filtered_finest name', observation_number=nrow(test), OTU_number=nrow(unique(test[c("observation_name")])), blast_number=nrow(unique(test[c("blast_taxonomy")])), rdp_number=nrow(unique(test[c("rdp_tax_and_bootstrap")])), total_abundance=sum(test$abundance))

# Exporting tracking and taxonomy (cf 2. Taxonomy table)
tracking_protists <- rbind(tracking_raw_data, tracking_no_blast_results, tracking_1a, tracking_2a, tracking_3a.1, tracking_3a.2, tracking_4a)
#write.csv(tracking_protists, "analyses/tracking/protists_2024-09-13.csv", row.names = FALSE)

tidy_protists_4a$finest_name <- paste(tidy_protists_4a$finest_name, tidy_protists_4a$observation_name, sep = "::")
tidy_protists_5 <- tidy_protists_4a %>% 
  select(code_uhbr, finest_name, abundance, date, method, community)
#write.csv(tidy_protists_5, "data/derived_data/tidy_data/tidy_protists_blast_cleaned_filtered_2024-09-13.csv", row.names = FALSE)
#saveRDS(tidy_protists_5, "data/derived_data/tidy_data/tidy_protists_blast_cleaned_filtered_2024-09-13.rds")

tidy_protists <- tidy_protists_4a # blast_taxonomy results
```

At this step, the files with cleaned data (suffix _1) are:
  macrofauna_surface
  macrofauna_aerial
  macrofauna_foliar
  nematodes
  micro_arthropodes
  bacteria
  fungi
  protists

# Export tidy_data

```{r}
# Select minimal data to export

tidy_macrofauna_surface_2 <- tidy_macrofauna_surface_1 %>% 
  select(code_uhbr, finest_name, abundance, date, method, community)
# stade and mass can be used for traits
# fullName, genusName, familyName, orderName, className, phylumName will be regenerated in homogenized format during the tax_table section
#write.csv(tidy_macrofauna_surface_2, "data/derived_data/tidy_data/tidy_macrofauna_surface_2024-09-13.csv", row.names = FALSE)

tidy_macrofauna_aerial_2 <- tidy_macrofauna_aerial_1 %>% 
  select(code_uhbr, finest_name, abundance, date, method, community)
# INat can be used for ID and stade can be used for traits
# fullName, genusName, familyName, orderName, className, phylumName will be regenerated in homogenized format during the tax_table section
#write.csv(tidy_macrofauna_aerial_2, "data/derived_data/tidy_data/tidy_macrofauna_aerial_2024-09-13.csv", row.names = FALSE)

tidy_macrofauna_foliar_2 <- tidy_macrofauna_foliar_1 %>% 
  select(code_uhbr, finest_name, abundance, date, method, community)
# Valid.Name (what is it?)
# fullName, genusName, familyName, orderName, className, phylumName will be regenerated in homogenized format during the tax_table section
#write.csv(tidy_macrofauna_foliar_2, "data/derived_data/tidy_data/tidy_macrofauna_foliar_2024-09-13.csv", row.names = FALSE)

tidy_nematodes_2 <- tidy_nematodes_1 %>% select(code_uhbr, finest_name, abundance, date, method, community)
tidy_nematodes_2$abundance = as.numeric(tidy_nematodes_2$abundance)
#write.csv(tidy_nematodes_2, "data/derived_data/tidy_data/tidy_nematodes_2024-09-13.csv", row.names = FALSE)

tidy_micro_arthropodes_2 <- tidy_micro_arthropodes_1 %>% select(code_uhbr, finest_name, abundance, date, method, community)
# stade can be used for traits
# higher.level and lower.level will be regenerated in another tax_table
#write.csv(tidy_micro_arthropodes_2, "data/derived_data/tidy_data/tidy_micro_arthropodes_2024-09-13.csv", row.names = FALSE)
```

```{r}
tidy_macrofauna_surface <- read.csv("data/derived_data/tidy_data/tidy_macrofauna_surface_2024-09-13.csv")
tidy_macrofauna_aerial <- read.csv("data/derived_data/tidy_data/tidy_macrofauna_aerial_2024-09-13.csv")
tidy_macrofauna_foliar <- read.csv("data/derived_data/tidy_data/tidy_macrofauna_foliar_2024-09-13.csv")
tidy_micro_arthropodes <- read.csv("data/derived_data/tidy_data/tidy_micro_arthropodes_2024-09-13.csv")
tidy_nematodes <- read.csv("data/derived_data/tidy_data/tidy_nematodes_2024-09-13.csv")

tidy_bacteria <- read.csv("data/derived_data/tidy_data/tidy_bacteria_blast_cleaned_filtered_2024-09-13.csv")
tidy_fungi <- read.csv("data/derived_data/tidy_data/tidy_fungi_blast_cleaned_filtered_2024-09-13.csv")
tidy_protists <- read.csv("data/derived_data/tidy_data/tidy_protists_blast_cleaned_filtered_2024-09-13.csv")
```

```{r}
tidy_ALL <- bind_rows(tidy_macrofauna_surface, tidy_macrofauna_aerial, tidy_macrofauna_foliar, tidy_nematodes, tidy_micro_arthropodes, tidy_bacteria, tidy_fungi, tidy_protists) # bind all communities in one file

tidy_ALL <- aggregate(tidy_ALL$abundance, by=list(code_uhbr=tidy_ALL$code_uhbr, finest_name=tidy_ALL$finest_name, date=tidy_ALL$date, method=tidy_ALL$method, community=tidy_ALL$community), FUN=sum) %>% rename(abundance = x) # aggregate abundances by code_uhbr, finest_name, date, method and community

tidy_ALL <- filter(tidy_ALL, abundance != 0) # remove rows with abundance = 0 (especially needed for amplicon barcoding since each sample has a row for all barcode/assigned taxa name)

#write.csv(tidy_ALL, "data/derived_data/tidy_data/tidy_all_2024-09-13.csv", row.names=F)
#saveRDS(tidy_ALL, "data/derived_data/tidy_data/tidy_all_2024-09-13.rds")
```

# Check tracking

## microorganismes

```{r}
# Import and combine track results
tracking_bacteria <- read.csv("analyses/tracking/bacteria_2024-09-13.csv")
tracking_fungi <- read.csv("analyses/tracking/fungi_2024-09-13.csv")
tracking_protists <- read.csv("analyses/tracking/protists_2024-09-13.csv")

tracking <- rbind(tracking_bacteria, tracking_fungi, tracking_protists)

#write.csv(tracking, "analyses/tracking/all_long_2024-09-13.csv", row.names = FALSE)

# Keep blast results
tracking <- filter(tracking, !grepl("rdp", step))

# Keep minimal steps
tracking <- filter(tracking, grepl("raw data|blast results_cleaned_filter 1|blast results_cleaned_filter 2", step))

# Keep last taxo filter for bacteria
tracking$step <- gsub("blast results_cleaned_filter 1.2", "blast results_cleaned_filter 1", tracking$step)
tracking <- filter(tracking, !grepl("1.1", step))

# Remove filter details in step name to homogeneize x axis
tracking$step <- gsub(" \\(.*?\\)", "", tracking$step)

#write.csv(tracking, "analyses/tracking/all_short_2024-09-13.csv", row.names = FALSE)

# Order x axis
order_step = c("raw data", "blast results_cleaned_filter 1", "blast results_cleaned_filter 2")
tracking$step <- ordered(tracking$step, order_step)

# Draw graphs
C1 <- ggplot(tracking, aes(step, OTU_number, fill=community)) + frame_simple + geom_bar(stat="identity", position="dodge") + theme(axis.text.x = element_text(angle = 45, hjust=1)) + theme(axis.text=element_text(size=8)) + labs(title="Number of OTUs at each cleaning step") + scale_y_continuous(labels = scales::comma)
C1
#ggsave("analyses/tracking/tracking_OTU_number_2024-09-13.png", width=4.5, height=4.5)

C2 <- ggplot(tracking, aes(step, total_abundance, fill=community)) + frame_simple + geom_bar(stat="identity", position="dodge") + theme(axis.text.x = element_text(angle = 45, hjust=1)) + theme(axis.text=element_text(size=8)) + labs(title="Total abundance of OTUs at each cleaning step") + scale_y_continuous(labels = scales::comma)
C2
#ggsave("analyses/tracking/tracking_total_abundance_2024-09-13.png", width=4.5, height=4.5)

C3 <- ggplot(tracking, aes(step, blast_number, fill=community)) + frame_simple + geom_bar(stat="identity", position="dodge") + theme(axis.text.x = element_text(angle = 45, hjust=1)) + theme(axis.text=element_text(size=8)) + labs(title="Number of BLAST assignations at each cleaning step") + scale_y_continuous(labels = scales::comma)
C3
#ggsave("analyses/tracking/tracking_blast_number_2024-09-13.png", width=5, height=4.5)
```

## all

```{r}
tidy_ALL <- read.csv("data/derived_data/tidy_data/tidy_all_2024-09-13.csv")

step_1 <- tidy_ALL %>% group_by(community) %>% summarise(total_abundance = sum(abundance), n_distinct_finest_name = n_distinct(finest_name), n_distinct_date = n_distinct(date), n_distinct_method = n_distinct(method)) %>% mutate(step = "step_1")

step_1$community <- ordered(step_1$community, order_community)

step_1 <- as.data.frame(step_1)
#write.csv(step_1, "analyses/tracking/step_1_2024-09-13.csv", row.names = F)

S1 <- ggplot(step_1, aes(n_distinct_finest_name, total_abundance)) + frame_simple + geom_point(aes(color = community, size = 4)) + labs(title="Total abundance of organisms = f(number of distinct finest taxa names)") + scale_y_continuous(labels = scales::comma) + scale_x_continuous(labels = scales::comma) + scale_colour_brewer(palette = "Set2") + guides(size = F)
S1
#ggsave("analyses/tracking/tracking_step_1_2024-09-13.png", width=7.1, height=4.5)

step_1 <- filter(step_1, !grepl("bacteria|fungi|protists|nematodes", community))

S1 <- ggplot(step_1, aes(n_distinct_finest_name, total_abundance)) + frame_simple + geom_point(aes(color = community, size = 4)) + labs(title="Total abundance of organisms = f(number of distinct finest taxa names)", subtitle = "Zoom on macrofauna and micro-arthropodes") + scale_y_continuous(labels = scales::comma) + scale_x_continuous(labels = scales::comma) + scale_colour_brewer(palette = "Set2") + guides(size = F)
S1
#ggsave("analyses/tracking/tracking_step_1_zoom_2024-09-13.png", width=7.1, height=4.5)
```

# 2. Taxonomy table

Files generated in this section:
prefix "tax_"
suffix "_1" = file to use for taxize input (query)
suffix "_2" = raw taxize output (ITIS and NCBI results)
suffix "_3" = cleaned taxonomy
suffix "_ALL" = all data combined (all communities)

To go faster while generating the taxonomy, use a NCBI entrez key.
First, generate a key cf https://ncbiinsights.ncbi.nlm.nih.gov/2017/11/02/new-api-keys-for-the-e-utilities/
Then, set your environment by doing once:
usethis::edit_r_environ()
ENTREZ_KEY='youractualkey' # To write in the environment

NB: Even with the key, error (400) appears but can be solved by using a loop.

## macrofauna_surface

```{r}
tax_macrofauna_surface_1 <- tidy_macrofauna_surface_2 %>% distinct(finest_name)
temp <- gnr_resolve(tax_macrofauna_surface_1$finest_name, preferred_data_sources = c(1,3,4,11,12), best_match_only = T) # 1 = CLC, 3 = ITIS, 4 = NCBI, 11 = GBIF, 12 = EOL (https://resolver.globalnames.org/data_sources)
diff <- setdiff(tax_macrofauna_surface_1$finest_name, temp$user_supplied_name) 
print(diff)
# Manual cleaning
## Pezottetix not found because wrong spelling
tax_macrofauna_surface_1$finest_name[tax_macrofauna_surface_1$finest_name == "Pezottetix"] <- "Pezotettix"
# Run again from temp... and check diff = 0 now

#write.csv(tax_macrofauna_surface_1, "data/derived_data/tax_table/intermediate_files/tax_macrofauna_surface_taxizeinput_2024-09-13.csv", row.names = F)
```

```{r, echo=FALSE}
tax_macrofauna_surface_2 = NULL
for (i in 1: nrow(tax_macrofauna_surface_1)) {
classes_i <- try(tax_name(sci = tax_macrofauna_surface_1$finest_name[i], get = c("kingdom", "phylum", "class", "order", "family", "genus", "species"), db = "both", ask = FALSE))
if (class(classes_i)=="try-error") {
Sys.sleep(10)
classes_i <- try(tax_name(sci = tax_macrofauna_surface_1$finest_name[i], get = c("kingdom", "phylum", "class", "order", "family", "genus", "species"), db = "both", ask = FALSE))}
tax_macrofauna_surface_2 <- rbind(tax_macrofauna_surface_2, classes_i)
}

tax_macrofauna_surface_2 <- tax_macrofauna_surface_2[- grep("Error", tax_macrofauna_surface_2$query),] # remove the rows with errors
#write.csv(tax_macrofauna_surface_2, "data/derived_data/tax_table/intermediate_files/tax_macrofauna_surface_taxizeoutput.csv")
```

```{r, echo=FALSE}
tax_macrofauna_surface_2 <- read.csv("data/derived_data/tax_table/intermediate_files/tax_macrofauna_surface_taxizeoutput.csv")
taxnotfoundboth <- tax_macrofauna_surface_2[!complete.cases(tax_macrofauna_surface_2$kingdom),]
taxnotfoundonedb <- taxnotfoundboth %>% group_by(query) %>% filter(n() == 1) # These are the taxa found in only one database (not found in the database in this object = found in the other database in macrofauna_foliar_2). Therefore, if we filter NCBI (by default) results in macrofauna_foliar_2, we must also keep ITIS results for the taxa with no NCBI results in this objects. These taxa to keep are saved in the following vector :
keepitis <- c(subset(taxnotfoundonedb, db == 'ncbi')$query)
taxnotfound_ncbianditis <- subset(taxnotfoundboth,duplicated(query)) # These are the taxa not found in any database. We will export the list, fill it manually and reimport it.
#write.csv(taxnotfound_ncbianditis[,-1], "data/derived_data/tax_table/intermediate_files/tax_macrofauna_surface_taxnotfound_ncbianditis.csv") # Fill cells with info found on internet, save it to x_mod and import it
taxfound_manually <- read.csv("data/derived_data/tax_table/intermediate_files/tax_macrofauna_surface_taxnotfound_ncbianditis_mod.csv", sep = ";")
taxfound_manually <- taxfound_manually[complete.cases(taxfound_manually$kingdom),-c(9,10)]

# Keep NCBI results in macrofauna_surface_tax and add ITIS results and manual taxonomy
taxfound_ncbi <- tax_macrofauna_surface_2[tax_macrofauna_surface_2$db == 'ncbi' & complete.cases(tax_macrofauna_surface_2$kingdom), -1]
taxfound_itisonly <- tax_macrofauna_surface_2[tax_macrofauna_surface_2$db == 'itis' & tax_macrofauna_surface_2$query %in% keepitis, ]
tax_macrofauna_surface_3 <- bind_rows(taxfound_ncbi, taxfound_itisonly[,-1], taxfound_manually) %>% mutate(community = "macrofauna_surface")

tax_macrofauna_surface_3[tax_macrofauna_surface_3 == "Animalia"] <- "Metazoa" # Replace Animalia (ITIS) par Metazoa (NCBI)
tax_macrofauna_surface_3 <- tax_macrofauna_surface_3 %>% replace(is.na(.), "unassigned")

tax_macrofauna_surface_3 <- tax_macrofauna_surface_3[,-1] %>% rename(finest_name=query)
#write.csv(tax_macrofauna_surface_3, "data/derived_data/tax_table/final_files/tax_macrofauna_surface_2024-09-13.csv", row.names = F)
```

## macrofauna_aerial

```{r}
tax_macrofauna_aerial_1 <- tidy_macrofauna_aerial_2 %>% distinct(finest_name)
temp <- gnr_resolve(tax_macrofauna_aerial_1$finest_name, preferred_data_sources = c(1,3,4,11,12), best_match_only = T) # 1 = CLC, 3 = ITIS, 4 = NCBI, 11 = GBIF, 12 = EOL (https://resolver.globalnames.org/data_sources)
diff <- setdiff(tax_macrofauna_aerial_1$finest_name, temp$user_supplied_name) 
print(diff)
# Manual cleaning
## Belytini not found because its a tribu of Belytinae
tax_macrofauna_aerial_1$finest_name[tax_macrofauna_aerial_1$finest_name == "Belytini"] <- "Belytinae"
# Run again from temp... and check diff = 0 now

#write.csv(tax_macrofauna_aerial_1, "data/derived_data/tax_table/intermediate_files/tax_macrofauna_aerial_taxizeinput_2024-09-13.csv", row.names = F)
```

```{r, echo=FALSE}
tax_macrofauna_aerial_2 = NULL
for (i in 1: nrow(tax_macrofauna_aerial_1)) {
classes_i <- try(tax_name(sci = tax_macrofauna_aerial_1$finest_name[i], get = c("kingdom", "phylum", "class", "order", "family", "genus", "species"), db = "both", ask = FALSE))
if (class(classes_i)=="try-error") {
Sys.sleep(10)
classes_i <- try(tax_name(sci = tax_macrofauna_aerial_1$finest_name[i], get = c("kingdom", "phylum", "class", "order", "family", "genus", "species"), db = "both", ask = FALSE))}
tax_macrofauna_aerial_2 <- rbind(tax_macrofauna_aerial_2, classes_i)
}

#write.csv(tax_macrofauna_aerial_2, "data/derived_data/tax_table/intermediate_files/tax_macrofauna_aerial_taxizeoutput.csv")
```

```{r, echo=FALSE}
tax_macrofauna_aerial_2 <- read.csv("data/derived_data/tax_table/intermediate_files/tax_macrofauna_aerial_taxizeoutput.csv")
taxnotfoundboth <- tax_macrofauna_aerial_2[!complete.cases(tax_macrofauna_aerial_2$kingdom),]
taxnotfoundonedb <- taxnotfoundboth %>% group_by(query) %>% filter(n() ==1) # These are the taxa found in only one database (not found in the database in this object = found in the other database in macrofauna_foliar_2). Therefore, if we filter NCBI (by default) results in macrofauna_foliar_2, we must also keep ITIS results for the taxa with no NCBI results in this objects. These taxa to keep are saved in the following vector :
keepitis <- c(subset(taxnotfoundonedb, db == 'ncbi')$query)
taxnotfound_ncbianditis <- subset(taxnotfoundboth,duplicated(query)) # These are the taxa not found in any database. We will export the list, fill it manually and reimport it.
#write.csv(taxnotfound_ncbianditis[,-1], "data/derived_data/tax_table/intermediate_files/tax_macrofauna_aerial_taxnotfound_ncbianditis.csv") # Fill cells with info found on internet, save it to x_mod and import it
taxfound_manually <- read.csv("data/derived_data/tax_table/intermediate_files/tax_macrofauna_aerial_taxnotfound_ncbianditis_mod.csv", sep = ";")
taxfound_manually <- taxfound_manually[complete.cases(taxfound_manually$kingdom),-c(9,10)]

# Keep NCBI results in macrofauna_aerial_tax and add ITIS results and manual taxonomy
taxfound_ncbi <- tax_macrofauna_aerial_2[tax_macrofauna_aerial_2$db == 'ncbi' & complete.cases(tax_macrofauna_aerial_2$kingdom), -1]
taxfound_itisonly <- tax_macrofauna_aerial_2[tax_macrofauna_aerial_2$db == 'itis' & tax_macrofauna_aerial_2$query %in% keepitis, ]
tax_macrofauna_aerial_3 <- bind_rows(taxfound_ncbi, taxfound_itisonly[,-1], taxfound_manually) %>% mutate(community = "macrofauna_aerial")

tax_macrofauna_aerial_3[tax_macrofauna_aerial_3 == "Animalia"] <- "Metazoa" # Replace Animalia (ITIS) par Metazoa (NCBI)
tax_macrofauna_aerial_3 <- tax_macrofauna_aerial_3 %>% replace(is.na(.), "unassigned")

tax_macrofauna_aerial_3 <- tax_macrofauna_aerial_3[,-1] %>% rename(finest_name=query)
#write.csv(tax_macrofauna_aerial_3, "data/derived_data/tax_table/final_files/tax_macrofauna_aerial_2024-09-13.csv", row.names = F)
```

## macrofauna_foliar

```{r}
tax_macrofauna_foliar_1 <- tidy_macrofauna_foliar_2 %>% distinct(finest_name)
temp <- gnr_resolve(tax_macrofauna_foliar_1$finest_name, preferred_data_sources = c(1,3,4,11,12), best_match_only = T) # 1 = CLC, 3 = ITIS, 4 = NCBI, 11 = GBIF, 12 = EOL (https://resolver.globalnames.org/data_sources)
diff <- setdiff(tax_macrofauna_foliar_1$finest_name, temp$user_supplied_name) 
print(diff)
# Manual cleaning
## Phonognathidae not found because wrong spelling
tax_macrofauna_foliar_1$finest_name[tax_macrofauna_foliar_1$finest_name == "Phonognathidae"] <- "Phonognathinae"
## vide should be removed
tax_macrofauna_foliar_1 <- tax_macrofauna_foliar_1[!(tax_macrofauna_foliar_1$finest_name %in% "vide"),]
# Run again from temp... and check diff = 0 now

#write.csv(tax_macrofauna_foliar_1, "data/derived_data/tax_table/intermediate_files/tax_macrofauna_foliar_taxizeinput_2024-09-13.csv")
```

```{r, echo=FALSE}
tax_macrofauna_foliar_2 = NULL
for (i in 1: nrow(tax_macrofauna_foliar_1)) {
classes_i <- try(tax_name(sci = tax_macrofauna_foliar_1$finest_name[i], get = c("kingdom", "phylum", "class", "order", "family", "genus", "species"), db = "both", ask = FALSE))
if (class(classes_i)=="try-error") {
Sys.sleep(10)
classes_i <- try(tax_name(sci = tax_macrofauna_foliar_1$finest_name[i], get = c("kingdom", "phylum", "class", "order", "family", "genus", "species"), db = "both", ask = FALSE))}
tax_macrofauna_foliar_2 <- rbind(tax_macrofauna_foliar_2, classes_i)
}

tax_macrofauna_foliar_2 <- tax_macrofauna_foliar_2[- grep("Error", tax_macrofauna_foliar_2$query),] # remove the rows with errors
#write.csv(tax_macrofauna_foliar_2, "data/derived_data/tax_table/intermediate_files/tax_macrofauna_foliar_taxizeoutput.csv")
```

```{r, echo=FALSE}
tax_macrofauna_foliar_2 <- read.csv("data/derived_data/tax_table/intermediate_files/tax_macrofauna_foliar_taxizeoutput.csv")
taxnotfoundboth <- tax_macrofauna_foliar_2[!complete.cases(tax_macrofauna_foliar_2$kingdom),]
taxnotfoundonedb <- taxnotfoundboth %>% group_by(query) %>% filter(n() ==1) # These are the taxa found in only one database (not found in the database in this object = found in the other database in macrofauna_foliar_2). Therefore, if we filter NCBI (by default) results in macrofauna_foliar_2, we must also keep ITIS results for the taxa with no NCBI results in this objects. These taxa to keep are saved in the following vector :
keepitis <- c(subset(taxnotfoundonedb, db == 'ncbi')$query)
taxnotfound_ncbianditis <- subset(taxnotfoundboth,duplicated(query)) # These are the taxa not found in any database. We will export the list, fill it manually and reimport it.
#write.csv(taxnotfound_ncbianditis[,-1], "data/derived_data/tax_table/intermediate_files/tax_macrofauna_foliar_taxnotfound_ncbianditis.csv") # Fill cells with info found on internet, save it to x_mod and import it
taxfound_manually <- read.csv("data/derived_data/tax_table/intermediate_files/tax_macrofauna_foliar_taxnotfound_ncbianditis_mod.csv", sep = ";")
taxfound_manually <- taxfound_manually[complete.cases(taxfound_manually$kingdom),-c(9,10)]

# Keep NCBI results in macrofauna_foliar_tax and add ITIS results and manual taxonomy
taxfound_ncbi <- tax_macrofauna_foliar_2[tax_macrofauna_foliar_2$db == 'ncbi' & complete.cases(tax_macrofauna_foliar_2$kingdom), -1]
taxfound_itisonly <- tax_macrofauna_foliar_2[tax_macrofauna_foliar_2$db == 'itis' & tax_macrofauna_foliar_2$query %in% keepitis, ]
tax_macrofauna_foliar_3 <- bind_rows(taxfound_ncbi, taxfound_itisonly[,-1], taxfound_manually) %>% mutate(community = "macrofauna_foliar")

tax_macrofauna_foliar_3[tax_macrofauna_foliar_3 == "Animalia"] <- "Metazoa" # Replace Animalia (ITIS) par Metazoa (NCBI)
tax_macrofauna_foliar_3 <- tax_macrofauna_foliar_3 %>% replace(is.na(.), "unassigned")

tax_macrofauna_foliar_3 <- tax_macrofauna_foliar_3[,-1] %>% rename(finest_name=query)
#write.csv(tax_macrofauna_foliar_3, "data/derived_data/tax_table/final_files/tax_macrofauna_foliar_2024-09-13.csv", row.names = F)
```                   

## nematodes
                    
```{r}
tax_nematodes_1 <- tidy_nematodes_2 %>% distinct(finest_name)
temp <- gnr_resolve(tax_nematodes_1$finest_name, preferred_data_sources = c(1,3,4,11,12), best_match_only = T) # 1 = CLC, 3 = ITIS, 4 = NCBI, 11 = GBIF, 12 = EOL (https://resolver.globalnames.org/data_sources)
diff <- setdiff(tax_nematodes_1$finest_name, temp$user_supplied_name) 
print(diff)
# diff = 0 so no need for manual cleaning

#write.csv(tax_nematodes_1, "data/derived_data/tax_table/intermediate_files/tax_nematodes_taxizeinput_2024-09-13.csv")
```

```{r, echo=FALSE}
tax_nematodes_2 = NULL
for (i in 1: nrow(tax_nematodes_1)) {
classes_i <- try(tax_name(sci = tax_nematodes_1$finest_name[i], get = c("kingdom", "phylum", "class", "order", "family", "genus", "species"), db = "both", ask = FALSE))
if (class(classes_i)=="try-error") {
Sys.sleep(10)
classes_i <- try(tax_name(sci = tax_nematodes_1$finest_name[i], get = c("kingdom", "phylum", "class", "order", "family", "genus", "species"), db = "both", ask = FALSE))}
tax_nematodes_2 <- rbind(tax_nematodes_2, classes_i)
}

#write.csv(tax_nematodes_2, "data/derived_data/tax_table/intermediate_files/tax_nematodes_taxizeoutput.csv")
```

```{r, echo=FALSE}
tax_nematodes_2 <- read.csv("data/derived_data/tax_table/intermediate_files/tax_nematodes_taxizeoutput.csv")
taxnotfoundboth <- tax_nematodes_2[!complete.cases(tax_nematodes_2$kingdom),] 
taxnotfound_ncbianditis <- subset(taxnotfoundboth,duplicated(query)) # All taxa are found with NCBI or ITIS.

# Keep NCBI results in nematodes_tax and add ITIS results and manual taxonomy
tax_nematodes_3 <- tax_nematodes_2[tax_nematodes_2$db == 'ncbi' & complete.cases(tax_nematodes_2$kingdom), -1] %>% mutate(community = "nematodes")

tax_nematodes_3[tax_nematodes_3 == "Animalia"] <- "Metazoa" # Replace Animalia (ITIS) par Metazoa (NCBI)
tax_nematodes_3 <- tax_nematodes_3 %>% replace(is.na(.), "unassigned")

#tax_nematodes_3 <- tax_nematodes_3[,-1] %>% rename(finest_name=query)
write.csv(tax_nematodes_3, "data/derived_data/tax_table/final_files/tax_nematodes_2024-09-13.csv", row.names = F)
```

## micro_arthropodes

```{r}
tax_micro_arthropodes_1 <- tidy_micro_arthropodes_2 %>% distinct(finest_name)
temp <- gnr_resolve(tax_micro_arthropodes_1$finest_name, preferred_data_sources = c(1,3,4,11,12), best_match_only = T) # 1 = CLC, 3 = ITIS, 4 = NCBI, 11 = GBIF, 12 = EOL (https://resolver.globalnames.org/data_sources)
diff <- setdiff(tax_micro_arthropodes_1$finest_name, temp$user_supplied_name) 
print(diff)
# Manual cleaning
## NA should be removed
tax_micro_arthropodes_1 <- tax_micro_arthropodes_1[!(tax_micro_arthropodes_1$finest_name %in% "NA"),]
tax_micro_arthropodes_1 <- data.frame(tax_micro_arthropodes_1)
# diff = 0 now

#write.csv(tax_micro_arthropodes_1, "data/derived_data/tax_table/intermediate_files/tax_micro_arthropodes_taxizeinput_2024-09-13.csv", row.names = F)
```

```{r, echo=FALSE}
tax_micro_arthropodes_2 = NULL
for (i in 1: nrow(tax_micro_arthropodes_1)) {
classes_i <- try(tax_name(sci = tax_micro_arthropodes_1$finest_name[i], get = c("kingdom", "phylum", "class", "order", "family", "genus", "species"), db = "both", ask = FALSE))
if (class(classes_i)=="try-error") {
Sys.sleep(10)
classes_i <- try(tax_name(sci = tax_micro_arthropodes_1$finest_name[i], get = c("kingdom", "phylum", "class", "order", "family", "genus", "species"), db = "both", ask = FALSE))}
tax_micro_arthropodes_2 <- rbind(tax_micro_arthropodes_2, classes_i)
}

#write.csv(tax_micro_arthropodes_2, "data/derived_data/tax_table/intermediate_files/tax_micro_arthropodes_taxizeoutput.csv")
```

```{r, echo=FALSE}
tax_micro_arthropodes_2 <- read.csv("data/derived_data/tax_table/intermediate_files/tax_micro_arthropodes_taxizeoutput.csv")
taxnotfoundboth <- tax_micro_arthropodes_2[!complete.cases(tax_micro_arthropodes_2$kingdom),]
taxnotfoundonedb <- taxnotfoundboth %>% group_by(query) %>% filter(n() ==1) # These are the taxa found in only one database (not found in the database in this object = found in the other database in micro_arthropodes_2). Therefore, if we filter NCBI (by default) results in micro_arthropodes_2, we must also keep ITIS results for the taxa with no NCBI results in this objects. These taxa to keep are saved in the following vector :
keepitis <- c(subset(taxnotfoundonedb, db == 'ncbi')$query)
taxnotfound_ncbianditis <- subset(taxnotfoundboth,duplicated(query)) # These are the taxa not found in any database. We will export the list, fill it manually and reimport it.
#write.csv(taxnotfound_ncbianditis[,-1], "data/derived_data/tax_table/intermediate_files/tax_micro_arthropodes_taxnotfound_ncbianditis.csv") # Fill cells with info found on internet, save it to x_mod and import it
taxfound_manually <- read.csv("data/derived_data/tax_table/intermediate_files/tax_micro_arthropodes_taxnotfound_ncbianditis_mod.csv", sep = ";")
taxfound_manually <- taxfound_manually[complete.cases(taxfound_manually$kingdom),-c(9,10)]

# Keep NCBI results in micro_arthropodes_tax and add ITIS results and manual taxonomy
taxfound_ncbi <- tax_micro_arthropodes_2[tax_micro_arthropodes_2$db == 'ncbi' & complete.cases(tax_micro_arthropodes_2$kingdom), -1]
taxfound_itisonly <- tax_micro_arthropodes_2[tax_micro_arthropodes_2$db == 'itis' & tax_micro_arthropodes_2$query %in% keepitis, ]
tax_micro_arthropodes_3 <- bind_rows(taxfound_ncbi, taxfound_itisonly[,-1], taxfound_manually) %>% mutate(community = "micro_arthropodes")

tax_micro_arthropodes_3[tax_micro_arthropodes_3 == "Animalia"] <- "Metazoa" # Replace Animalia (ITIS) par Metazoa (NCBI)
tax_micro_arthropodes_3 <- tax_micro_arthropodes_3 %>% replace(is.na(.), "unassigned")

tax_micro_arthropodes_3 <- tax_micro_arthropodes_3[,-1] %>% rename(finest_name=query)
#write.csv(tax_micro_arthropodes_3, "data/derived_data/tax_table/final_files/tax_micro_arthropodes_2024-09-13.csv", row.names = F)
```

## microorganisms

### bacteria

```{r}
# Choose the option you want to work with
tidy_bacteria <- tidy_bacteria_4a # blast_taxonomy results
#tidy_bacteria <- read.csv("data/derived_data/tax_table/final_files/tax_bacteria_blast_cleaned_filtered_2024-09-13.csv")

tax_bacteria <- 
  unique(tidy_bacteria[, c("finest_name", "kingdom", "phylum", "class", "order", "family", "genus", "species", "community")])

#write.csv(tax_bacteria, "data/derived_data/tax_table/final_files/tax_bacteria_blast_cleaned_filtered_2024-09-13.csv", row.names = FALSE)
#saveRDS(tax_bacteria, "data/derived_data/tax_table/final_files/tax_bacteria_blast_cleaned_filtered_2024-09-13.rds")
```

### fungi

```{r}
# Choose the option you want to work with
tidy_fungi <- tidy_fungi_4a # blast_taxonomy results
#tidy_fungi <- read.csv("data/derived_data/tax_table/final_files/tax_fungi_blast_cleaned_filtered_2024-09-13.csv")

tax_fungi <- 
  unique(tidy_fungi[, c("finest_name", "kingdom", "phylum", "class", "order", "family", "genus", "species", "community")])

#write.csv(tax_fungi, "data/derived_data/tax_table/final_files/tax_fungi_blast_cleaned_filtered_2024-09-13.csv", row.names = FALSE)
#saveRDS(tax_fungi, "data/derived_data/tax_table/final_files/tax_fungi_blast_cleaned_filtered_2024-09-13.rds")
```

### protists

```{r}
# Choose the option you want to work with
tidy_protists <- tidy_protists_4a # blast_taxonomy results
#tidy_protists <- read.csv("data/derived_data/tax_table/final_files/tax_protists_blast_cleaned_filtered_2024-09-13.csv")

tax_protists <- 
  unique(tidy_protists[, c("finest_name", "kingdom", "phylum", "class", "order", "family", "genus", "species", "community")])

#write.csv(tax_protists, "data/derived_data/tax_table/final_files/tax_protists_blast_cleaned_filtered_2024-09-13.csv", row.names = FALSE)
#saveRDS(tax_protists, "data/derived_data/tax_table/final_files/tax_protists_blast_cleaned_filtered_2024-09-13.rds")
```

### all

```{r}
tax_mo <- rbind(tax_bacteria, tax_fungi, tax_protists)

tax_mo_unique <- tax_mo %>%
  group_by(community) %>%
  summarise(across(kingdom:species, n_distinct, na.rm = TRUE))
#write.csv(tax_mo_unique, "analyses/tracking/blast_number_level_2024-09-13.csv", row.names = F)

tax_mo_unique_long <- pivot_longer(tax_mo_unique, cols = c(2:8), names_to = "level", values_to = "taxa_number")
tax_mo_unique_long$level <- ordered(tax_mo_unique_long$level, order_level)

T <- ggplot(tax_mo_unique_long, aes(level, taxa_number, fill=community)) + frame_simple + geom_bar(stat="identity", position="dodge") + theme(axis.text.x = element_text(angle = 45, hjust=1)) + theme(axis.text=element_text(size=8)) + labs(title="Number of distinct BLAST assignations per taxonomic level") + scale_y_continuous(labels = scales::comma)
T
#ggsave("analyses/tracking/blast_number_level_2024-09-13.png", width=6, height=4.5)
```

# Export tax_table

```{r}
tax_macrofauna_surface_3 <- read.csv("data/derived_data/tax_table/final_files/tax_macrofauna_surface_2024-09-13.csv")
tax_macrofauna_aerial_3 <- read.csv("data/derived_data/tax_table/final_files/tax_macrofauna_aerial_2024-09-13.csv")
tax_macrofauna_foliar_3 <- read.csv("data/derived_data/tax_table/final_files/tax_macrofauna_foliar_2024-09-13.csv")
tax_micro_arthropodes_3 <- read.csv("data/derived_data/tax_table/final_files/tax_micro_arthropodes_2024-09-13.csv")
tax_nematodes_3 <- read.csv("data/derived_data/tax_table/final_files/tax_nematodes_2024-09-13.csv")
tax_bacteria <- read.csv("data/derived_data/tax_table/final_files/tax_bacteria_blast_cleaned_filtered_2024-09-13.csv")
tax_fungi <- read.csv("data/derived_data/tax_table/final_files/tax_fungi_blast_cleaned_filtered_2024-09-13.csv")
tax_protists <- read.csv("data/derived_data/tax_table/final_files/tax_protists_blast_cleaned_filtered_2024-09-13.csv")

tax_ALL <- bind_rows(tax_macrofauna_surface_3, tax_macrofauna_aerial_3, tax_macrofauna_foliar_3, tax_nematodes_3, tax_micro_arthropodes_3, tax_bacteria, tax_fungi, tax_protists) # bind all communities in one file
#write.csv(tax_ALL, "data/derived_data/tax_table/final_files/tax_all_2024-09-13.csv", row.names = F)
#write_rds(tax_ALL, "data/derived_data/tax_table/final_files/tax_all_2024-09-13.rds")
```

# Check tracking

## all

```{r}
tax_ALL <- read.csv("data/derived_data/tax_table/final_files/tax_all_2024-09-13.csv")

step_2 <- tax_ALL %>% group_by(community) %>% summarise(n_distinct_finest_name = n_distinct(finest_name)) %>% mutate(step = "step_2")

step_2$community <- ordered(step_2$community, order_community)

#write.csv(step_2, "analyses/tracking/step_2_2024-09-13.csv", row.names = F)

S2 <- ggplot(step_2, aes(x = step, y = n_distinct_finest_name, fill = community)) + frame_simple + geom_bar(stat="identity", position="dodge") + labs(title="Number of distinct taxa names at finest level") + scale_y_continuous(labels = scales::comma) + scale_fill_brewer(palette = "Set2") + guides(size = F)
S2
#ggsave("analyses/tracking/tracking_step_2_2024-09-13.png", width=6, height=4.5)
```

```{r}
tax_ALL_long <- pivot_longer(tax_ALL, cols=c(2:8), names_to = "level", values_to = "taxa")

tax_ALL_long$level <- ordered(tax_ALL_long$level, order_level)

tax_ALL_long_sum <- tax_ALL_long %>%
    group_by(community, level) %>%
    summarise(n_distinct_taxa = n_distinct(taxa))

tax_ALL_long_sum$community <- ordered(tax_ALL_long_sum$community, order_community)

T1 <- ggplot(tax_ALL_long_sum, aes(level, n_distinct_taxa, fill=community)) + geom_bar(stat="identity") + frame_simple  + facet_grid(vars(community), scales="free") + labs(title="Taxonomic assignation", subtitle = "Number of distinct taxa names per taxonomic level",x ="Taxonomic level") + scale_y_continuous(labels = scales::comma) + scale_fill_brewer(palette = "Set2") + theme(axis.text.x = element_text(angle = 45, hjust=1))
T1
#ggsave("analyses/tracking/tracking_taxononomic_assignation_2024-09-19.png", width=5.5, height=11)

tax_ALL_long_sum_wide <- pivot_wider(tax_ALL_long_sum, names_from = level, values_from = n_distinct_taxa)
#write.csv(tax_ALL_long_sum_wide, "analyses/tracking/step_2_bylevel_2024-09-19.csv", row.names = F)
```

```{r}
tax_ALL_au <- tax_ALL

tax_ALL_au[, 2:8] <- apply(tax_ALL_au[, 2:8], 2, function(x) ifelse(x == "unassigned", "unassigned", "assigned"))

tidy_ALL <- read.csv("data/derived_data/tidy_data/tidy_all_2024-09-13.csv")
tidy_ALL <- aggregate(abundance ~ finest_name, data = tidy_ALL, FUN = sum)

tidytax_ALL_au <- merge(tax_ALL_au, tidy_ALL, by = "finest_name")

tidytax_ALL_au_long <- pivot_longer(tidytax_ALL_au, cols=c(2:8), names_to = "level", values_to = "assignation")

tidytax_ALL_au_long_abund <- aggregate(abundance ~ community*level*assignation, data = tidytax_ALL_au_long, FUN = sum)
#write.csv(tidytax_ALL_au_long_abund, "analyses/tracking/tidytax_assignation_abund_2024-09-13.csv", row.names = F)

tidytax_ALL_au_long_distinct <- tidytax_ALL_au_long %>%  group_by(community, level, assignation) %>% summarise(n_distinct_finest_name = n_distinct(finest_name))
#write.csv(tidytax_ALL_au_long_distinct, "analyses/tracking/tidytax_assignation_distinct_2024-09-13.csv", row.names = F)

tidytax_ALL_au_long$community <- ordered(tidytax_ALL_au_long$community, order_community)
tidytax_ALL_au_long$level <- ordered(tidytax_ALL_au_long$level, order_level)

L1 <- ggplot(tidytax_ALL_au_long, aes(level, abundance, fill=assignation)) + geom_bar(stat="identity") + frame_simple + facet_grid(vars(community), scales="free") + labs(title="Abundance of assigned or unassigned organisms", subtitle="Taxonomic assignation", x ="Taxonomic level", y ="abundance") + scale_y_continuous(labels = scales::comma) + scale_fill_brewer(palette = "Set2") + theme(axis.text.x = element_text(angle = 45, hjust=1))
L1
#ggsave("analyses/tracking/tracking_taxononomic_assignation_binary_2024-09-04.png", width=5.5, height=11)
```

# 3. Functional table

```{r}
endpoint_url <- "http://129.88.204.79:7200/repositories/gratin-20240723"
sources <- get.sources(endpoint = endpoint_url)
print(sources)

tax_ALL <- read.csv("data/derived_data/tax_table/final_files/tax_all_2024-08-21.csv")
tax_macrofauna_surface <- subset(tax_ALL, community == 'macrofauna_surface')
tax_macrofauna_aerial <- subset(tax_ALL, community == 'macrofauna_aerial')
tax_macrofauna_foliar <- subset(tax_ALL, community == 'macrofauna_foliar')
tax_nematodes <- subset(tax_ALL, community == 'nematodes')
tax_micro_arthropodes <- subset(tax_ALL, community == 'micro_arthropodes')
tax_bacteria <- subset(tax_ALL, community == 'bacteria')
tax_fungi <- subset(tax_ALL, community == 'fungi')
tax_protists <- subset(tax_ALL, community == 'protists')

# Chose the levels to focus on
taxlevels_focus = c("species", "genus", "family")
```

## macrofauna_surface

```{r}
# Guilds
list_macrofauna_surface_guilds <- list()

for (i in seq_along(taxlevels_focus)) {
  taxlevel <- taxlevels_focus[i]
  
  guilds <- get.guilds(sciName = unique(tax_macrofauna_surface[[taxlevel]]), endpoint = endpoint_url)
  list_macrofauna_surface_guilds[[taxlevel]] <- guilds
}

macrofauna_surface_guilds_species <- as.data.frame(list_macrofauna_surface_guilds$species) %>% mutate(level = "species")
macrofauna_surface_guilds_genus <- as.data.frame(list_macrofauna_surface_guilds$genus) %>% mutate(level = "genus")
macrofauna_surface_guilds_family <- as.data.frame(list_macrofauna_surface_guilds$family) %>% mutate(level = "family")

macrofauna_surface_guilds <- rbind(macrofauna_surface_guilds_species, macrofauna_surface_guilds_genus, macrofauna_surface_guilds_family) %>% mutate(community = "macrofauna_surface")

#write.csv(apply(macrofauna_surface_guilds,2,as.character), "data/derived_data/fun_table/fun_macrofauna_surface_guilds_2024-08-21.csv", row.names = F)
#write_rds(apply(macrofauna_surface_guilds,2,as.character), "data/derived_data/fun_table/fun_macrofauna_surface_guilds_2024-08-20.rds")

# Interactions
list_macrofauna_surface_interactions <- list()

for (i in seq_along(taxlevels_focus)) {
  taxlevel <- taxlevels_focus[i]
  
  interactions <- get.interactions(sciName = unique(tax_macrofauna_surface[[taxlevel]]), endpoint = endpoint_url)
  list_macrofauna_surface_interactions[[taxlevel]] <- interactions
}

macrofauna_surface_interactions_species <- as.data.frame(list_macrofauna_surface_interactions$species) %>% mutate(level = "species")
macrofauna_surface_interactions_genus <- as.data.frame(list_macrofauna_surface_interactions$genus) %>% mutate(level = "genus")
macrofauna_surface_interactions_family <- as.data.frame(list_macrofauna_surface_interactions$family) %>% mutate(level = "family")

macrofauna_surface_interactions <- rbind(macrofauna_surface_interactions_species, macrofauna_surface_interactions_genus, macrofauna_surface_interactions_family) %>% mutate(community = "macrofauna_surface")

#write.csv(apply(macrofauna_surface_interactions,2,as.character), "data/derived_data/fun_table/fun_macrofauna_surface_interactions_2024-08-21.csv", row.names = F)
#write_rds(apply(macrofauna_surface_interactions,2,as.character), "data/derived_data/fun_table/fun_macrofauna_surface_interactions_2024-08-21.rds")
```

## macrofauna_aerial

```{r}
# Guilds
list_macrofauna_aerial_guilds <- list()

for (i in seq_along(taxlevels_focus)) {
  taxlevel <- taxlevels_focus[i]
  
  guilds <- get.guilds(sciName = unique(tax_macrofauna_aerial[[taxlevel]]), endpoint = endpoint_url)
  list_macrofauna_aerial_guilds[[taxlevel]] <- guilds
}

macrofauna_aerial_guilds_species <- as.data.frame(list_macrofauna_aerial_guilds$species) %>% mutate(level = "species")
macrofauna_aerial_guilds_genus <- as.data.frame(list_macrofauna_aerial_guilds$genus) %>% mutate(level = "genus")
macrofauna_aerial_guilds_family <- as.data.frame(list_macrofauna_aerial_guilds$family) %>% mutate(level = "family")

macrofauna_aerial_guilds <- rbind(macrofauna_aerial_guilds_species, macrofauna_aerial_guilds_genus, macrofauna_aerial_guilds_family) %>% mutate(community = "macrofauna_aerial")

#write.csv(apply(macrofauna_aerial_guilds,2,as.character), "data/derived_data/fun_table/fun_macrofauna_aerial_guilds_2024-08-20.csv", row.names = F)
#write_rds(apply(macrofauna_aerial_guilds,2,as.character), "data/derived_data/fun_table/fun_macrofauna_aerial_guilds_2024-08-20.rds")

# Interactions
list_macrofauna_aerial_interactions <- list()

for (i in seq_along(taxlevels_focus)) {
  taxlevel <- taxlevels_focus[i]
  
  interactions <- get.interactions(sciName = unique(tax_macrofauna_aerial[[taxlevel]]), endpoint = endpoint_url)
  list_macrofauna_aerial_interactions[[taxlevel]] <- interactions
}

macrofauna_aerial_interactions_species <- as.data.frame(list_macrofauna_aerial_interactions$species) %>% mutate(level = "species")
macrofauna_aerial_interactions_genus <- as.data.frame(list_macrofauna_aerial_interactions$genus) %>% mutate(level = "genus")
macrofauna_aerial_interactions_family <- as.data.frame(list_macrofauna_aerial_interactions$family) %>% mutate(level = "family")

macrofauna_aerial_interactions <- rbind(macrofauna_aerial_interactions_species, macrofauna_aerial_interactions_genus, macrofauna_aerial_interactions_family) %>% mutate(community = "macrofauna_aerial")

#write.csv(apply(macrofauna_aerial_interactions,2,as.character), "data/derived_data/fun_table/fun_macrofauna_aerial_interactions_2024-08-20.csv", row.names = F)
#write_rds(apply(macrofauna_aerial_interactions,2,as.character), "data/derived_data/fun_table/fun_macrofauna_aerial_interactions_2024-08-20.rds")
```

## macrofauna_foliar

```{r}
# Guilds
list_macrofauna_foliar_guilds <- list()

for (i in seq_along(taxlevels_focus)) {
  taxlevel <- taxlevels_focus[i]
  
  guilds <- get.guilds(sciName = unique(tax_macrofauna_foliar[[taxlevel]]), endpoint = endpoint_url)
  list_macrofauna_foliar_guilds[[taxlevel]] <- guilds
}

macrofauna_foliar_guilds_species <- as.data.frame(list_macrofauna_foliar_guilds$species) %>% mutate(level = "species")
macrofauna_foliar_guilds_genus <- as.data.frame(list_macrofauna_foliar_guilds$genus) %>% mutate(level = "genus")
macrofauna_foliar_guilds_family <- as.data.frame(list_macrofauna_foliar_guilds$family) %>% mutate(level = "family")

macrofauna_foliar_guilds <- rbind(macrofauna_foliar_guilds_species, macrofauna_foliar_guilds_genus, macrofauna_foliar_guilds_family) %>% mutate(community = "macrofauna_foliar")

#write.csv(apply(macrofauna_foliar_guilds,2,as.character), "data/derived_data/fun_table/fun_macrofauna_foliar_guilds_2024-08-20.csv", row.names = F)
#write_rds(apply(macrofauna_foliar_guilds,2,as.character), "data/derived_data/fun_table/fun_macrofauna_foliar_guilds_2024-08-20.rds")

# Interactions
list_macrofauna_foliar_interactions <- list()

for (i in seq_along(taxlevels_focus)) {
  taxlevel <- taxlevels_focus[i]
  
  interactions <- get.interactions(sciName = unique(tax_macrofauna_foliar[[taxlevel]]), endpoint = endpoint_url)
  list_macrofauna_foliar_interactions[[taxlevel]] <- interactions
}

macrofauna_foliar_interactions_species <- as.data.frame(list_macrofauna_foliar_interactions$species) %>% mutate(level = "species")
macrofauna_foliar_interactions_genus <- as.data.frame(list_macrofauna_foliar_interactions$genus) %>% mutate(level = "genus")
macrofauna_foliar_interactions_family <- as.data.frame(list_macrofauna_foliar_interactions$family) %>% mutate(level = "family")

macrofauna_foliar_interactions <- rbind(macrofauna_foliar_interactions_species, macrofauna_foliar_interactions_genus, macrofauna_foliar_interactions_family) %>% mutate(community = "macrofauna_foliar")

#write.csv(apply(macrofauna_foliar_interactions,2,as.character), "data/derived_data/fun_table/fun_macrofauna_foliar_interactions_2024-08-20.csv", row.names = F)
#write_rds(apply(macrofauna_foliar_interactions,2,as.character), "data/derived_data/fun_table/fun_macrofauna_foliar_interactions_2024-08-20.rds")
```

## nematodes

```{r}
# Guilds
list_nematodes_guilds <- list()

for (i in seq_along(taxlevels_focus)) {
  taxlevel <- taxlevels_focus[i]
  
  guilds <- get.guilds(sciName = unique(tax_nematodes[[taxlevel]]), endpoint = endpoint_url)
  list_nematodes_guilds[[taxlevel]] <- guilds
}

nematodes_guilds_species <- as.data.frame(list_nematodes_guilds$species) %>% mutate(level = "species")
nematodes_guilds_genus <- as.data.frame(list_nematodes_guilds$genus) %>% mutate(level = "genus")
nematodes_guilds_family <- as.data.frame(list_nematodes_guilds$family) %>% mutate(level = "family")

nematodes_guilds <- rbind(nematodes_guilds_species, nematodes_guilds_genus, nematodes_guilds_family) %>% mutate(community = "nematodes")

#write.csv(apply(nematodes_guilds,2,as.character), "data/derived_data/fun_table/fun_nematodes_guilds_2024-08-20.csv", row.names = F)
#write_rds(apply(nematodes_guilds,2,as.character), "data/derived_data/fun_table/fun_nematodes_guilds_2024-08-20.rds")

# Interactions
list_nematodes_interactions <- list()

for (i in seq_along(taxlevels_focus)) {
  taxlevel <- taxlevels_focus[i]
  
  interactions <- get.interactions(sciName = unique(tax_nematodes[[taxlevel]]), endpoint = endpoint_url)
  list_nematodes_interactions[[taxlevel]] <- interactions
}

nematodes_interactions_species <- as.data.frame(list_nematodes_interactions$species) %>% mutate(level = "species")
nematodes_interactions_genus <- as.data.frame(list_nematodes_interactions$genus) %>% mutate(level = "genus")
nematodes_interactions_family <- as.data.frame(list_nematodes_interactions$family) %>% mutate(level = "family")

nematodes_interactions <- rbind(nematodes_interactions_species, nematodes_interactions_genus, nematodes_interactions_family) %>% mutate(community = "nematodes")

#write.csv(apply(nematodes_interactions,2,as.character), "data/derived_data/fun_table/fun_nematodes_interactions_2024-08-20.csv", row.names = F)
#write_rds(apply(nematodes_interactions,2,as.character), "data/derived_data/fun_table/fun_nematodes_interactions_2024-08-20.rds")
```

## micro_arthropodes

```{r}
# Guilds
list_micro_arthropodes_guilds <- list()

for (i in seq_along(taxlevels_focus)) {
  taxlevel <- taxlevels_focus[i]
  
  guilds <- get.guilds(sciName = unique(tax_micro_arthropodes[[taxlevel]]), endpoint = endpoint_url)
  list_micro_arthropodes_guilds[[taxlevel]] <- guilds
}

micro_arthropodes_guilds_species <- as.data.frame(list_micro_arthropodes_guilds$species) %>% mutate(level = "species")
micro_arthropodes_guilds_genus <- as.data.frame(list_micro_arthropodes_guilds$genus) %>% mutate(level = "genus")
micro_arthropodes_guilds_family <- as.data.frame(list_micro_arthropodes_guilds$family) %>% mutate(level = "family")

micro_arthropodes_guilds <- rbind(micro_arthropodes_guilds_species, micro_arthropodes_guilds_genus, micro_arthropodes_guilds_family) %>% mutate(community = "micro_arthropodes")

#write.csv(apply(micro_arthropodes_guilds,2,as.character), "data/derived_data/fun_table/fun_micro_arthropodes_guilds_2024-08-20.csv", row.names = F)
#write_rds(apply(micro_arthropodes_guilds,2,as.character), "data/derived_data/fun_table/fun_micro_arthropodes_guilds_2024-08-20.rds")

# Interactions
list_micro_arthropodes_interactions <- list()

for (i in seq_along(taxlevels_focus)) {
  taxlevel <- taxlevels_focus[i]
  
  interactions <- get.interactions(sciName = unique(tax_micro_arthropodes[[taxlevel]]), endpoint = endpoint_url)
  list_micro_arthropodes_interactions[[taxlevel]] <- interactions
}

micro_arthropodes_interactions_species <- as.data.frame(list_micro_arthropodes_interactions$species) %>% mutate(level = "species")
micro_arthropodes_interactions_genus <- as.data.frame(list_micro_arthropodes_interactions$genus) %>% mutate(level = "genus")
micro_arthropodes_interactions_family <- as.data.frame(list_micro_arthropodes_interactions$family) %>% mutate(level = "family")

micro_arthropodes_interactions <- rbind(micro_arthropodes_interactions_species, micro_arthropodes_interactions_genus, micro_arthropodes_interactions_family) %>% mutate(community = "micro_arthropodes")

#write.csv(apply(micro_arthropodes_interactions,2,as.character), "data/derived_data/fun_table/fun_micro_arthropodes_interactions_2024-08-20.csv", row.names = F)
#write_rds(apply(micro_arthropodes_interactions,2,as.character), "data/derived_data/fun_table/fun_micro_arthropodes_interactions_2024-08-20.rds")
```

## microorganisms

### bacteria

```{r}
# Split taxa in 10 lists
unique_species <- unique(tax_bacteria$species)
n <- length(unique_species)
size <- ceiling(n/10)
lists_species <- split(unique_species, ceiling(seq_along(unique_species)/size))

unique_genus <- unique(tax_bacteria$genus)
n <- length(unique_genus)
size <- ceiling(n/10)
lists_genus <- split(unique_genus, ceiling(seq_along(unique_genus)/size))

unique_family <- unique(tax_bacteria$family)
n <- length(unique_family)
size <- ceiling(n/10)
lists_family <- split(unique_family, ceiling(seq_along(unique_family)/size))

unique_order <- unique(tax_bacteria$order)
n <- length(unique_order)
size <- ceiling(n/10)
lists_order <- split(unique_order, ceiling(seq_along(unique_order)/size))
```

```{r}
# Guilds
list_bacteria_guilds_species <- list()
list_bacteria_guilds_genus <- list()
list_bacteria_guilds_family <- list()
list_bacteria_guilds_order <- list()

for (i in 1:10) {
  guilds_species <- get.guilds(sciName = lists_species[[i]], endpoint = endpoint_url)
  list_bacteria_guilds_species[[i]] <- guilds_species
  
  guilds_genus <- get.guilds(sciName = lists_genus[[i]], endpoint = endpoint_url)
  list_bacteria_guilds_genus[[i]] <- guilds_genus
  
  guilds_family <- get.guilds(sciName = lists_family[[i]], endpoint = endpoint_url)
  list_bacteria_guilds_family[[i]] <- guilds_family
  
  guilds_order <- get.guilds(sciName = lists_order[[i]], endpoint = endpoint_url)
  list_bacteria_guilds_order[[i]] <- guilds_order
}

bacteria_guilds_species <- do.call(rbind.data.frame, list_bacteria_guilds_species) %>% mutate(level = "species")
bacteria_guilds_genus <- do.call(rbind.data.frame, list_bacteria_guilds_genus) %>% mutate(level = "genus")
bacteria_guilds_family <- do.call(rbind.data.frame, list_bacteria_guilds_family) %>% mutate(level = "family")
bacteria_guilds_order <- do.call(rbind.data.frame, list_bacteria_guilds_order) %>% mutate(level = "order")

bacteria_guilds <- rbind(bacteria_guilds_species, bacteria_guilds_genus, bacteria_guilds_family, bacteria_guilds_order)  %>% mutate(community = "bacteria")

#write.csv(apply(bacteria_guilds,2,as.character), "data/derived_data/fun_table/fun_bacteria_guilds_2024-08-21.csv", row.names = F)
#write_rds(apply(bacteria_guilds,2,as.character), "data/derived_data/fun_table/fun_bacteria_guilds_2024-08-21.rds")

# Interactions
list_bacteria_interactions_species <- list()
list_bacteria_interactions_genus <- list()
list_bacteria_interactions_family <- list()
list_bacteria_interactions_order <- list()

for (i in 1:10) {
  interactions_species <- get.interactions(sciName = lists_species[[i]], endpoint = endpoint_url)
  list_bacteria_interactions_species[[i]] <- interactions_species
  
  interactions_genus <- get.interactions(sciName = lists_genus[[i]], endpoint = endpoint_url)
  list_bacteria_interactions_genus[[i]] <- interactions_genus
  
  interactions_family <- get.interactions(sciName = lists_family[[i]], endpoint = endpoint_url)
  list_bacteria_interactions_family[[i]] <- interactions_family
  
  interactions_order <- get.interactions(sciName = lists_order[[i]], endpoint = endpoint_url)
  list_bacteria_interactions_order[[i]] <- interactions_order
}

bacteria_interactions_species <- do.call(rbind.data.frame, list_bacteria_interactions_species) %>% mutate(level = "species")
bacteria_interactions_genus <- do.call(rbind.data.frame, list_bacteria_interactions_genus) %>% mutate(level = "genus")
bacteria_interactions_family <- do.call(rbind.data.frame, list_bacteria_interactions_family) %>% mutate(level = "family")
bacteria_interactions_order <- do.call(rbind.data.frame, list_bacteria_interactions_order) %>% mutate(level = "order")

bacteria_interactions <- rbind(bacteria_interactions_species, bacteria_interactions_genus, bacteria_interactions_family, bacteria_interactions_order) %>% mutate(community = "bacteria")

#write_delim(bacteria_interactions, "data/derived_data/fun_table/fun_bacteria_interactions_2024-08-21.csv", delim=",")
#write_rds(apply(bacteria_interactions,2,as.character), "data/derived_data/fun_table/fun_bacteria_interactions_2024-08-21.rds")
```

### fungi

```{r}
# Split taxa in 9 lists
unique_species <- unique(tax_fungi$species)
n <- length(unique_species)
size <- ceiling(n/9)
lists_species <- split(unique_species, ceiling(seq_along(unique_species)/size))

unique_genus <- unique(tax_fungi$genus)
n <- length(unique_genus)
size <- ceiling(n/9)
lists_genus <- split(unique_genus, ceiling(seq_along(unique_genus)/size))

unique_family <- unique(tax_fungi$family)
n <- length(unique_family)
size <- ceiling(n/9)
lists_family <- split(unique_family, ceiling(seq_along(unique_family)/size))

unique_order <- unique(tax_fungi$order)
n <- length(unique_order)
size <- ceiling(n/9)
lists_order <- split(unique_order, ceiling(seq_along(unique_order)/size))
```

```{r}
# Guilds
list_fungi_guilds_species <- list()
list_fungi_guilds_genus <- list()
list_fungi_guilds_family <- list()
list_fungi_guilds_order <- list()

for (i in 1:9) {
  guilds_species <- get.guilds(sciName = lists_species[[i]], endpoint = endpoint_url)
  list_fungi_guilds_species[[i]] <- guilds_species
  
  guilds_genus <- get.guilds(sciName = lists_genus[[i]], endpoint = endpoint_url)
  list_fungi_guilds_genus[[i]] <- guilds_genus
  
  guilds_family <- get.guilds(sciName = lists_family[[i]], endpoint = endpoint_url)
  list_fungi_guilds_family[[i]] <- guilds_family
  
  guilds_order <- get.guilds(sciName = lists_order[[i]], endpoint = endpoint_url)
  list_fungi_guilds_order[[i]] <- guilds_order
}

fungi_guilds_species <- do.call(rbind.data.frame, list_fungi_guilds_species) %>% mutate(level = "species")
fungi_guilds_genus <- do.call(rbind.data.frame, list_fungi_guilds_genus) %>% mutate(level = "genus")
fungi_guilds_family <- do.call(rbind.data.frame, list_fungi_guilds_family) %>% mutate(level = "family")
fungi_guilds_order <- do.call(rbind.data.frame, list_fungi_guilds_order) %>% mutate(level = "order")

fungi_guilds <- rbind(fungi_guilds_species, fungi_guilds_genus, fungi_guilds_family, fungi_guilds_order) %>% mutate(community = "fungi")

#write.csv(apply(fungi_guilds,2,as.character), "data/derived_data/fun_table/fun_fungi_guilds_2024-08-21.csv", row.names = F)
#write_rds(apply(fungi_guilds,2,as.character), "data/derived_data/fun_table/fun_fungi_guilds_2024-08-21.rds")

# Interactions
list_fungi_interactions_species <- list()
list_fungi_interactions_genus <- list()
list_fungi_interactions_family <- list()
list_fungi_interactions_order <- list()

for (i in 1:9) {
  interactions_species <- get.interactions(sciName = lists_species[[i]], endpoint = endpoint_url)
  list_fungi_interactions_species[[i]] <- interactions_species
  
  interactions_genus <- get.interactions(sciName = lists_genus[[i]], endpoint = endpoint_url)
  list_fungi_interactions_genus[[i]] <- interactions_genus
  
  interactions_family <- get.interactions(sciName = lists_family[[i]], endpoint = endpoint_url)
  list_fungi_interactions_family[[i]] <- interactions_family
  
  interactions_order <- get.interactions(sciName = lists_order[[i]], endpoint = endpoint_url)
  list_fungi_interactions_order[[i]] <- interactions_order
}

fungi_interactions_species <- do.call(rbind.data.frame, list_fungi_interactions_species) %>% mutate(level = "species")
fungi_interactions_genus <- do.call(rbind.data.frame, list_fungi_interactions_genus) %>% mutate(level = "genus")
fungi_interactions_family <- do.call(rbind.data.frame, list_fungi_interactions_family) %>% mutate(level = "family")
fungi_interactions_order <- do.call(rbind.data.frame, list_fungi_interactions_order) %>% mutate(level = "order")

fungi_interactions <- rbind(fungi_interactions_species, fungi_interactions_genus, fungi_interactions_family, fungi_interactions_order) %>% mutate(community = "fungi")

#write.csv(apply(fungi_interactions,2,as.character), "data/derived_data/fun_table/fun_fungi_interactions_2024-08-21.csv", row.names = F)
#write_rds(apply(fungi_interactions,2,as.character), "data/derived_data/fun_table/fun_fungi_interactions_2024-08-21.rds")
```

### protists

```{r}
# Split taxa in 10 lists
unique_species <- unique(tax_protists$species)
n <- length(unique_species)
size <- ceiling(n/3)
lists_species <- split(unique_species, ceiling(seq_along(unique_species)/size))

unique_genus <- unique(tax_protists$genus)
n <- length(unique_genus)
size <- ceiling(n/3)
lists_genus <- split(unique_genus, ceiling(seq_along(unique_genus)/size))

unique_family <- unique(tax_protists$family)
n <- length(unique_family)
size <- ceiling(n/3)
lists_family <- split(unique_family, ceiling(seq_along(unique_family)/size))

unique_order <- unique(tax_protists$order)
n <- length(unique_order)
size <- ceiling(n/3)
lists_order <- split(unique_order, ceiling(seq_along(unique_order)/size))
```

```{r}
# Guilds
list_protists_guilds_species <- list()
list_protists_guilds_genus <- list()
list_protists_guilds_family <- list()
list_protists_guilds_order <- list()

for (i in 1:3) {
  guilds_species <- get.guilds(sciName = lists_species[[i]], endpoint = endpoint_url)
  list_protists_guilds_species[[i]] <- guilds_species
  
  guilds_genus <- get.guilds(sciName = lists_genus[[i]], endpoint = endpoint_url)
  list_protists_guilds_genus[[i]] <- guilds_genus
  
  guilds_family <- get.guilds(sciName = lists_family[[i]], endpoint = endpoint_url)
  list_protists_guilds_family[[i]] <- guilds_family
  
  guilds_order <- get.guilds(sciName = lists_order[[i]], endpoint = endpoint_url)
  list_protists_guilds_order[[i]] <- guilds_order
}

protists_guilds_species <- do.call(rbind.data.frame, list_protists_guilds_species) %>% mutate(level = "species")
protists_guilds_genus <- do.call(rbind.data.frame, list_protists_guilds_genus) %>% mutate(level = "genus")
protists_guilds_family <- do.call(rbind.data.frame, list_protists_guilds_family) %>% mutate(level = "family")
protists_guilds_order <- do.call(rbind.data.frame, list_protists_guilds_order) %>% mutate(level = "order")

protists_guilds <- rbind(protists_guilds_species, protists_guilds_genus, protists_guilds_family, protists_guilds_order)  %>% mutate(community = "protists")

#write.csv(apply(protists_guilds,2,as.character), "data/derived_data/fun_table/fun_protists_guilds_2024-08-21.csv", row.names = F)
#write_rds(apply(protists_guilds,2,as.character), "data/derived_data/fun_table/fun_protists_guilds_2024-08-21.rds")

# Interactions
list_protists_interactions_species <- list()
list_protists_interactions_genus <- list()
list_protists_interactions_family <- list()
list_protists_interactions_order <- list()

for (i in 1:3) {
  interactions_species <- get.interactions(sciName = lists_species[[i]], endpoint = endpoint_url)
  list_protists_interactions_species[[i]] <- interactions_species
  
  interactions_genus <- get.interactions(sciName = lists_genus[[i]], endpoint = endpoint_url)
  list_protists_interactions_genus[[i]] <- interactions_genus
  
  interactions_family <- get.interactions(sciName = lists_family[[i]], endpoint = endpoint_url)
  list_protists_interactions_family[[i]] <- interactions_family
  
  interactions_order <- get.interactions(sciName = lists_order[[i]], endpoint = endpoint_url)
  list_protists_interactions_order[[i]] <- interactions_order
}

protists_interactions_species <- do.call(rbind.data.frame, list_protists_interactions_species) %>% mutate(level = "species")
protists_interactions_genus <- do.call(rbind.data.frame, list_protists_interactions_genus) %>% mutate(level = "genus")
protists_interactions_family <- do.call(rbind.data.frame, list_protists_interactions_family) %>% mutate(level = "family")
protists_interactions_order <- do.call(rbind.data.frame, list_protists_interactions_order) %>% mutate(level = "order")

protists_interactions <- rbind(protists_interactions_species, protists_interactions_genus, protists_interactions_family, protists_interactions_order) %>% mutate(community = "protists")

#write.csv(apply(protists_interactions,2,as.character), "data/derived_data/fun_table/fun_protists_interactions_2024-08-21.csv", row.names = F)
#write_rds(apply(protists_interactions,2,as.character), "data/derived_data/fun_table/fun_protists_interactions_2024-08-21.rds")
```

# Export fun_table


```{r}
fun_guilds_macrofauna_surface <- read.csv("data/derived_data/fun_table/fun_macrofauna_surface_guilds_2024-08-21.csv")
fun_guilds_macrofauna_aerial <- read.csv("data/derived_data/fun_table/fun_macrofauna_aerial_guilds_2024-08-21.csv")
fun_guilds_macrofauna_foliar <- read.csv("data/derived_data/fun_table/fun_macrofauna_foliar_guilds_2024-08-21.csv")
fun_guilds_nematodes <- read.csv("data/derived_data/fun_table/fun_nematodes_guilds_2024-08-21.csv")
fun_guilds_micro_arthropodes <- read.csv("data/derived_data/fun_table/fun_micro_arthropodes_guilds_2024-08-21.csv")
fun_guilds_bacteria <- read.csv("data/derived_data/fun_table/fun_bacteria_guilds_2024-08-21.csv")
fun_guilds_fungi <- read.csv("data/derived_data/fun_table/fun_fungi_guilds_2024-08-21.csv")
fun_guilds_protists <- read.csv("data/derived_data/fun_table/fun_protists_guilds_2024-08-21.csv")

fun_guilds_ALL <- bind_rows(fun_guilds_macrofauna_surface, fun_guilds_macrofauna_aerial, fun_guilds_macrofauna_foliar, fun_guilds_micro_arthropodes, fun_guilds_nematodes, fun_guilds_bacteria, fun_guilds_fungi, fun_guilds_protists) # bind all communities in one file
#write.csv(fun_guilds_ALL, "data/derived_data/fun_table/fun_guilds_all_2024-08-21.csv", row.names = F)
#write_rds(fun_guilds_ALL, "data/derived_data/fun_table/fun_guilds_all_2024-08-21.rds")
```

```{r}
fun_interactions_macrofauna_surface <- read.csv("data/derived_data/fun_table/fun_macrofauna_surface_interactions_2024-08-21.csv")
fun_interactions_macrofauna_aerial <- read.csv("data/derived_data/fun_table/fun_macrofauna_aerial_interactions_2024-08-21.csv")
fun_interactions_macrofauna_foliar <- read.csv("data/derived_data/fun_table/fun_macrofauna_foliar_interactions_2024-08-21.csv")
fun_interactions_nematodes <- read.csv("data/derived_data/fun_table/fun_nematodes_interactions_2024-08-21.csv")
fun_interactions_micro_arthropodes <- read.csv("data/derived_data/fun_table/fun_micro_arthropodes_interactions_2024-08-21.csv")
fun_interactions_bacteria <- read.csv("data/derived_data/fun_table/fun_bacteria_interactions_2024-08-21.csv")
fun_interactions_fungi <- read.csv("data/derived_data/fun_table/fun_fungi_interactions_2024-08-21.csv")
fun_interactions_protists <- read.csv("data/derived_data/fun_table/fun_protists_interactions_2024-08-21.csv")

fun_interactions_ALL <- bind_rows(fun_interactions_macrofauna_surface, fun_interactions_macrofauna_aerial, fun_interactions_macrofauna_foliar, fun_interactions_micro_arthropodes, fun_interactions_nematodes, fun_interactions_bacteria, fun_interactions_fungi, fun_interactions_protists) # bind all communities in one file
#write.csv(fun_interactions_ALL, "data/derived_data/fun_table/fun_interactions_all_2024-08-21.csv", row.names = F)
#write_rds(fun_interactions_ALL, "data/derived_data/fun_table/fun_interactions_all_2024-08-21.rds")
```

# Check tracking

## all

```{r}
fun_guilds_ALL <- read.csv("data/derived_data/fun_table/fun_guilds_all_2024-08-21.csv")
fun_interactions_ALL <- read.csv("data/derived_data/fun_table/fun_interactions_all_2024-08-21.csv")

step_3_guilds_common <- fun_guilds_ALL %>% group_by(community, level) %>% summarise(n_distinct_queryName = n_distinct(queryName), n_distinct_matchName = n_distinct(matchName), n_distinct_exact_match = n_distinct(queryName [queryName == matchName])) %>% mutate(fun = "guilds", step = "step_3")

step_3_interactions_common <- fun_interactions_ALL %>% group_by(community, level) %>% summarise(n_distinct_queryName = n_distinct(queryName), n_distinct_matchName = n_distinct(matchName), n_distinct_exact_match = n_distinct(queryName [queryName == matchName])) %>% mutate(fun = "interactions", step = "step_3")

step_3_common <- rbind(step_3_guilds_common, step_3_interactions_common)

#write.csv(step_3_common, "analyses/tracking/step_3_common_2024-09-13.csv", row.names = F)

step_3_common$community <- ordered(step_3_common$community, order_community)
step_3_common$level <- ordered(step_3_common$level, order_level)

M1 <- ggplot(step_3_common, aes(x = level, y = n_distinct_queryName, fill = community)) + frame_simple + geom_bar(stat="identity", position = position_dodge2(preserve = "single")) + labs(title="Number of distinct assigned taxa") + scale_y_continuous(labels = scales::comma) + scale_fill_brewer(palette = "Set2") + guides(size = F) + facet_grid("fun", scales="free")
M1
#ggsave("analyses/tracking/tracking_step3_n_distinct_queryName_2024-09-13.png", width=6, height=4.5)

M2 <- ggplot(step_3_common, aes(x = level, y = n_distinct_matchName, fill = community)) + frame_simple + geom_bar(stat="identity", position = position_dodge2(preserve = "single")) + labs(title="Number of distinct taxa matching assigned taxa", subtitle = "(exact taxa or lower taxonomic levels)") + scale_y_continuous(labels = scales::comma) + scale_fill_brewer(palette = "Set2") + guides(size = F) + facet_grid("fun", scales="free")
M2
#ggsave("analyses/tracking/tracking_step3_n_distinct_matchName_2024-09-13.png", width=6, height=4.5)

M3 <- ggplot(step_3_common, aes(x = level, y = n_distinct_exact_match, fill = community)) + frame_simple + geom_bar(stat="identity", position = position_dodge2(preserve = "single")) + labs(title="Number of distinct taxa matching exactly assigned taxa") + scale_y_continuous(labels = scales::comma) + scale_fill_brewer(palette = "Set2") + guides(size = F) + facet_grid("fun", scales="free")
M3
#ggsave("analyses/tracking/tracking_step3_n_distinct_exact_match_2024-09-13.png", width=6, height=4.5)
```

```{r}
step_3_guilds_spe <- fun_guilds_ALL %>% group_by(community, level) %>% summarise(n_distinct_guildName = n_distinct(guildName)) %>% mutate(fun = "guilds", step = "step_3")
#write.csv(step_3_guilds_spe, "analyses/tracking/step_3_guilds_spe_2024-09-13.csv", row.names = F)

step_3_guilds_spe$community <- ordered(step_3_guilds_spe$community, order_community)
step_3_guilds_spe$level <- ordered(step_3_guilds_spe$level, order_level)

S1 <- ggplot(step_3_guilds_spe, aes(x = level, y = n_distinct_guildName, fill = community)) + frame_simple + geom_bar(stat="identity", position = position_dodge2(preserve = "single")) + labs(title="Number of distinct guilds") + scale_y_continuous(labels = scales::comma) + scale_fill_brewer(palette = "Set2") + guides(size = F) + facet_grid("fun", scales="free")
S1
#ggsave("analyses/tracking/tracking_step3_n_distinct_guildName_2024-09-13.png", width=6, height=4.5)

step_3_interactions_spe <- fun_interactions_ALL %>% group_by(community, level) %>% summarise(n_distinct_interactionName = n_distinct(interactionName), n_distinct_resourceName = n_distinct(resourceName)) %>% mutate(fun = "interactions", step = "step_3")
#write.csv(step_3_interactions_spe, "analyses/tracking/step_3_interactions_spe_2024-09-13.csv", row.names = F)

step_3_interactions_spe$community <- ordered(step_3_interactions_spe$community, order_community)
step_3_interactions_spe$level <- ordered(step_3_interactions_spe$level, order_level)

S2 <- ggplot(step_3_interactions_spe, aes(x = level, y = n_distinct_interactionName, fill = community)) + frame_simple + geom_bar(stat="identity", position = position_dodge2(preserve = "single")) + labs(title="Number of distinct interactions") + scale_y_continuous(labels = scales::comma) + scale_fill_brewer(palette = "Set2") + guides(size = F) + facet_grid("fun", scales="free")
S2
#ggsave("analyses/tracking/tracking_step3_n_distinct_interactionName_2024-09-13.png", width=6, height=4.5)

S3 <- ggplot(step_3_interactions_spe, aes(x = level, y = n_distinct_resourceName, fill = community)) + frame_simple + geom_bar(stat="identity", position = position_dodge2(preserve = "single")) + labs(title="Number of distinct resources") + scale_y_continuous(labels = scales::comma) + scale_fill_brewer(palette = "Set2") + guides(size = F) + facet_grid("fun", scales="free")
S3
#ggsave("analyses/tracking/tracking_step3_n_distinct_resourceName_2024-09-13.png", width=6, height=4.5)
```

```{r}
step_2 <- read.csv("analyses/tracking/step_2_bylevel_2024-09-13.csv")[,-c(2:4)]

step_2_long <- pivot_longer(step_2, cols=c(2:5), names_to = "level", values_to = "n_distinct_name")

## queryName

step_3_perc_queryName <- step_3_common %>% inner_join(step_2_long, by=c("community","level")) %>% mutate(perc_queryName = n_distinct_queryName*100/n_distinct_name)

step_3_perc_queryName$community <- ordered(step_3_perc_queryName$community, order_community)
step_3_perc_queryName$level <- ordered(step_3_perc_queryName$level, order_level)

step_3_wide_queryName <- pivot_wider(step_3_perc_queryName[,-c(3,4,5,7,8)], names_from = level, values_from = perc_queryName)
step_3_wide_queryName <- step_3_wide_queryName[,c(1,5,3,4,6,2)] %>% mutate_all(~replace(., is.na(.), 0)) %>% mutate_if(is.numeric, round, digits = 2) %>% mutate(version = "queryName")
#write.csv(step_3_wide_queryName, "analyses/tracking/step_3_perc_queryName_2024-09-13.csv", row.names = F)

P1 <- ggplot(step_3_perc_queryName, aes(x = level, y = perc_queryName, fill = community)) + frame_simple + geom_bar(stat="identity", position = position_dodge2(preserve = "single")) + labs(title="Percentage of assigned taxa") + scale_y_continuous(labels = scales::comma) + scale_fill_brewer(palette = "Set2") + guides(size = F) + facet_grid("fun", scales="free") + ylim(0, 100)
P1
#ggsave("analyses/tracking/tracking_step3_perc_queryName_2024-09-13.png", width=6, height=4.5)

## matchName

step_3_perc_matchName <- step_3_common %>% inner_join(step_2_long, by=c("community","level")) %>% mutate(perc_matchName = n_distinct_matchName*100/n_distinct_name)

step_3_perc_matchName$community <- ordered(step_3_perc_matchName$community, order_community)
step_3_perc_matchName$level <- ordered(step_3_perc_matchName$level, order_level)

step_3_wide_matchName <- pivot_wider(step_3_perc_matchName[,-c(3,4,5,7,8)], names_from = level, values_from = perc_matchName)
step_3_wide_matchName <- step_3_wide_matchName[,c(1,5,3,4,6,2)] %>% mutate_all(~replace(., is.na(.), 0)) %>% mutate_if(is.numeric, round, digits = 2) %>% mutate(version = "matchName")
#write.csv(step_3_wide_matchName, "analyses/tracking/step_3_perc_matchName_2024-09-13.csv", row.names = F)

P2 <- ggplot(step_3_perc_matchName, aes(x = level, y = perc_matchName, fill = community)) + frame_simple + geom_bar(stat="identity", position = position_dodge2(preserve = "single")) + labs(title="Percentage of taxa matching assigned taxa", subtitle = "(exact taxa or lower taxonomic levels)") + scale_y_continuous(labels = scales::comma) + scale_fill_brewer(palette = "Set2") + guides(size = F) + facet_grid("fun", scales="free")
P2
#ggsave("analyses/tracking/tracking_step3_perc_matchName_2024-09-13.png", width=6, height=4.5)

## matchName

step_3_perc_exact_match <- step_3_common %>% inner_join(step_2_long, by=c("community","level")) %>% mutate(perc_exact_match = n_distinct_exact_match*100/n_distinct_name)

step_3_perc_exact_match$community <- ordered(step_3_perc_exact_match$community, order_community)
step_3_perc_exact_match$level <- ordered(step_3_perc_exact_match$level, order_level)

step_3_wide_exact_match <- pivot_wider(step_3_perc_exact_match[,-c(3,4,5,7,8)], names_from = level, values_from = perc_exact_match)
step_3_wide_exact_match <- step_3_wide_exact_match[,c(1,5,3,4,6,2)] %>% mutate_all(~replace(., is.na(.), 0)) %>% mutate_if(is.numeric, round, digits = 2) %>% mutate(version = "exact_match")
#write.csv(step_3_wide_exact_match, "analyses/tracking/step_3_perc_exact_match_2024-09-13.csv", row.names = F)

P3 <- ggplot(step_3_perc_exact_match, aes(x = level, y = perc_exact_match, fill = community)) + frame_simple + geom_bar(stat="identity", position = position_dodge2(preserve = "single")) + labs(title="Percentage of taxa matching exactly assigned taxa") + scale_y_continuous(labels = scales::comma) + scale_fill_brewer(palette = "Set2") + guides(size = F) + facet_grid("fun", scales="free") + ylim(0, 100)
P3
#ggsave("analyses/tracking/tracking_step3_perc_exact_match_2024-09-13.png", width=6, height=4.5)

step_3_wide <- rbind(step_3_wide_queryName, step_3_wide_matchName, step_3_wide_exact_match)
#write.csv(step_3_wide, "analyses/tracking/step_3_perc_bind_2024-09-13.csv", row.names = F)
```

```{r}
fun_guilds_ALL <- read.csv("data/derived_data/fun_table/fun_guilds_all_2024-08-21.csv")
tax_ALL <- read.csv("data/derived_data/tax_table/final_files/tax_all_2024-09-13.csv")

fun_guilds_ALL_family <- fun_guilds_ALL %>% filter(level %in% c("family"))
family_match_guilds <- unique(fun_guilds_ALL_family$matchName)

fun_guilds_ALL_family_au <- tax_ALL[,-c(2:5)] %>% mutate(
  family = ifelse(family %in% family_match_guilds, "assigned", "unassigned"),
  genus = ifelse(genus %in% family_match_guilds, "assigned", "unassigned"),
  species = ifelse(species %in% family_match_guilds, "assigned", "unassigned"))

tidy_ALL <- read.csv("data/derived_data/tidy_data/tidy_all_2024-09-13.csv")
tidy_ALL_agg <- aggregate(abundance ~ finest_name, data = tidy_ALL, FUN = sum)
tidytaxfun_ALL_au <- merge(fun_guilds_ALL_family_au, tidy_ALL_agg, by = "finest_name")

tidytaxfun_ALL_au <- tidytaxfun_ALL_au %>% mutate(any_level = if_else(family == "assigned" | genus == "assigned" | species == "assigned", "assigned", "unassigned"))

tidytaxfun_ALL_au_long <- pivot_longer(tidytaxfun_ALL_au, cols=c(2:4,7), names_to = "level", values_to = "assignation")

tidytaxfun_ALL_au_long_abund <- aggregate(abundance ~ community*level*assignation, data = tidytaxfun_ALL_au_long, FUN = sum)
#write.csv(tidytaxfun_ALL_au_long_abund, "analyses/tracking/tidytaxfun_assignation_abund_2024-09-19.csv", row.names = F)

tidytaxfun_ALL_au_long_distinct <- tidytaxfun_ALL_au_long %>%  group_by(community, level, assignation) %>% summarise(n_distinct_finest_name = n_distinct(finest_name))
#write.csv(tidytaxfun_ALL_au_long_distinct, "analyses/tracking/tidytaxfun_assignation_distinct_2024-09-19.csv", row.names = F)

tidytaxfun_ALL_au_long_abund_graph <- filter(tidytaxfun_ALL_au_long_abund, level != "any_level")

tidytaxfun_ALL_au_long_abund_graph$community <- ordered(tidytaxfun_ALL_au_long_abund_graph$community, order_community)
tidytaxfun_ALL_au_long_abund_graph$level <- ordered(tidytaxfun_ALL_au_long_abund_graph$level, order_level)

L2 <- ggplot(tidytaxfun_ALL_au_long_abund_graph, aes(level, abundance, fill=assignation)) + geom_bar(stat="identity") + frame_simple + facet_grid(vars(community), scales="free") + labs(title="Abundance of assigned or unassigned organisms", subtitle="Functional assignation", x ="Taxonomic level", y ="abundance") + scale_y_continuous(labels = scales::comma) + scale_fill_brewer(palette = "Set2") + theme(axis.text.x = element_text(angle = 45, hjust=1))
L2
#ggsave("analyses/tracking/tracking_taxononomic_assignation_functional_binary_2024-09-13.png", width=5.5, height=11)
```

```{r}
# To obtain the list of families that are not assigned with Gratin (not in queryName)
family_match_guilds_queryName <- unique(fun_guilds_ALL_family$queryName)

fun_guilds_ALL_family_au_queryName <- tax_ALL[,-c(2:5,7,8)] %>% mutate(
  family = ifelse(family %in% family_match_guilds_queryName, "assigned", family))

fun_guilds_ALL_family_au_queryName <- fun_guilds_ALL_family_au_queryName[!fun_guilds_ALL_family_au_queryName$family %in% c("assigned", "unassigned"),]

fun_guilds_ALL_family_au_queryName <- fun_guilds_ALL_family_au_queryName %>% 
   separate(col = finest_name , into = c("finest_name","OTU"), sep = "::")

fun_guilds_ALL_family_au_queryName <- fun_guilds_ALL_family_au_queryName[,-2]

fun_guilds_ALL_family_au_queryName <- fun_guilds_ALL_family_au_queryName[!duplicated(fun_guilds_ALL_family_au_queryName), ]

#write.csv(fun_guilds_ALL_family_au_queryName, "analyses/tracking/list_unassigned_2024-09-19.csv", row.names = F)
```

```{r}
L3_tax <- L1 + theme(legend.position = "none", title.position = "none") + labs(title="Abundance of assigned organisms")

L3_fun <- ggplot(tidytaxfun_ALL_au_long_abund_graph, aes(level, abundance, fill=assignation), aes(level, abundance, fill=assignation)) + geom_bar(stat="identity") + frame_simple + facet_grid(vars(community), scales="free") + labs(title=" ", subtitle="Functional assignation (guilds)", x ="Taxonomic level", y ="abundance") + scale_y_continuous(labels = scales::comma) + scale_fill_brewer(palette = "Set2") + theme(axis.text.x = element_text(angle = 45, hjust=1))

L3_both <- plot_grid(L3_tax, L3_fun, labels = c('A', 'B'), ncol=2, rel_widths = c(1, 0.75))
L3_both
#ggsave("analyses/tracking/tracking_taxfun_assignation_binary_2024-09-19.png", width=10, height=11)

#svglite(file="analyses/tracking/tracking_taxfun_assignation_binary_2024-09-19.svg", width = 10, height = 11, bg = "white", pointsize = 16)
L3_both
#dev.off()
```

```{r}
# Calculate percentage assigned

## tidytax
tidytax_ALL_au_long_perc <- pivot_wider(tidytax_ALL_au_long_abund, names_from = assignation, values_from = abundance)

tidytax_ALL_au_long_perc[is.na(tidytax_ALL_au_long_perc)] <- 0

tidytax_ALL_au_long_perc <- tidytax_ALL_au_long_perc %>% mutate(perc_assigned = assigned*100/(assigned+unassigned), assignation = "taxonomic") %>% mutate_if(is.numeric, round, digits = 2)

## tidytaxfun

tidytaxfun_ALL_au_long_perc <- pivot_wider(tidytaxfun_ALL_au_long_abund, names_from = assignation, values_from = abundance)

tidytaxfun_ALL_au_long_perc[is.na(tidytaxfun_ALL_au_long_perc)] <- 0

tidytaxfun_ALL_au_long_perc <- tidytaxfun_ALL_au_long_perc %>% mutate(perc_assigned = assigned*100/(assigned+unassigned), assignation = "functional") %>% mutate_if(is.numeric, round, digits = 2)

## bind and export
#write.csv(rbind(tidytax_ALL_au_long_perc, tidytaxfun_ALL_au_long_perc), "analyses/tracking/assignation_perc_abund_2024-09-13.csv", row.names = F)
```

```{r}
family_match_guilds <- unique(fun_guilds_ALL_family$matchName)
fun_guilds_ALL_genus <- fun_guilds_ALL %>% filter(level %in% c("genus"))
genus_match_guilds <- unique(fun_guilds_ALL_genus$matchName)
fun_guilds_ALL_species <- fun_guilds_ALL %>% filter(level %in% c("species"))
species_match_guilds <- unique(fun_guilds_ALL_species$matchName)

tax_ALL_tf_match_guilds <- tax_ALL_tf %>% mutate(
  family = ifelse(family %in% family_match_guilds, family, "unassigned"),
  genus = ifelse(genus %in% genus_match_guilds, genus, "unassigned"),
  species = ifelse(species %in% species_match_guilds, species, "unassigned"))

tax_ALL_tf_match_guilds_long <- pivot_longer(tax_ALL_tf_match_guilds, cols=c(2:4), names_to = "level", values_to = "taxa")

tax_ALL_tf_match_guilds_long_agg <- tax_ALL_tf_match_guilds_long %>% group_by(community, level) %>% summarise(n_distinct_taxa = n_distinct(taxa))

tax_ALL_tf_match_guilds_long_agg$community <- ordered(tax_ALL_tf_match_guilds_long_agg$community, order_community)
tax_ALL_tf_match_guilds_long_agg$level <- ordered(tax_ALL_tf_match_guilds_long_agg$level, order_level)

L4 <- ggplot(tax_ALL_tf_match_guilds_long_agg, aes(level, n_distinct_taxa, fill=community)) + geom_bar(stat="identity") + frame_simple  + facet_grid(vars(community), scales="free") + labs(title = "Number of taxa assigned to a guild", subtitle="Functional assignation", x ="Taxonomic level", y ="Number of distinct taxa names") + scale_y_continuous(labels = scales::comma) + scale_fill_brewer(palette = "Set2") + theme(axis.text.x = element_text(angle = 45, hjust=1))
L4
#ggsave("analyses/tracking/tracking_taxononomic_assignation_functional_unique_2024-09-19.png", width=4, height=11)

tax_ALL_tf_match_guilds_long_agg_wide <- pivot_wider(tax_ALL_tf_match_guilds_long_agg, names_from = level, values_from = n_distinct_taxa)
#write.csv(tax_ALL_tf_match_guilds_long_agg_wide, "analyses/tracking/step_3_bylevel_2024-09-19.csv", row.names = F)
```

```{r}
L5_tax <- ggplot(tax_ALL_long_sum, aes(level, n_distinct_taxa, fill=community)) + geom_bar(stat="identity") + frame_simple  + facet_grid(vars(community), scales="free") + labs(title="Number of assigned organisms", subtitle = "Taxonomic assignation",x ="Taxonomic level") + scale_y_continuous(labels = scales::comma) + scale_fill_brewer(palette = "Set2") + theme(axis.text.x = element_text(angle = 45, hjust=1)) + theme(legend.position = "none") # cf T1

L5_fun <- ggplot(tax_ALL_tf_match_guilds_long_agg, aes(level, n_distinct_taxa, fill=community)) + geom_bar(stat="identity") + frame_simple  + facet_grid(vars(community), scales="free") + labs(title = " ", subtitle="Functional assignation (guilds)", x ="Taxonomic level", y =" ") + scale_y_continuous(labels = scales::comma) + scale_fill_brewer(palette = "Set2") + theme(axis.text.x = element_text(angle = 45, hjust=1)) # cf L4

L5_both <- plot_grid(L5_tax, L5_fun, labels = c('A', 'B'), ncol=2, rel_widths = c(1, 1))
L5_both
#ggsave("analyses/tracking/tracking_taxfun_assignation_number_2024-09-19.png", width=8, height=11)

#svglite(file="analyses/tracking/tracking_taxfun_assignation_number_2024-09-19.svg", width = 8, height = 11, bg = "white", pointsize = 16)
L5_both
#dev.off()
```

# 4.  Create phyloseq objects

help with https://vaulot.github.io/tutorials/Phyloseq_tutorial.html

```{r}
## import data
tidy_data <- read.csv("data/derived_data/tidy_data/tidy_all_2024-09-13.csv")
tax_table <- read.csv("data/derived_data/tax_table/final_files/tax_all_2024-09-13.csv")

## add separate info columns and recreate code columns
tidy_data_PO <- tidy_data %>% 
   separate(col = code_uhbr , into = c("land_use","habitat","block","replicate"), sep = "_")

tidy_data_PO$code_uhbr <- paste(tidy_data_PO$land_use, tidy_data_PO$habitat, tidy_data_PO$block, tidy_data_PO$replicate, sep = "_")

tidy_data_PO$code_uh <- paste(tidy_data_PO$land_use, tidy_data_PO$habitat, sep = "_")

tidy_data_PO$code_uhb <- paste(tidy_data_PO$land_use, tidy_data_PO$habitat, tidy_data_PO$block, sep = "_")

tidy_data_PO$date <- gsub("/", ".", tidy_data_PO$date)

tidy_data_PO$code_uhbrd <- paste(tidy_data_PO$land_use, tidy_data_PO$habitat, tidy_data_PO$block, tidy_data_PO$replicate, tidy_data_PO$date, sep = "_")

tidy_data_PO$code_uhbrdmc <- paste(tidy_data_PO$code_uhbr, tidy_data_PO$date, tidy_data_PO$method, tidy_data_PO$community, sep = "_")

tidy_data_PO <- tidy_data_PO %>% mutate(
    spatial = case_when(
      code_uh == "AF_C" | code_uh == "AF_T" ~ "hetero",
      code_uh == "MC_C" | code_uh == "TP_T" ~ "homo"))

## reorder columns
tidy_data_PO <- tidy_data_PO[, c(5,9,1:4,6:8,11,12,10,13,14,15)]

# samples_df = metadata
## remove useless columns
metadata_PO <- tidy_data_PO[,-c(1,2)]

## remove duplicates
metadata_PO <- metadata_PO[!duplicated(metadata_PO),]
```

```{r}
## import data
fun_table <- read.csv("data/derived_data/fun_table/fun_guilds_all_2024-08-21.csv")

## select the level to make assignations
fun_table_PO <- fun_table %>% filter(level %in% c("family"))

## remove "plant exudate feeder" for convenience in subsetting potential auxiliaries (and why aren't they classified in herbivores ?)
fun_table_PO <- fun_table_PO %>% filter(guildName != "plant exudate feeder")

## Prepare guild_levels_table
guild_table <- read.csv("data/raw_data/guilds.csv")[,-1]

guild_table$level_all <- paste(guild_table$Ancestors, guild_table$Guild, sep = "|")

guild_table <- guild_table %>% separate_wider_delim(level_all, "|", too_few = "align_start", names = c("guild_level_1", "guild_level_2", "guild_level_3", "guild_level_4", "guild_level_5"))

#write.csv(guild_table, "data/derived_data/fun_table/guild_levels_table.csv")

## remove useless columns
fun_table_PO <- fun_table_PO[,c(3,6)]

## Add the guild levels
fun_table_PO <- merge(guild_table, fun_table_PO, by.x='Guild', by.y='guildName')

## replace NA by unassigned
fun_table_PO[is.na(fun_table_PO)] <- "unassigned"

## remove useless columns
fun_table_PO <- fun_table_PO[,-c(1,2)]

## remove duplicates
fun_table_PO <- fun_table_PO[!duplicated(fun_table_PO), ]

lignes_solitaires_level_2 <- fun_table_PO %>%
  group_by(matchName, guild_level_2) %>%
  mutate(count_guild = n()) %>%
  ungroup() %>%
  filter(count_guild == 1) %>%
  select(-count_guild)

lignes_dedupliquees_level_3 <- fun_table_PO %>%
  group_by(guild_level_2) %>%
  mutate(has_multiple_values = n_distinct(guild_level_3) > 1) %>%
  filter(!(guild_level_3 == "unassigned" & has_multiple_values)) %>%
  select(-has_multiple_values) %>%
  ungroup()

cleaned_level_3 <- bind_rows(lignes_solitaires_level_2, lignes_dedupliquees_level_3)

lignes_solitaires_level_3 <- cleaned_level_3 %>%
  group_by(matchName, guild_level_3) %>%
  mutate(count_guild = n()) %>%
  ungroup() %>%
  filter(count_guild == 1) %>%
  select(-count_guild)

lignes_dedupliquees_level_4 <- cleaned_level_3 %>% anti_join(lignes_solitaires_level_3, by = colnames(cleaned_level_3)) %>% 
  group_by(guild_level_3) %>%
  mutate(has_multiple_values = n_distinct(guild_level_4) > 1) %>%
  filter(!(guild_level_4 == "unassigned" & has_multiple_values)) %>%
  select(-has_multiple_values) %>%
  ungroup()

cleaned_level_4 <- bind_rows(lignes_solitaires_level_3, lignes_dedupliquees_level_4)

lignes_solitaires_level_4 <- cleaned_level_4 %>%
  group_by(matchName, guild_level_4) %>%
  mutate(count_guild = n()) %>%
  ungroup() %>%
  filter(count_guild == 1) %>%
  select(-count_guild)

lignes_dedupliquees_level_5 <- cleaned_level_4 %>% anti_join(lignes_solitaires_level_4, by = colnames(cleaned_level_4)) %>% 
  group_by(guild_level_4) %>%
  mutate(has_multiple_values = n_distinct(guild_level_5) > 1) %>%
  filter(!(guild_level_5 == "unassigned" & has_multiple_values)) %>%
  select(-has_multiple_values) %>%
  ungroup()

cleaned_level_5 <- bind_rows(lignes_solitaires_level_4, lignes_dedupliquees_level_5)

## adjust columns to manage multiple assignations
adjusted_PO <- cleaned_level_5
adjusted_PO <- adjusted_PO %>% mutate(abr_4_guild_level_5 = paste0(substr(guild_level_4, 1, 3), "_", guild_level_5))
adjusted_PO <- adjusted_PO %>% mutate(abr_3_guild_level_4 = paste0(substr(guild_level_3, 1, 3), "_", guild_level_4))
adjusted_PO <- adjusted_PO %>% mutate(abr_2_guild_level_3 = paste0(substr(guild_level_2, 1, 3), "_", guild_level_3))
adjusted_PO <- adjusted_PO %>% mutate(abr_1_guild_level_2 = paste0(substr(guild_level_1, 1, 3), "_", guild_level_2))
adjusted_PO <- adjusted_PO %>% mutate(abr_all_guild_level = paste0(substr(guild_level_1, 1, 3), "_", substr(guild_level_2, 1, 3), "_", substr(guild_level_3, 1, 3), "_", substr(guild_level_4, 1, 3), "_", substr(guild_level_5, 1, 3)))

adjusted_PO <- adjusted_PO %>%
  group_by(matchName) %>%
  mutate(abr_1_guild_level_2_AND = str_c(sort(unique(abr_1_guild_level_2)), collapse = "AND"),
         abr_2_guild_level_3_AND = str_c(sort(unique(abr_2_guild_level_3)), collapse = "AND"),
         abr_3_guild_level_4_AND = str_c(sort(unique(abr_3_guild_level_4)), collapse = "AND"),
         abr_4_guild_level_5_AND = str_c(sort(unique(abr_4_guild_level_5)), collapse = "AND"),
         abr_all_guild_level_AND = str_c(sort(unique(abr_all_guild_level)), collapse = "AND")) %>%
  ungroup()

adjusted_PO_min <- adjusted_PO[,c(6,12:16)]

fun_table_PO <- adjusted_PO_min[!duplicated(adjusted_PO_min), ]
```

```{r}
## import data
fun_table_inter <- read.csv("data/derived_data/fun_table/fun_interactions_all_2024-08-21.csv")

## select the level to make assignations
fun_table_inter_PO <- fun_table_inter %>% filter(level %in% c("family"))

## remove useless columns
fun_table_inter_PO <- fun_table_inter_PO[,c(1,3,6,8)]
```

## macrofauna_surface

### Version raw

```{r}
# otu_table = tidy_data
## select the community
tidy_data_PO_macrofauna_surface <- tidy_data_PO %>% filter(community %in% c("macrofauna_surface"))

## remove useless columns
tidy_data_PO_macrofauna_surface <- tidy_data_PO_macrofauna_surface[,c(1,2,13)]

## transform table to taxa_are_rows = TRUE
tidy_data_PO_macrofauna_surface <- pivot_wider(tidy_data_PO_macrofauna_surface, names_from = code_uhbrd, values_from = abundance)

## replace NA by 0
tidy_data_PO_macrofauna_surface[is.na(tidy_data_PO_macrofauna_surface)] <- 0

## use "finest_name" as rownames
tidy_data_PO_macrofauna_surface <- tidy_data_PO_macrofauna_surface %>% tibble::column_to_rownames("finest_name")

## transform tbl_df to matrix table
tidy_data_PO_macrofauna_surface <- as.matrix(tidy_data_PO_macrofauna_surface)

# tax_table = tax_table # + fun_table
## select the community
tax_table_PO_macrofauna_surface <- tax_table %>% filter(community %in% c("macrofauna_surface"))

## remove useless columns
tax_table_PO_macrofauna_surface <- tax_table_PO_macrofauna_surface[,-c(9)]

########## Add fun_table here ########## 
tax_table_PO_macrofauna_surface_fun <- left_join(tax_table_PO_macrofauna_surface, fun_table_PO, by = c("family" = "matchName"))

tax_table_PO_macrofauna_surface_fun[is.na(tax_table_PO_macrofauna_surface_fun)] <- "unassigned"

tax_table_PO_macrofauna_surface_fun_inter <- tax_table_PO_macrofauna_surface_fun %>%
  rowwise() %>%
  mutate(interaction = if_else(any(c_across(1:8) %in% fun_table_inter_PO$matchName), "observed", "not observed")) %>%
  ungroup()

tax_table_PO_macrofauna_surface <- tax_table_PO_macrofauna_surface_fun_inter
########## END ########## 

## use "finest_name" as rownames
tax_table_PO_macrofauna_surface <- tax_table_PO_macrofauna_surface %>% tibble::column_to_rownames("finest_name")

## transform tbl_df to matrix table
tax_table_PO_macrofauna_surface <- as.matrix(tax_table_PO_macrofauna_surface)

# sam_data = metadata
## select the community
metadata_PO_macrofauna_surface <- metadata_PO %>% filter(community %in% c("macrofauna_surface"))

## choose the rownames
rownames(metadata_PO_macrofauna_surface) <- metadata_PO_macrofauna_surface[,11] # code_uhbrd

# create phyloseq object
tidy_data_PO_macrofauna_surface = otu_table(tidy_data_PO_macrofauna_surface, taxa_are_rows = TRUE)

tax_table_PO_macrofauna_surface = tax_table(tax_table_PO_macrofauna_surface)

metadata_PO_macrofauna_surface = sample_data(metadata_PO_macrofauna_surface)
  
PO_macrofauna_surface <- phyloseq(tidy_data_PO_macrofauna_surface, tax_table_PO_macrofauna_surface, metadata_PO_macrofauna_surface)

# check PO
PO_macrofauna_surface
sample_names(PO_macrofauna_surface)
rank_names(PO_macrofauna_surface)
sample_variables(PO_macrofauna_surface)

#write_rds(PO_macrofauna_surface, "data/derived_data/phyloseq_objects/PO_macrofauna_surface_raw_2024-09-23.rds")
#PO_macrofauna_surface_df <- phyloseq_to_df(PO_macrofauna_surface, addtax = T, addtot = T, addmaxrank = F, sorting = "NULL")
#write.csv(PO_macrofauna_surface_df, "data/derived_data/phyloseq_objects/PO_macrofauna_surface_raw_2024-09-23.csv", row.names = FALSE)
```

### Version stand

```{r}
# standardize abundances to the median sequencing depth
total = median(sample_sums(PO_macrofauna_surface))
standf = function(x, t=total) round(t*(x/sum(x)))
PO_macrofauna_surface_stand = transform_sample_counts(PO_macrofauna_surface, standf)

# check PO
PO_macrofauna_surface_stand

#write_rds(PO_macrofauna_surface_stand, "data/derived_data/phyloseq_objects/PO_macrofauna_surface_stand_2024-09-23.rds")
#PO_macrofauna_surface_df <- phyloseq_to_df(PO_macrofauna_surface_stand, addtax = T, addtot = T, addmaxrank = F, sorting = "NULL")
#write.table(PO_macrofauna_surface_df, "data/derived_data/phyloseq_objects/PO_macrofauna_surface_stand_2024-09-23.csv", row.names = FALSE)

propfun = function(x){
  xprop = (x / sum(x))
  return(xprop)
}
PO_macrofauna_surface_stand_prop <- transform_sample_counts(PO_macrofauna_surface_stand, fun = propfun)

#write_rds(PO_macrofauna_surface_stand_prop, "data/derived_data/phyloseq_objects/PO_macrofauna_surface_stand_prop_2024-09-23.rds")
#PO_macrofauna_surface_df <- phyloseq_to_df(PO_macrofauna_surface_stand_prop, addtax = T, addtot = T, addmaxrank = F, sorting = "NULL")
#write.table(PO_macrofauna_surface_df, "data/derived_data/phyloseq_objects/PO_macrofauna_surface_stand_prop_2024-09-23.csv", row.names = FALSE)
```

### Version norm

```{r}
# normalize
PO_macrofauna_surface_norm <- normalize_prop_pq(PO_macrofauna_surface, base_log = 2, digits = 0)

# check PO
PO_macrofauna_surface_norm

#write_rds(PO_macrofauna_surface_norm, "data/derived_data/phyloseq_objects/PO_macrofauna_surface_norm_2024-09-23.rds")
#PO_macrofauna_surface_df <- phyloseq_to_df(PO_macrofauna_surface_norm, addtax = T, addtot = T, addmaxrank = F, sorting = "NULL")
#write.table(PO_macrofauna_surface_df, "data/derived_data/phyloseq_objects/PO_macrofauna_surface_norm_2024-09-23.csv", row.names = FALSE)

propfun = function(x){
  xprop = (x / sum(x))
  return(xprop)
}
PO_macrofauna_surface_norm_prop <- transform_sample_counts(PO_macrofauna_surface_norm, fun = propfun)

#write_rds(PO_macrofauna_surface_norm_prop, "data/derived_data/phyloseq_objects/PO_macrofauna_surface_norm_prop_2024-09-23.rds")
#PO_macrofauna_surface_df <- phyloseq_to_df(PO_macrofauna_surface_norm_prop, addtax = T, addtot = T, addmaxrank = F, sorting = "NULL")
#write.csv(PO_macrofauna_surface_df[,-1], "data/derived_data/phyloseq_objects/PO_macrofauna_surface_norm_prop_2024-09-23.csv", row.names = FALSE)
```

### Pre-analyses

```{r}
tidy_data_PO_macrofauna_surface_before <- as.data.frame(t(tidy_data_PO_macrofauna_surface))
sample_names = rownames(tidy_data_PO_macrofauna_surface_before)
curve_before <- rarecurve(tidy_data_PO_macrofauna_surface_before, step = 5, label = F)
# save image to rarecurves_macrofauna_surface_raw_2024-09-23

tidy_data_PO_macrofauna_surface_stand <- otu_table(PO_macrofauna_surface_stand)
class(tidy_data_PO_macrofauna_surface_stand) <- "matrix"
tidy_data_PO_macrofauna_surface_stand <- t(tidy_data_PO_macrofauna_surface_stand)
curve_stand <- rarecurve(tidy_data_PO_macrofauna_surface_stand, step=5, label=F)
# save image to rarecurves_macrofauna_surface_stand_2024-09-23

tidy_data_PO_macrofauna_surface_norm <- otu_table(PO_macrofauna_surface_norm)
class(tidy_data_PO_macrofauna_surface_norm) <- "matrix"
tidy_data_PO_macrofauna_surface_norm <- t(tidy_data_PO_macrofauna_surface_norm)
curve_stand <- rarecurve(tidy_data_PO_macrofauna_surface_norm, step=5, label=F)
# save image to rarecurves_macrofauna_surface_norm_2024-09-23
```

```{r}
# Choose the version
PO <- PO_macrofauna_surface # raw
PO <- PO_macrofauna_surface_stand # this one used first
PO <- PO_macrofauna_surface_stand_prop
PO <- PO_macrofauna_surface_norm
PO <- PO_macrofauna_surface_norm_prop
PO <- PO_bioag
PO <- PO_auxil

# Filter date
PO <- subset_samples(PO, date == "17.04.2023")
```

```{r}
# Order levels
PO@sam_data$date <- factor(PO@sam_data$date, levels = order_date)
PO@sam_data$code_uh <- factor(PO@sam_data$code_uh, levels = order_uh)
PO@sam_data$spatial <- factor(PO@sam_data$spatial, levels = order_spatial)
```

```{r}
alpha_table <- estimate_richness(PO)
#write.table(alpha_table, "analyses/pre-analysis/alpha_macrofauna_surface_stand_2024-09-23.csv", sep = ",")

PO_alpha <- merge(PO@sam_data, alpha_table, by = 0)
quickanova <- aov(Observed ~ spatial, PO_alpha)
summary(quickanova)
quickanova <- aov(Shannon ~ spatial, PO_alpha)
summary(quickanova)
quickanova <- aov(InvSimpson ~ spatial, PO_alpha)
summary(quickanova)
```

```{r}
observed <- plot_richness(PO, x="code_uh", shape="block", color="habitat", measures=c("Observed")) +
  geom_boxplot(shape="block") +
  geom_point(position = "jitter") +
  theme_bw() + frame_simple + scale_x_discrete(limits = order_uh) +
  theme(axis.text.x=element_text(angle = -90, hjust = 0)) +
  labs(title="Alpha diversity of macrofauna_surface", subtitle="Taxonomy table", x ="land use_habitat", y="Observed richness") + facet_wrap(date ~ .)
observed$layers <- observed$layers[-1]

shannon <- plot_richness(PO, x="code_uh", shape="block", color="habitat", measures=c("Shannon")) +
  geom_boxplot(shape="block") +
  geom_point(position = "jitter") +
  theme_bw() + frame_simple + scale_x_discrete(limits = order_uh) +
  theme(axis.text.x=element_text(angle = -90, hjust = 0)) +
  labs(title="Alpha diversity of macrofauna_surface", subtitle="Taxonomy table", x ="land use_habitat", y="Shannon index") + facet_wrap(date ~ .)
shannon$layers <- shannon$layers[-1]

simpson <- plot_richness(PO, x="code_uh", shape="block", color="habitat", measures=c("InvSimpson")) +
  geom_boxplot(shape="block") +
  geom_point(position = "jitter") +
  theme_bw() + frame_simple + scale_x_discrete(limits = order_uh) +
  theme(axis.text.x=element_text(angle = -90, hjust = 0)) +
  labs(title="Alpha diversity of macrofauna_surface", subtitle="Taxonomy table", x ="land use_habitat", y="Simpson inverse index") + facet_wrap(date ~ .)
simpson$layers <- simpson$layers[-1]

alpha_graph <- plot_grid(observed, shannon, simpson, ncol=1)
alpha_graph

#ggsave("analyses/pre-analysis/alpha_macrofauna_surface_stand_uh_2024-09-23.png", width=5, height=12)
```

```{r}
PO_ord <- ordinate(PO, "PCoA", distance = "bray")
ord <- plot_ordination(PO, PO_ord, color = "code_uh", shape = "block") + theme_bw() + labs(title="Beta diversity of macrofauna_surface", subtitle="Taxonomy table")
ord

#ggsave("analyses/pre-analysis/beta_macrofauna_surface_plot_ordination_stand_uhb_2024-09-23.png", width=5, height=4)
```

```{r}
#ord_explore(PO)

PO %>%
 tax_transform(rank = "unique", trans = "identity") %>%
 dist_calc(dist = "bray") %>%
 ord_calc(
  method = "PCoA"
 ) %>% 
 ord_plot(
  axes = c(1, 2),
  colour = "date", fill = "date",
  shape = "habitat", alpha = 0.5,
  size = 2
 ) + 
 scale_shape_girafe_filled() +
 ggplot2::stat_ellipse(
  ggplot2::aes(colour = date)
 )

#ggsave("analyses/pre-analysis/beta_macrofauna_surface_microViz_stand_hd_2024-09-23.png", width=5, height=4, bg = "white")
```

```{r}
# Filter before to continue here

observed <- plot_richness(PO, x="habitat", shape="block", color="habitat", measures=c("Observed")) +
  geom_boxplot(shape="block") +
  geom_point(position = "jitter") +
  theme_bw() + frame_simple +
  labs(title="Alpha diversity of macrofauna_surface", subtitle="Taxonomy table", x =" ", y="Observed richness") + facet_wrap(spatial ~ .)
observed$layers <- observed$layers[-1]

shannon <- plot_richness(PO, x="habitat", shape="block", color="habitat", measures=c("Shannon")) +
  geom_boxplot(shape="block") +
  geom_point(position = "jitter") +
  theme_bw() + frame_simple +
  labs(title="Alpha diversity of macrofauna_surface", subtitle="Taxonomy table", x =" ", y="Shannon index") + facet_wrap(spatial ~ .)
shannon$layers <- shannon$layers[-1]

simpson <- plot_richness(PO, x="habitat", shape="block", color="habitat", measures=c("InvSimpson")) +
  geom_boxplot(shape="block") +
  geom_point(position = "jitter") +
  theme_bw() + frame_simple +
  labs(title="Alpha diversity of macrofauna_surface", subtitle="Taxonomy table", x =" ", y="Simpson inverse index") + facet_wrap(spatial ~ .)
simpson$layers <- simpson$layers[-1]

alpha_graph <- plot_grid(observed, shannon, simpson, ncol=1)
alpha_graph

#ggsave("analyses/pre-analysis/alpha_macrofauna_surface_stand_filterdate_spatial_2024-09-23.png", width=4, height=10)
#ggsave("analyses/pre-analysis/alpha_macrofauna_surface_stand_filterdate_spatial_bioag_2024-09-23.png", width=4, height=10)
#ggsave("analyses/pre-analysis/alpha_macrofauna_surface_stand_filterdate_spatial_auxil_2024-09-23.png", width=4, height=10)
```

```{r}
PO_ord <- ordinate(PO, "PCoA", distance = "bray")
ord <- plot_ordination(PO, PO_ord, color = "spatial", shape = "block") + theme_bw() + labs(title="Beta diversity of macrofauna_surface", subtitle="Taxonomy table")
ord

#ggsave("analyses/pre-analysis/beta_macrofauna_surface_plot_ordination_stand_filter_date_spatial_2024-09-23.png", width=5, height=4)
```

```{r}
#ord_explore(PO)

PO %>%
 tax_transform(rank = "unique", trans = "identity") %>%
 dist_calc(dist = "bray") %>%
 ord_calc(
  method = "PCoA"
 ) %>% 
 ord_plot(
  axes = c(1, 2),
  colour = "spatial", fill = "spatial",
  shape = "habitat", alpha = 0.5,
  size = 2
 ) + 
 scale_shape_girafe_filled() +
 ggplot2::stat_ellipse(
  ggplot2::aes(colour = spatial)
 )

#ggsave("analyses/pre-analysis/beta_macrofauna_surface_microViz_stand_filterdate_spatial_2024-09-23.png", width=5, height=4, bg = "white")
```

```{r}
adonis_pq(PO, "spatial", na_remove = TRUE)
adonis_pq(PO, "spatial*habitat", na_remove = TRUE)
adonis_pq(PO, "land_use*habitat", na_remove = TRUE)
adonis_pq(PO, "code_uh", na_remove = TRUE)
adonis_pq(PO, "code_uh*block", na_remove = TRUE)
adonis_pq(PO, "code_uh*replicate", na_remove = TRUE)
```

```{r}
#topOTU = names(sort(taxa_sums(PO), TRUE)[1:100])
#PO_top = prune_taxa(topOTU, PO)
PO_top <- PO # if no pruning
PO_top_uh = merge_samples(PO_top, "code_uh")
PO_top_uh_order = tax_glom(PO_top_uh, "order")
#abund_table <- psmelt(PO_top_uh_order)
#abund_table$order <- as.character(abund_table$order)
#abund_table$order[abund_table$abundance < 0.01] <- "abundunce < 1%" 

order_tot <- plot_bar(PO_top_uh_order, fill="order") + geom_col(stat="identity", position="stack", color = "black") + frame_simple + scale_x_discrete(limits = order_uh) + labs(title="Abundance of macrofauna_surface at order level ", subtitle="Total abundance", x ="land use_habitat") + scale_y_continuous(labels = scales::comma) + guides(fill=guide_legend(nrow=12)) + fill_manual_order

PO_top_uh_order_relabun = transform_sample_counts(PO_top_uh_order, function(x) x/sum(x) * 100)

order_rel <- plot_bar(PO_top_uh_order_relabun, fill="order") + geom_col(stat="identity", position="stack", color = "black") + frame_simple + scale_x_discrete(limits = order_uh) + labs(title="Abundance of macrofauna_surface at order level ", subtitle="Relative abundance", x ="land use_habitat") + scale_y_continuous(labels = scales::comma) + guides(fill=guide_legend(nrow=12)) + fill_manual_order

abund_order <- plot_grid(order_tot, order_rel, ncol=1)
abund_order

#ggsave("analyses/pre-analysis/abund_order_macrofauna_surface_stand_filterdate_uh_2024-09-23.png", width=6.5, height=9)
```

```{r}
#topOTU = names(sort(taxa_sums(PO), TRUE)[1:100])
#PO_top = prune_taxa(topOTU, PO)
PO_top <- PO # if no pruning
PO_top_spatial = merge_samples(PO_top, "spatial")
PO_top_spatial_order = tax_glom(PO_top_spatial, "order")
#abund_table <- psmelt(PO_top_spatial_order)
#abund_table$order <- as.character(abund_table$order)
#abund_table$order[abund_table$abundance < 0.01] <- "abundunce < 1%" 

order_tot <- plot_bar(PO_top_spatial_order, fill="order") + geom_col(stat="identity", position="stack", color = "black") + frame_simple + labs(title="Abundance of macrofauna_surface at order level ", subtitle="Total abundance", x ="Spatial heterogeneity") + scale_y_continuous(labels = scales::comma) + guides(fill=guide_legend(nrow=12)) + fill_manual_order + scale_x_discrete(limits = order_spatial)

PO_top_spatial_order_relabun = transform_sample_counts(PO_top_spatial_order, function(x) x/sum(x) * 100)

order_rel <- plot_bar(PO_top_spatial_order_relabun, fill="order") + geom_col(stat="identity", position="stack", color = "black") + frame_simple + labs(title="Abundance of macrofauna_surface at order level ", subtitle="Relative abundance", x ="Spatial heterogeneity") + scale_y_continuous(labels = scales::comma) + guides(fill=guide_legend(nrow=12)) + fill_manual_order + scale_x_discrete(limits = order_spatial)

abund_order <- plot_grid(order_tot, order_rel, ncol=1)
abund_order

#ggsave("analyses/pre-analysis/abund_order_macrofauna_surface_stand_filterdate_spatial_2024-09-23.png", width=5.5, height=9)
```

```{r}
# Filter functions
PO_bioag <- subset_taxa(PO, grepl("her|pla", abr_all_guild_level_AND))
PO_bioag_obs <- subset_taxa(PO_bioag, interaction == "observed")

PO_auxil <- subset_taxa(PO, grepl("pre|par", abr_all_guild_level_AND))
PO_auxil_obs <- subset_taxa(PO_auxil, interaction == "observed")
```

```{r}
PO_top <- PO_bioag
PO_top_spatial = merge_samples(PO_top, "spatial")
PO_top_spatial_family = tax_glom(PO_top_spatial, "family")

family_tot_bioag <- plot_bar(PO_top_spatial_family, fill="family") + geom_col(stat="identity", position="stack", color = "black") + frame_simple + labs(title="Abundance of macrofauna_surface at family level ", subtitle="Total abundance of potential bioagressors", x ="Spatial heterogeneity") + scale_y_continuous(labels = scales::comma)  + guides(fill=guide_legend(ncol=3))
family_tot_bioag

PO_top <- PO_auxil
PO_top_spatial = merge_samples(PO_top, "spatial")
PO_top_spatial_family = tax_glom(PO_top_spatial, "family")

family_tot_auxil <- plot_bar(PO_top_spatial_family, fill="family") + geom_col(stat="identity", position="stack", color = "black") + frame_simple + labs(title="Abundance of macrofauna_surface at family level ", subtitle="Total abundance of potential auxiliaries", x ="Spatial heterogeneity") + scale_y_continuous(labels = scales::comma)  + guides(fill=guide_legend(ncol=3))
family_tot_auxil

abund_tot_family <- plot_grid(family_tot_bioag, family_tot_auxil, ncol=1)
abund_tot_family

#ggsave("analyses/pre-analysis/abund_tot_family_macrofauna_surface_stand_filterdate_spatial_bioagANDauxil_2024-09-23.png", width=6, height=9)
```

```{r}
PO_top <- PO_bioag
PO_top_spatial = merge_samples(PO_top, "spatial")
PO_top_spatial_order = tax_glom(PO_top_spatial, "order")

order_tot_bioag <- plot_bar(PO_top_spatial_order, fill="order") + geom_col(stat="identity", position="stack", color = "black") + frame_simple + labs(title="Abundance of macrofauna_surface at order level ", subtitle="Total abundance of potential bioagressors", x ="Spatial heterogeneity") + scale_y_continuous(labels = scales::comma) + guides(fill=guide_legend(nrow=10)) + scale_x_discrete(limits = order_spatial) + fill_manual_order
order_tot_bioag

PO_top <- PO_auxil
PO_top_spatial = merge_samples(PO_top, "spatial")
PO_top_spatial_order = tax_glom(PO_top_spatial, "order")

order_tot_auxil <- plot_bar(PO_top_spatial_order, fill="order") + geom_col(stat="identity", position="stack", color = "black") + frame_simple + labs(title="Abundance of macrofauna_surface at order level ", subtitle="Total abundance of potential auxiliaries", x ="Spatial heterogeneity") + scale_y_continuous(labels = scales::comma) + guides(fill=guide_legend(nrow=10)) + scale_x_discrete(limits = order_spatial) + fill_manual_order
order_tot_auxil

abund_tot_order <- plot_grid(order_tot_bioag, order_tot_auxil, ncol=1)
abund_tot_order

#ggsave("analyses/pre-analysis/abund_tot_order_macrofauna_surface_stand_filterdate_spatial_bioagANDauxil_2024-09-23.png", width=5, height=9)
```

```{r}
PO_top <- PO_bioag_obs
PO_top_spatial = merge_samples(PO_top, "spatial")
PO_top_spatial_family = tax_glom(PO_top_spatial, "family")

family_tot_bioag <- plot_bar(PO_top_spatial_family, fill="family") + geom_col(stat="identity", position="stack", color = "black") + frame_simple + labs(title="Abundance of macrofauna_surface at family level ", subtitle="Total abundance of potential bioagressors with observed interactions", x ="Spatial heterogeneity") + scale_y_continuous(labels = scales::comma)  + guides(fill=guide_legend(ncol=1)) + scale_x_discrete(limits = order_spatial) + fill_manual_family
family_tot_bioag

PO_top <- PO_auxil_obs
PO_top_spatial = merge_samples(PO_top, "spatial")
PO_top_spatial_family = tax_glom(PO_top_spatial, "family")

family_tot_auxil <- plot_bar(PO_top_spatial_family, fill="family") + geom_col(stat="identity", position="stack", color = "black") + frame_simple + labs(title="Abundance of macrofauna_surface at family level ", subtitle="Total abundance of potential auxiliaries with observed interactions", x ="Spatial heterogeneity") + scale_y_continuous(labels = scales::comma)  + guides(fill=guide_legend(ncol=1)) + scale_x_discrete(limits = order_spatial) + fill_manual_family
family_tot_auxil

abund_tot_family <- plot_grid(family_tot_bioag, family_tot_auxil, ncol=1)
abund_tot_family

#ggsave("analyses/pre-analysis/abund_tot_family_macrofauna_surface_stand_filterdate_spatial_bioagANDauxil_obs_2024-09-23.png", width=5, height=9)
```

```{r}
list_PO_bioag_obs <- PO_bioag_obs@tax_table
#write.csv(list_PO_bioag_obs, "analyses/pre-analysis/list_macrofauna_surface_bioag_obs_2024-09-23.csv", row.names = T)

list_PO_auxil_obs <- PO_auxil_obs@tax_table
#write.csv(list_PO_auxil_obs, "analyses/pre-analysis/list_macrofauna_surface_auxil_obs_2024-09-23.csv", row.names = T)
```

```{r}
V_spatial_bioag <- ggvenn_pq(PO_bioag, fact = "spatial", min_nb_seq = 0, taxonomic_rank = "family", add_nb_samples = F, set_size = 3) + labs(title = "Number of potential bioagressors at family level") # split_by = "date" # label = "count"
V_spatial_bioag

#save manually to ggvenn_macrofauna_surface_spatial_bioag_family_2024-09-23

V_spatial_auxil <- ggvenn_pq(PO_auxil, fact = "spatial", min_nb_seq = 0, taxonomic_rank = "family", add_nb_samples = F, set_size = 3) + labs(title = "Number of potential auxiliaries at family level") # split_by = "date" # label = "count"
V_spatial_auxil

#save manually to ggvenn_macrofauna_surface_spatial_auxil_family_2024-09-23
```

## macrofauna_aerial

### Version raw

```{r}
# otu_table = tidy_data
## select the community
tidy_data_PO_macrofauna_aerial <- tidy_data_PO %>% filter(community %in% c("macrofauna_aerial"))

## remove useless columns
tidy_data_PO_macrofauna_aerial <- tidy_data_PO_macrofauna_aerial[,c(1,2,13)]

## transform table to taxa_are_rows = TRUE
tidy_data_PO_macrofauna_aerial <- pivot_wider(tidy_data_PO_macrofauna_aerial, names_from = code_uhbrd, values_from = abundance)

## replace NA by 0
tidy_data_PO_macrofauna_aerial[is.na(tidy_data_PO_macrofauna_aerial)] <- 0

## use "finest_name" as rownames
tidy_data_PO_macrofauna_aerial <- tidy_data_PO_macrofauna_aerial %>% tibble::column_to_rownames("finest_name")

## transform tbl_df to matrix table
tidy_data_PO_macrofauna_aerial <- as.matrix(tidy_data_PO_macrofauna_aerial)

# tax_table = tax_table # + fun_table
## select the community
tax_table_PO_macrofauna_aerial <- tax_table %>% filter(community %in% c("macrofauna_aerial"))

## remove useless columns
tax_table_PO_macrofauna_aerial <- tax_table_PO_macrofauna_aerial[,-c(9)]

########## Add fun_table here ########## 
tax_table_PO_macrofauna_aerial_fun <- left_join(tax_table_PO_macrofauna_aerial, fun_table_PO, by = c("family" = "matchName"))

tax_table_PO_macrofauna_aerial_fun[is.na(tax_table_PO_macrofauna_aerial_fun)] <- "unassigned"

tax_table_PO_macrofauna_aerial_fun_inter <- tax_table_PO_macrofauna_aerial_fun %>%
  rowwise() %>%
  mutate(interaction = if_else(any(c_across(1:8) %in% fun_table_inter_PO$matchName), "observed", "not observed")) %>%
  ungroup()

tax_table_PO_macrofauna_aerial <- tax_table_PO_macrofauna_aerial_fun_inter
########## END ########## 

## use "finest_name" as rownames
tax_table_PO_macrofauna_aerial <- tax_table_PO_macrofauna_aerial %>% tibble::column_to_rownames("finest_name")

## transform tbl_df to matrix table
tax_table_PO_macrofauna_aerial <- as.matrix(tax_table_PO_macrofauna_aerial)

# sam_data = metadata
## select the community
metadata_PO_macrofauna_aerial <- metadata_PO %>% filter(community %in% c("macrofauna_aerial"))

## choose the rownames
rownames(metadata_PO_macrofauna_aerial) <- metadata_PO_macrofauna_aerial[,11] # code_uhbrd

# create phyloseq object
tidy_data_PO_macrofauna_aerial = otu_table(tidy_data_PO_macrofauna_aerial, taxa_are_rows = TRUE)

tax_table_PO_macrofauna_aerial = tax_table(tax_table_PO_macrofauna_aerial)

metadata_PO_macrofauna_aerial = sample_data(metadata_PO_macrofauna_aerial)
  
PO_macrofauna_aerial <- phyloseq(tidy_data_PO_macrofauna_aerial, tax_table_PO_macrofauna_aerial, metadata_PO_macrofauna_aerial)

# check PO
PO_macrofauna_aerial
sample_names(PO_macrofauna_aerial)
rank_names(PO_macrofauna_aerial)
sample_variables(PO_macrofauna_aerial)

#write_rds(PO_macrofauna_aerial, "data/derived_data/phyloseq_objects/PO_macrofauna_aerial_raw_2024-09-23.rds")
#PO_macrofauna_aerial_df <- phyloseq_to_df(PO_macrofauna_aerial, addtax = T, addtot = T, addmaxrank = F, sorting = "NULL")
#write.csv(PO_macrofauna_aerial_df, "data/derived_data/phyloseq_objects/PO_macrofauna_aerial_raw_2024-09-23.csv", row.names = FALSE)
```

### Version stand

```{r}
# standardize abundances to the median sequencing depth
total = median(sample_sums(PO_macrofauna_aerial))
standf = function(x, t=total) round(t*(x/sum(x)))
PO_macrofauna_aerial_stand = transform_sample_counts(PO_macrofauna_aerial, standf)

# check PO
PO_macrofauna_aerial_stand

#write_rds(PO_macrofauna_aerial_stand, "data/derived_data/phyloseq_objects/PO_macrofauna_aerial_stand_2024-09-23.rds")
#PO_macrofauna_aerial_df <- phyloseq_to_df(PO_macrofauna_aerial_stand, addtax = T, addtot = T, addmaxrank = F, sorting = "NULL")
#write.table(PO_macrofauna_aerial_df, "data/derived_data/phyloseq_objects/PO_macrofauna_aerial_stand_2024-09-23.csv", row.names = FALSE)

propfun = function(x){
  xprop = (x / sum(x))
  return(xprop)
}
PO_macrofauna_aerial_stand_prop <- transform_sample_counts(PO_macrofauna_aerial_stand, fun = propfun)

#write_rds(PO_macrofauna_aerial_stand_prop, "data/derived_data/phyloseq_objects/PO_macrofauna_aerial_stand_prop_2024-09-23.rds")
#PO_macrofauna_aerial_df <- phyloseq_to_df(PO_macrofauna_aerial_stand_prop, addtax = T, addtot = T, addmaxrank = F, sorting = "NULL")
#write.table(PO_macrofauna_aerial_df, "data/derived_data/phyloseq_objects/PO_macrofauna_aerial_stand_prop_2024-09-23.csv", row.names = FALSE)
```

### Version norm

```{r}
# normalize
PO_macrofauna_aerial_norm <- normalize_prop_pq(PO_macrofauna_aerial, base_log = 2, digits = 0)

# check PO
PO_macrofauna_aerial_norm

#write_rds(PO_macrofauna_aerial_norm, "data/derived_data/phyloseq_objects/PO_macrofauna_aerial_norm_2024-09-23.rds")
#PO_macrofauna_aerial_df <- phyloseq_to_df(PO_macrofauna_aerial_norm, addtax = T, addtot = T, addmaxrank = F, sorting = "NULL")
#write.table(PO_macrofauna_aerial_df, "data/derived_data/phyloseq_objects/PO_macrofauna_aerial_norm_2024-09-23.csv", row.names = FALSE)

propfun = function(x){
  xprop = (x / sum(x))
  return(xprop)
}
PO_macrofauna_aerial_norm_prop <- transform_sample_counts(PO_macrofauna_aerial_norm, fun = propfun)

#write_rds(PO_macrofauna_aerial_norm_prop, "data/derived_data/phyloseq_objects/PO_macrofauna_aerial_norm_prop_2024-09-23.rds")
#PO_macrofauna_aerial_df <- phyloseq_to_df(PO_macrofauna_aerial_norm_prop, addtax = T, addtot = T, addmaxrank = F, sorting = "NULL")
#write.csv(PO_macrofauna_aerial_df[,-1], "data/derived_data/phyloseq_objects/PO_macrofauna_aerial_norm_prop_2024-09-23.csv", row.names = FALSE)
```

### Pre-analyses

```{r}
tidy_data_PO_macrofauna_aerial_before <- as.data.frame(t(tidy_data_PO_macrofauna_aerial))
sample_names = rownames(tidy_data_PO_macrofauna_aerial_before)
curve_before <- rarecurve(tidy_data_PO_macrofauna_aerial_before, step = 5, label = F)
# save image to rarecurves_macrofauna_aerial_raw_2024-09-23

tidy_data_PO_macrofauna_aerial_stand <- otu_table(PO_macrofauna_aerial_stand)
class(tidy_data_PO_macrofauna_aerial_stand) <- "matrix"
tidy_data_PO_macrofauna_aerial_stand <- t(tidy_data_PO_macrofauna_aerial_stand)
curve_stand <- rarecurve(tidy_data_PO_macrofauna_aerial_stand, step=5, label=F)
# save image to rarecurves_macrofauna_aerial_stand_2024-09-23

tidy_data_PO_macrofauna_aerial_norm <- otu_table(PO_macrofauna_aerial_norm)
class(tidy_data_PO_macrofauna_aerial_norm) <- "matrix"
tidy_data_PO_macrofauna_aerial_norm <- t(tidy_data_PO_macrofauna_aerial_norm)
curve_stand <- rarecurve(tidy_data_PO_macrofauna_aerial_norm, step=5, label=F)
# save image to rarecurves_macrofauna_aerial_norm_2024-09-23
```

```{r}
# Choose the version
PO <- PO_macrofauna_aerial # raw
PO <- PO_macrofauna_aerial_stand # this one used first
PO <- PO_macrofauna_aerial_stand_prop
PO <- PO_macrofauna_aerial_norm
PO <- PO_macrofauna_aerial_norm_prop
PO <- PO_bioag
PO <- PO_auxil

# Filter date
PO <- subset_samples(PO, date == "17.04.2023")
```

```{r}
# Order levels
PO@sam_data$date <- factor(PO@sam_data$date, levels = order_date)
PO@sam_data$code_uh <- factor(PO@sam_data$code_uh, levels = order_uh)
PO@sam_data$spatial <- factor(PO@sam_data$spatial, levels = order_spatial)
```

```{r}
alpha_table <- estimate_richness(PO)
#write.table(alpha_table, "analyses/pre-analysis/alpha_macrofauna_aerial_stand_2024-09-23.csv", sep = ",")

PO_alpha <- merge(PO@sam_data, alpha_table, by = 0)
quickanova <- aov(Observed ~ spatial, PO_alpha)
summary(quickanova)
quickanova <- aov(Shannon ~ spatial, PO_alpha)
summary(quickanova)
quickanova <- aov(InvSimpson ~ spatial, PO_alpha)
summary(quickanova)
```

```{r}
observed <- plot_richness(PO, x="code_uh", shape="block", color="habitat", measures=c("Observed")) +
  geom_boxplot(shape="block") +
  geom_point(position = "jitter") +
  theme_bw() + frame_simple + scale_x_discrete(limits = order_uh) +
  theme(axis.text.x=element_text(angle = -90, hjust = 0)) +
  labs(title="Alpha diversity of macrofauna_aerial", subtitle="Taxonomy table", x ="land use_habitat", y="Observed richness") + facet_wrap(date ~ .)
observed$layers <- observed$layers[-1]

shannon <- plot_richness(PO, x="code_uh", shape="block", color="habitat", measures=c("Shannon")) +
  geom_boxplot(shape="block") +
  geom_point(position = "jitter") +
  theme_bw() + frame_simple + scale_x_discrete(limits = order_uh) +
  theme(axis.text.x=element_text(angle = -90, hjust = 0)) +
  labs(title="Alpha diversity of macrofauna_aerial", subtitle="Taxonomy table", x ="land use_habitat", y="Shannon index") + facet_wrap(date ~ .)
shannon$layers <- shannon$layers[-1]

simpson <- plot_richness(PO, x="code_uh", shape="block", color="habitat", measures=c("InvSimpson")) +
  geom_boxplot(shape="block") +
  geom_point(position = "jitter") +
  theme_bw() + frame_simple + scale_x_discrete(limits = order_uh) +
  theme(axis.text.x=element_text(angle = -90, hjust = 0)) +
  labs(title="Alpha diversity of macrofauna_aerial", subtitle="Taxonomy table", x ="land use_habitat", y="Simpson inverse index") + facet_wrap(date ~ .)
simpson$layers <- simpson$layers[-1]

alpha_graph <- plot_grid(observed, shannon, simpson, ncol=1)
alpha_graph

#ggsave("analyses/pre-analysis/alpha_macrofauna_aerial_stand_uh_2024-09-19.png", width=5, height=12)
```

```{r}
PO_ord <- ordinate(PO, "PCoA", distance = "bray")
ord <- plot_ordination(PO, PO_ord, color = "code_uh", shape = "block") + theme_bw() + labs(title="Beta diversity of macrofauna_aerial", subtitle="Taxonomy table")
ord

#ggsave("analyses/pre-analysis/beta_macrofauna_aerial_plot_ordination_stand_uhb_2024-09-23.png", width=5, height=4)
```

```{r}
#ord_explore(PO)

PO %>%
 tax_transform(rank = "unique", trans = "identity") %>%
 dist_calc(dist = "bray") %>%
 ord_calc(
  method = "PCoA"
 ) %>% 
 ord_plot(
  axes = c(1, 2),
  colour = "date", fill = "date",
  shape = "habitat", alpha = 0.5,
  size = 2
 ) + 
 scale_shape_girafe_filled() +
 ggplot2::stat_ellipse(
  ggplot2::aes(colour = date)
 )

#ggsave("analyses/pre-analysis/beta_macrofauna_aerial_microViz_stand_hd_2024-09-23.png", width=5, height=4, bg = "white")
```

```{r}
# Filter before to continue here

observed <- plot_richness(PO, x="habitat", shape="block", color="habitat", measures=c("Observed")) +
  geom_boxplot(shape="block") +
  geom_point(position = "jitter") +
  theme_bw() + frame_simple +
  labs(title="Alpha diversity of macrofauna_aerial", subtitle="Taxonomy table", x =" ", y="Observed richness") + facet_wrap(spatial ~ .)
observed$layers <- observed$layers[-1]

shannon <- plot_richness(PO, x="habitat", shape="block", color="habitat", measures=c("Shannon")) +
  geom_boxplot(shape="block") +
  geom_point(position = "jitter") +
  theme_bw() + frame_simple +
  labs(title="Alpha diversity of macrofauna_aerial", subtitle="Taxonomy table", x =" ", y="Shannon index") + facet_wrap(spatial ~ .)
shannon$layers <- shannon$layers[-1]

simpson <- plot_richness(PO, x="habitat", shape="block", color="habitat", measures=c("InvSimpson")) +
  geom_boxplot(shape="block") +
  geom_point(position = "jitter") +
  theme_bw() + frame_simple +
  labs(title="Alpha diversity of macrofauna_aerial", subtitle="Taxonomy table", x =" ", y="Simpson inverse index") + facet_wrap(spatial ~ .)
simpson$layers <- simpson$layers[-1]

alpha_graph <- plot_grid(observed, shannon, simpson, ncol=1)
alpha_graph

#ggsave("analyses/pre-analysis/alpha_macrofauna_aerial_stand_filterdate_spatial_2024-09-23.png", width=4, height=10)
#ggsave("analyses/pre-analysis/alpha_macrofauna_aerial_stand_filterdate_spatial_bioag_2024-09-23.png", width=4, height=10)
ggsave("analyses/pre-analysis/alpha_macrofauna_aerial_stand_filterdate_spatial_auxil_2024-09-23.png", width=4, height=10)
```

```{r}
PO_ord <- ordinate(PO, "PCoA", distance = "bray")
ord <- plot_ordination(PO, PO_ord, color = "spatial", shape = "block") + theme_bw() + labs(title="Beta diversity of macrofauna_aerial", subtitle="Taxonomy table")
ord

#ggsave("analyses/pre-analysis/beta_macrofauna_aerial_plot_ordination_stand_filter_date_spatial_2024-09-23.png", width=5, height=4)
```

```{r}
#ord_explore(PO)

PO %>%
 tax_transform(rank = "unique", trans = "identity") %>%
 dist_calc(dist = "bray") %>%
 ord_calc(
  method = "PCoA"
 ) %>% 
 ord_plot(
  axes = c(1, 2),
  colour = "spatial", fill = "spatial",
  shape = "habitat", alpha = 0.5,
  size = 2
 ) + 
 scale_shape_girafe_filled() +
 ggplot2::stat_ellipse(
  ggplot2::aes(colour = spatial)
 )

#ggsave("analyses/pre-analysis/beta_macrofauna_aerial_microViz_stand_filterdate_spatial_2024-09-23.png", width=5, height=4, bg = "white")
```

```{r}
adonis_pq(PO, "spatial", na_remove = TRUE)
adonis_pq(PO, "spatial*habitat", na_remove = TRUE)
adonis_pq(PO, "land_use*habitat", na_remove = TRUE)
adonis_pq(PO, "code_uh", na_remove = TRUE)
adonis_pq(PO, "code_uh*block", na_remove = TRUE)
adonis_pq(PO, "code_uh*replicate", na_remove = TRUE)
```

```{r}
#topOTU = names(sort(taxa_sums(PO), TRUE)[1:100])
#PO_top = prune_taxa(topOTU, PO)
PO_top <- PO # if no pruning
PO_top_uh = merge_samples(PO_top, "code_uh")
PO_top_uh_order = tax_glom(PO_top_uh, "order")
#abund_table <- psmelt(PO_top_uh_order)
#abund_table$order <- as.character(abund_table$order)
#abund_table$order[abund_table$abundance < 0.01] <- "abundunce < 1%" 

order_tot <- plot_bar(PO_top_uh_order, fill="order") + geom_col(stat="identity", position="stack", color = "black") + frame_simple + scale_x_discrete(limits = order_uh) + labs(title="Abundance of macrofauna_aerial at order level ", subtitle="Total abundance", x ="land use_habitat") + scale_y_continuous(labels = scales::comma) + guides(fill=guide_legend(nrow=10)) + fill_manual_order

PO_top_uh_order_relabun = transform_sample_counts(PO_top_uh_order, function(x) x/sum(x) * 100)

order_rel <- plot_bar(PO_top_uh_order_relabun, fill="order") + geom_col(stat="identity", position="stack", color = "black") + frame_simple + scale_x_discrete(limits = order_uh) + labs(title="Abundance of macrofauna_aerial at order level ", subtitle="Relative abundance", x ="land use_habitat") + scale_y_continuous(labels = scales::comma) + guides(fill=guide_legend(nrow=10)) + fill_manual_order

abund_order <- plot_grid(order_tot, order_rel, ncol=1)
abund_order

#ggsave("analyses/pre-analysis/abund_order_macrofauna_aerial_stand_filterdate_uh_2024-09-23.png", width=6, height=9)
```

```{r}
#topOTU = names(sort(taxa_sums(PO), TRUE)[1:100])
#PO_top = prune_taxa(topOTU, PO)
PO_top <- PO # if no pruning
PO_top_spatial = merge_samples(PO_top, "spatial")
PO_top_spatial_order = tax_glom(PO_top_spatial, "order")
#abund_table <- psmelt(PO_top_spatial_order)
#abund_table$order <- as.character(abund_table$order)
#abund_table$order[abund_table$abundance < 0.01] <- "abundunce < 1%" 

order_tot <- plot_bar(PO_top_spatial_order, fill="order") + geom_col(stat="identity", position="stack", color = "black") + frame_simple + labs(title="Abundance of macrofauna_aerial at order level ", subtitle="Total abundance", x ="Spatial heterogeneity") + scale_y_continuous(labels = scales::comma) + guides(fill=guide_legend(nrow=10)) + fill_manual_order + scale_x_discrete(limits = order_spatial)

PO_top_spatial_order_relabun = transform_sample_counts(PO_top_spatial_order, function(x) x/sum(x) * 100)

order_rel <- plot_bar(PO_top_spatial_order_relabun, fill="order") + geom_col(stat="identity", position="stack", color = "black") + frame_simple + labs(title="Abundance of macrofauna_aerial at order level ", subtitle="Relative abundance", x ="Spatial heterogeneity") + scale_y_continuous(labels = scales::comma) + guides(fill=guide_legend(nrow=10)) + fill_manual_order + scale_x_discrete(limits = order_spatial)

abund_order <- plot_grid(order_tot, order_rel, ncol=1)
abund_order

#ggsave("analyses/pre-analysis/abund_order_macrofauna_aerial_stand_filterdate_spatial_2024-09-23.png", width=4.5, height=9)
```

```{r}
# Filter functions
PO_bioag <- subset_taxa(PO, grepl("her|pla", abr_all_guild_level_AND))
PO_bioag_obs <- subset_taxa(PO_bioag, interaction == "observed")

PO_auxil <- subset_taxa(PO, grepl("pre|par", abr_all_guild_level_AND))
PO_auxil_obs <- subset_taxa(PO_auxil, interaction == "observed")
```

```{r}
PO_top <- PO_bioag
PO_top_spatial = merge_samples(PO_top, "spatial")
PO_top_spatial_family = tax_glom(PO_top_spatial, "family")

family_tot_bioag <- plot_bar(PO_top_spatial_family, fill="family") + geom_col(stat="identity", position="stack", color = "black") + frame_simple + labs(title="Abundance of macrofauna_aerial at family level ", subtitle="Total abundance of potential bioagressors", x ="Spatial heterogeneity") + scale_y_continuous(labels = scales::comma)  + guides(fill=guide_legend(ncol=3))
family_tot_bioag

PO_top <- PO_auxil
PO_top_spatial = merge_samples(PO_top, "spatial")
PO_top_spatial_family = tax_glom(PO_top_spatial, "family")

family_tot_auxil <- plot_bar(PO_top_spatial_family, fill="family") + geom_col(stat="identity", position="stack", color = "black") + frame_simple + labs(title="Abundance of macrofauna_aerial at family level ", subtitle="Total abundance of potential auxiliaries", x ="Spatial heterogeneity") + scale_y_continuous(labels = scales::comma)  + guides(fill=guide_legend(ncol=3))
family_tot_auxil

abund_tot_family <- plot_grid(family_tot_bioag, family_tot_auxil, ncol=1)
abund_tot_family

#ggsave("analyses/pre-analysis/abund_tot_family_macrofauna_aerial_stand_filterdate_spatial_bioagANDauxil_2024-09-23.png", width=6, height=9)
```

```{r}
PO_top <- PO_bioag
PO_top_spatial = merge_samples(PO_top, "spatial")
PO_top_spatial_order = tax_glom(PO_top_spatial, "order")

order_tot_bioag <- plot_bar(PO_top_spatial_order, fill="order") + geom_col(stat="identity", position="stack", color = "black") + frame_simple + labs(title="Abundance of macrofauna_aerial at order level ", subtitle="Total abundance of potential bioagressors", x ="Spatial heterogeneity") + scale_y_continuous(labels = scales::comma) + guides(fill=guide_legend(nrow=10)) + scale_x_discrete(limits = order_spatial) + fill_manual_order
order_tot_bioag

PO_top <- PO_auxil
PO_top_spatial = merge_samples(PO_top, "spatial")
PO_top_spatial_order = tax_glom(PO_top_spatial, "order")

order_tot_auxil <- plot_bar(PO_top_spatial_order, fill="order") + geom_col(stat="identity", position="stack", color = "black") + frame_simple + labs(title="Abundance of macrofauna_aerial at order level ", subtitle="Total abundance of potential auxiliaries", x ="Spatial heterogeneity") + scale_y_continuous(labels = scales::comma) + guides(fill=guide_legend(nrow=10)) + scale_x_discrete(limits = order_spatial) + fill_manual_order
order_tot_auxil

abund_tot_order <- plot_grid(order_tot_bioag, order_tot_auxil, ncol=1)
abund_tot_order

#ggsave("analyses/pre-analysis/abund_tot_order_macrofauna_aerial_stand_filterdate_spatial_bioagANDauxil_2024-09-23.png", width=5, height=9)
```

```{r}
PO_top <- PO_bioag_obs
PO_top_spatial = merge_samples(PO_top, "spatial")
PO_top_spatial_family = tax_glom(PO_top_spatial, "family")

family_tot_bioag <- plot_bar(PO_top_spatial_family, fill="family") + geom_col(stat="identity", position="stack", color = "black") + frame_simple + labs(title="Abundance of macrofauna_aerial at family level ", subtitle="Total abundance of potential bioagressors with observed interactions", x ="Spatial heterogeneity") + scale_y_continuous(labels = scales::comma)  + guides(fill=guide_legend(ncol=1)) + scale_x_discrete(limits = order_spatial) + fill_manual_family
family_tot_bioag

PO_top <- PO_auxil_obs
PO_top_spatial = merge_samples(PO_top, "spatial")
PO_top_spatial_family = tax_glom(PO_top_spatial, "family")

family_tot_auxil <- plot_bar(PO_top_spatial_family, fill="family") + geom_col(stat="identity", position="stack", color = "black") + frame_simple + labs(title="Abundance of macrofauna_aerial at family level ", subtitle="Total abundance of potential auxiliaries with observed interactions", x ="Spatial heterogeneity") + scale_y_continuous(labels = scales::comma)  + guides(fill=guide_legend(ncol=1)) + scale_x_discrete(limits = order_spatial) + fill_manual_family
family_tot_auxil

abund_tot_family <- plot_grid(family_tot_bioag, family_tot_auxil, ncol=1)
abund_tot_family

#ggsave("analyses/pre-analysis/abund_tot_family_macrofauna_aerial_stand_filterdate_spatial_bioagANDauxil_obs_2024-09-23.png", width=5, height=9)
```

```{r}
list_PO_bioag_obs <- PO_bioag_obs@tax_table
#write.csv(list_PO_bioag_obs, "analyses/pre-analysis/list_macrofauna_aerial_bioag_obs_2024-09-23.csv", row.names = T)

list_PO_auxil_obs <- PO_auxil_obs@tax_table
#write.csv(list_PO_auxil_obs, "analyses/pre-analysis/list_macrofauna_aerial_auxil_obs_2024-09-23.csv", row.names = T)
```

```{r}
V_spatial_bioag <- ggvenn_pq(PO_bioag, fact = "spatial", min_nb_seq = 0, taxonomic_rank = "family", add_nb_samples = F, set_size = 3) + labs(title = "Number of potential bioagressors at family level") # split_by = "date" # label = "count"
V_spatial_bioag

#save manually to ggvenn_macrofauna_aerial_spatial_bioag_family_2024-09-23

V_spatial_auxil <- ggvenn_pq(PO_auxil, fact = "spatial", min_nb_seq = 0, taxonomic_rank = "family", add_nb_samples = F, set_size = 3) + labs(title = "Number of potential auxiliaries at family level") # split_by = "date" # label = "count"
V_spatial_auxil

#save manually to ggvenn_macrofauna_aerial_spatial_auxil_family_2024-09-23
```

## macrofauna_foliar

### Version raw

```{r}
# otu_table = tidy_data
## select the community
tidy_data_PO_macrofauna_foliar <- tidy_data_PO %>% filter(community %in% c("macrofauna_foliar"))

## remove useless columns
tidy_data_PO_macrofauna_foliar <- tidy_data_PO_macrofauna_foliar[,c(1,2,13)]

## transform table to taxa_are_rows = TRUE
tidy_data_PO_macrofauna_foliar <- pivot_wider(tidy_data_PO_macrofauna_foliar, names_from = code_uhbrd, values_from = abundance)

## replace NA by 0
tidy_data_PO_macrofauna_foliar[is.na(tidy_data_PO_macrofauna_foliar)] <- 0

## use "finest_name" as rownames
tidy_data_PO_macrofauna_foliar <- tidy_data_PO_macrofauna_foliar %>% tibble::column_to_rownames("finest_name")

## transform tbl_df to matrix table
tidy_data_PO_macrofauna_foliar <- as.matrix(tidy_data_PO_macrofauna_foliar)

# tax_table = tax_table # + fun_table
## select the community
tax_table_PO_macrofauna_foliar <- tax_table %>% filter(community %in% c("macrofauna_foliar"))

## remove useless columns
tax_table_PO_macrofauna_foliar <- tax_table_PO_macrofauna_foliar[,-c(9)]

########## Add fun_table here ########## 
tax_table_PO_macrofauna_foliar_fun <- left_join(tax_table_PO_macrofauna_foliar, fun_table_PO, by = c("family" = "matchName"))

tax_table_PO_macrofauna_foliar_fun[is.na(tax_table_PO_macrofauna_foliar_fun)] <- "unassigned"

tax_table_PO_macrofauna_foliar_fun_inter <- tax_table_PO_macrofauna_foliar_fun %>%
  rowwise() %>%
  mutate(interaction = if_else(any(c_across(1:8) %in% fun_table_inter_PO$matchName), "observed", "not observed")) %>%
  ungroup()

tax_table_PO_macrofauna_foliar <- tax_table_PO_macrofauna_foliar_fun_inter
########## END ########## 

## use "finest_name" as rownames
tax_table_PO_macrofauna_foliar <- tax_table_PO_macrofauna_foliar %>% tibble::column_to_rownames("finest_name")

## transform tbl_df to matrix table
tax_table_PO_macrofauna_foliar <- as.matrix(tax_table_PO_macrofauna_foliar)

# sam_data = metadata
## select the community
metadata_PO_macrofauna_foliar <- metadata_PO %>% filter(community %in% c("macrofauna_foliar"))

## choose the rownames
rownames(metadata_PO_macrofauna_foliar) <- metadata_PO_macrofauna_foliar[,11] # code_uhbrd

# create phyloseq object
tidy_data_PO_macrofauna_foliar = otu_table(tidy_data_PO_macrofauna_foliar, taxa_are_rows = TRUE)

tax_table_PO_macrofauna_foliar = tax_table(tax_table_PO_macrofauna_foliar)

metadata_PO_macrofauna_foliar = sample_data(metadata_PO_macrofauna_foliar)
  
PO_macrofauna_foliar <- phyloseq(tidy_data_PO_macrofauna_foliar, tax_table_PO_macrofauna_foliar, metadata_PO_macrofauna_foliar)

# check PO
PO_macrofauna_foliar
sample_names(PO_macrofauna_foliar)
rank_names(PO_macrofauna_foliar)
sample_variables(PO_macrofauna_foliar)

#write_rds(PO_macrofauna_foliar, "data/derived_data/phyloseq_objects/PO_macrofauna_foliar_raw_2024-09-23.rds")
#PO_macrofauna_foliar_df <- phyloseq_to_df(PO_macrofauna_foliar, addtax = T, addtot = T, addmaxrank = F, sorting = "NULL")
#write.csv(PO_macrofauna_foliar_df, "data/derived_data/phyloseq_objects/PO_macrofauna_foliar_raw_2024-09-23.csv", row.names = FALSE)
```

### Version stand

```{r}
# standardize abundances to the median sequencing depth
total = median(sample_sums(PO_macrofauna_foliar))
standf = function(x, t=total) round(t*(x/sum(x)))
PO_macrofauna_foliar_stand = transform_sample_counts(PO_macrofauna_foliar, standf)

# check PO
PO_macrofauna_foliar_stand

#write_rds(PO_macrofauna_foliar_stand, "data/derived_data/phyloseq_objects/PO_macrofauna_foliar_stand_2024-09-23.rds")
#PO_macrofauna_foliar_df <- phyloseq_to_df(PO_macrofauna_foliar_stand, addtax = T, addtot = T, addmaxrank = F, sorting = "NULL")
#write.table(PO_macrofauna_foliar_df, "data/derived_data/phyloseq_objects/PO_macrofauna_foliar_stand_2024-09-23.csv", row.names = FALSE)

propfun = function(x){
  xprop = (x / sum(x))
  return(xprop)
}
PO_macrofauna_foliar_stand_prop <- transform_sample_counts(PO_macrofauna_foliar_stand, fun = propfun)

#write_rds(PO_macrofauna_foliar_stand_prop, "data/derived_data/phyloseq_objects/PO_macrofauna_foliar_stand_prop_2024-09-23.rds")
#PO_macrofauna_foliar_df <- phyloseq_to_df(PO_macrofauna_foliar_stand_prop, addtax = T, addtot = T, addmaxrank = F, sorting = "NULL")
#write.table(PO_macrofauna_foliar_df, "data/derived_data/phyloseq_objects/PO_macrofauna_foliar_stand_prop_2024-09-23.csv", row.names = FALSE)
```

### Version norm

```{r}
# normalize
PO_macrofauna_foliar_norm <- normalize_prop_pq(PO_macrofauna_foliar, base_log = 2, digits = 0)

# check PO
PO_macrofauna_foliar_norm

#write_rds(PO_macrofauna_foliar_norm, "data/derived_data/phyloseq_objects/PO_macrofauna_foliar_norm_2024-09-23.rds")
#PO_macrofauna_foliar_df <- phyloseq_to_df(PO_macrofauna_foliar_norm, addtax = T, addtot = T, addmaxrank = F, sorting = "NULL")
#write.table(PO_macrofauna_foliar_df, "data/derived_data/phyloseq_objects/PO_macrofauna_foliar_norm_2024-09-23.csv", row.names = FALSE)

propfun = function(x){
  xprop = (x / sum(x))
  return(xprop)
}
PO_macrofauna_foliar_norm_prop <- transform_sample_counts(PO_macrofauna_foliar_norm, fun = propfun)

#write_rds(PO_macrofauna_foliar_norm_prop, "data/derived_data/phyloseq_objects/PO_macrofauna_foliar_norm_prop_2024-09-23.rds")
#PO_macrofauna_foliar_df <- phyloseq_to_df(PO_macrofauna_foliar_norm_prop, addtax = T, addtot = T, addmaxrank = F, sorting = "NULL")
#write.csv(PO_macrofauna_foliar_df[,-1], "data/derived_data/phyloseq_objects/PO_macrofauna_foliar_norm_prop_2024-09-23.csv", row.names = FALSE)
```

### Pre-analyses

```{r}
tidy_data_PO_macrofauna_foliar_before <- as.data.frame(t(tidy_data_PO_macrofauna_foliar))
sample_names = rownames(tidy_data_PO_macrofauna_foliar_before)
curve_before <- rarecurve(tidy_data_PO_macrofauna_foliar_before, step = 5, label = F)
# save image to rarecurves_macrofauna_foliar_raw_2024-09-23

tidy_data_PO_macrofauna_foliar_stand <- otu_table(PO_macrofauna_foliar_stand)
class(tidy_data_PO_macrofauna_foliar_stand) <- "matrix"
tidy_data_PO_macrofauna_foliar_stand <- t(tidy_data_PO_macrofauna_foliar_stand)
curve_stand <- rarecurve(tidy_data_PO_macrofauna_foliar_stand, step=5, label=F)
# save image to rarecurves_macrofauna_foliar_stand_2024-09-23

tidy_data_PO_macrofauna_foliar_norm <- otu_table(PO_macrofauna_foliar_norm)
class(tidy_data_PO_macrofauna_foliar_norm) <- "matrix"
tidy_data_PO_macrofauna_foliar_norm <- t(tidy_data_PO_macrofauna_foliar_norm)
curve_stand <- rarecurve(tidy_data_PO_macrofauna_foliar_norm, step=5, label=F)
# save image to rarecurves_macrofauna_foliar_norm_2024-09-23
```

```{r}
# Choose the version
PO <- PO_macrofauna_foliar # raw
PO <- PO_macrofauna_foliar_stand # this one used first
PO <- PO_macrofauna_foliar_stand_prop
PO <- PO_macrofauna_foliar_norm
PO <- PO_macrofauna_foliar_norm_prop
PO <- PO_bioag
PO <- PO_auxil

# Filter date
PO <- subset_samples(PO, date == "17.04.2023")

# Filter method
PO <- subset_samples(PO, method == "inflorescence")
PO <- subset_samples(PO, method == "twig")
```

```{r}
# Order levels
PO@sam_data$date <- factor(PO@sam_data$date, levels = order_date)
PO@sam_data$code_uh <- factor(PO@sam_data$code_uh, levels = order_uh)
PO@sam_data$spatial <- factor(PO@sam_data$spatial, levels = order_spatial)
```

```{r}
alpha_table <- estimate_richness(PO)
#write.table(alpha_table, "analyses/pre-analysis/alpha_macrofauna_foliar_stand_inflo_2024-09-23.csv", sep = ",")
#write.table(alpha_table, "analyses/pre-analysis/alpha_macrofauna_foliar_stand_twig_2024-09-23.csv", sep = ",")

PO_alpha <- merge(PO@sam_data, alpha_table, by = 0)
quickanova <- aov(Observed ~ spatial, PO_alpha)
summary(quickanova)
quickanova <- aov(Shannon ~ spatial, PO_alpha)
summary(quickanova)
quickanova <- aov(InvSimpson ~ spatial, PO_alpha)
summary(quickanova)
```

```{r}
observed <- plot_richness(PO, x="code_uh", shape="block", color="habitat", measures=c("Observed")) +
  geom_boxplot(shape="block") +
  geom_point(position = "jitter") +
  theme_bw() + frame_simple +
  theme(axis.text.x=element_text(angle = -90, hjust = 0)) +
  labs(title="Alpha diversity of macrofauna_foliar", subtitle="Taxonomy table", x ="land use_habitat", y="Observed richness") + facet_wrap(date ~ method)
observed$layers <- observed$layers[-1]

shannon <- plot_richness(PO, x="code_uh", shape="block", color="habitat", measures=c("Shannon")) +
  geom_boxplot(shape="block") +
  geom_point(position = "jitter") +
  theme_bw() + frame_simple +
  theme(axis.text.x=element_text(angle = -90, hjust = 0)) +
  labs(title="Alpha diversity of macrofauna_foliar", subtitle="Taxonomy table", x ="land use_habitat", y="Shannon index") + facet_wrap(date ~ method)
shannon$layers <- shannon$layers[-1]

simpson <- plot_richness(PO, x="code_uh", shape="block", color="habitat", measures=c("InvSimpson")) +
  geom_boxplot(shape="block") +
  geom_point(position = "jitter") +
  theme_bw() + frame_simple +
  theme(axis.text.x=element_text(angle = -90, hjust = 0)) +
  labs(title="Alpha diversity of macrofauna_foliar", subtitle="Taxonomy table", x ="land use_habitat", y="Simpson inverse index") + facet_wrap(date ~ method)
simpson$layers <- simpson$layers[-1]

alpha_graph <- plot_grid(observed, shannon, simpson, ncol=1)
alpha_graph

#ggsave("analyses/pre-analysis/alpha_macrofauna_foliar_stand_uh_T_2024-09-23.png", width=4, height=12)
```

```{r}
PO_ord <- ordinate(PO, "PCoA", distance = "bray")
ord <- plot_ordination(PO, PO_ord, color = "code_uh", shape = "block") + theme_bw() + labs(title="Beta diversity of macrofauna_foliar", subtitle="Taxonomy table")
ord

#ggsave("analyses/pre-analysis/beta_macrofauna_foliar_plot_ordination_stand_inflo_uhb_2024-09-23.png", width=5, height=4)
#ggsave("analyses/pre-analysis/beta_macrofauna_foliar_plot_ordination_stand_twig_uhb_2024-09-23.png", width=5, height=4)
```

```{r}
#ord_explore(PO)

PO %>%
 tax_transform(rank = "unique", trans = "identity") %>%
 dist_calc(dist = "bray") %>%
 ord_calc(
  method = "PCoA"
 ) %>% 
 ord_plot(
  axes = c(1, 2),
  colour = "date", fill = "date",
  shape = "habitat", alpha = 0.5,
  size = 2
 ) + 
 scale_shape_girafe_filled() +
 ggplot2::stat_ellipse(
  ggplot2::aes(colour = date)
 )

#ggsave("analyses/pre-analysis/beta_macrofauna_foliar_microViz_stand_inflo_hd_2024-09-23.png", width=5, height=4, bg = "white")
#ggsave("analyses/pre-analysis/beta_macrofauna_foliar_microViz_stand_twig_hd_2024-09-23.png", width=5, height=4, bg = "white")
```

```{r}
# Filter before to continue here

observed <- plot_richness(PO, x="habitat", shape="block", color="habitat", measures=c("Observed")) +
  geom_boxplot(shape="block") +
  geom_point(position = "jitter") +
  theme_bw() + frame_simple +
  labs(title="Alpha diversity of macrofauna_foliar", subtitle="Taxonomy table", x =" ", y="Observed richness") + facet_wrap(spatial ~ .)
observed$layers <- observed$layers[-1]

shannon <- plot_richness(PO, x="habitat", shape="block", color="habitat", measures=c("Shannon")) +
  geom_boxplot(shape="block") +
  geom_point(position = "jitter") +
  theme_bw() + frame_simple +
  labs(title="Alpha diversity of macrofauna_foliar", subtitle="Taxonomy table", x =" ", y="Shannon index") + facet_wrap(spatial ~ .)
shannon$layers <- shannon$layers[-1]

simpson <- plot_richness(PO, x="habitat", shape="block", color="habitat", measures=c("InvSimpson")) +
  geom_boxplot(shape="block") +
  geom_point(position = "jitter") +
  theme_bw() + frame_simple +
  labs(title="Alpha diversity of macrofauna_foliar", subtitle="Taxonomy table", x =" ", y="Simpson inverse index") + facet_wrap(spatial ~ .)
simpson$layers <- simpson$layers[-1]

alpha_graph <- plot_grid(observed, shannon, simpson, ncol=1)
alpha_graph

#ggsave("analyses/pre-analysis/alpha_macrofauna_foliar_stand_filterdate_spatial_2024-09-23.png", width=3.5, height=10)
#ggsave("analyses/pre-analysis/alpha_macrofauna_foliar_stand_filterdate_spatial_bioag_2024-09-23.png", width=3.5, height=10)
#ggsave("analyses/pre-analysis/alpha_macrofauna_foliar_stand_filterdate_spatial_auxil_2024-09-23.png", width=3.5, height=10)
```

```{r}
PO_ord <- ordinate(PO, "PCoA", distance = "bray")
ord <- plot_ordination(PO, PO_ord, color = "spatial", shape = "block") + theme_bw() + labs(title="Beta diversity of macrofauna_foliar", subtitle="Taxonomy table")
ord

#ggsave("analyses/pre-analysis/beta_macrofauna_foliar_plot_ordination_stand_filterdate_spatial_2024-09-23.png", width=5, height=4)
```

```{r}
#ord_explore(PO)

PO %>%
 tax_transform(rank = "unique", trans = "identity") %>%
 dist_calc(dist = "bray") %>%
 ord_calc(
  method = "PCoA"
 ) %>% 
 ord_plot(
  axes = c(1, 2),
  colour = "spatial", fill = "spatial",
  shape = "habitat", alpha = 0.5,
  size = 2
 ) + 
 scale_shape_girafe_filled() +
 ggplot2::stat_ellipse(
  ggplot2::aes(colour = spatial)
 )

#ggsave("analyses/pre-analysis/beta_macrofauna_foliar_microViz_stand_filterdate_spatial_2024-09-23.png", width=5, height=4, bg = "white")
```

```{r}
adonis_pq(PO, "spatial", na_remove = TRUE)
adonis_pq(PO, "code_uh", na_remove = TRUE)
adonis_pq(PO, "code_uh*block", na_remove = TRUE)
adonis_pq(PO, "code_uh*replicate", na_remove = TRUE)
```

```{r}
#topOTU = names(sort(taxa_sums(PO), TRUE)[1:100])
#PO_top = prune_taxa(topOTU, PO)
PO_top <- PO # if no pruning
PO_top_uh = merge_samples(PO_top, "code_uh")
PO_top_uh_order = tax_glom(PO_top_uh, "order")
#abund_table <- psmelt(PO_top_uh_order)
#abund_table$order <- as.character(abund_table$order)
#abund_table$order[abund_table$abundance < 0.01] <- "abundunce < 1%" 

order_tot <- plot_bar(PO_top_uh_order, fill="order") + geom_col(stat="identity", position="stack", color = "black") + frame_simple + labs(title="Abundance of macrofauna_foliar at order level ", subtitle="Total abundance", x ="land use_habitat") + scale_y_continuous(labels = scales::comma) + guides(fill=guide_legend(nrow=10)) + fill_manual_order

PO_top_uh_order_relabun = transform_sample_counts(PO_top_uh_order, function(x) x/sum(x) * 100)

order_rel <- plot_bar(PO_top_uh_order_relabun, fill="order") + geom_col(stat="identity", position="stack", color = "black") + frame_simple + labs(title="Abundance of macrofauna_foliar at order level ", subtitle="Relative abundance", x ="land use_habitat") + scale_y_continuous(labels = scales::comma) + guides(fill=guide_legend(nrow=10)) + fill_manual_order

abund_order <- plot_grid(order_tot, order_rel, ncol=1)
abund_order

#ggsave("analyses/pre-analysis/abund_order_macrofauna_foliar_stand_filterdate_uh_2024-09-23.png", width=4.5, height=9)
```

```{r}
#topOTU = names(sort(taxa_sums(PO), TRUE)[1:100])
#PO_top = prune_taxa(topOTU, PO)
PO_top <- PO # if no pruning
PO_top_spatial = merge_samples(PO_top, "spatial")
PO_top_spatial_order = tax_glom(PO_top_spatial, "order")
#abund_table <- psmelt(PO_top_spatial_order)
#abund_table$order <- as.character(abund_table$order)
#abund_table$order[abund_table$abundance < 0.01] <- "abundunce < 1%" 

order_tot <- plot_bar(PO_top_spatial_order, fill="order") + geom_col(stat="identity", position="stack", color = "black") + frame_simple + labs(title="Abundance of macrofauna_foliar at order level ", subtitle="Total abundance", x ="Spatial heterogeneity") + scale_y_continuous(labels = scales::comma) + guides(fill=guide_legend(nrow=10)) + fill_manual_order + scale_x_discrete(limits = order_spatial)

PO_top_spatial_order_relabun = transform_sample_counts(PO_top_spatial_order, function(x) x/sum(x) * 100)

order_rel <- plot_bar(PO_top_spatial_order_relabun, fill="order") + geom_col(stat="identity", position="stack", color = "black") + frame_simple + labs(title="Abundance of macrofauna_foliar at order level ", subtitle="Relative abundance", x ="Spatial heterogeneity") + scale_y_continuous(labels = scales::comma) + guides(fill=guide_legend(nrow=10)) + fill_manual_order + scale_x_discrete(limits = order_spatial)

abund_order <- plot_grid(order_tot, order_rel, ncol=1)
abund_order

#ggsave("analyses/pre-analysis/abund_order_macrofauna_foliar_stand_filterdate_spatial_2024-09-23.png", width=4.5, height=9)
```

```{r}
# Filter functions
PO_bioag <- subset_taxa(PO, grepl("her|pla", abr_all_guild_level_AND))
#PO_bioag_obs <- subset_taxa(PO_bioag, interaction == "observed") # no observed interaction

PO_auxil <- subset_taxa(PO, grepl("pre|par", abr_all_guild_level_AND))
PO_auxil_obs <- subset_taxa(PO_auxil, interaction == "observed")
```

```{r}
PO_top <- PO_bioag
PO_top_spatial = merge_samples(PO_top, "spatial")
PO_top_spatial_family = tax_glom(PO_top_spatial, "family")

family_tot_bioag <- plot_bar(PO_top_spatial_family, fill="family") + geom_col(stat="identity", position="stack", color = "black") + frame_simple + labs(title="Abundance of macrofauna_foliar at family level ", subtitle="Total abundance of potential bioagressors", x ="Spatial heterogeneity") + scale_y_continuous(labels = scales::comma)  + guides(fill=guide_legend(ncol=3))
family_tot_bioag

PO_top <- PO_auxil
PO_top_spatial = merge_samples(PO_top, "spatial")
PO_top_spatial_family = tax_glom(PO_top_spatial, "family")

family_tot_auxil <- plot_bar(PO_top_spatial_family, fill="family") + geom_col(stat="identity", position="stack", color = "black") + frame_simple + labs(title="Abundance of macrofauna_foliar at family level ", subtitle="Total abundance of potential auxiliaries", x ="Spatial heterogeneity") + scale_y_continuous(labels = scales::comma)  + guides(fill=guide_legend(ncol=3))
family_tot_auxil

abund_tot_family <- plot_grid(family_tot_bioag, family_tot_auxil, ncol=1)
abund_tot_family

#ggsave("analyses/pre-analysis/abund_tot_family_macrofauna_foliar_stand_filterdate_spatial_bioagANDauxil_2024-09-23.png", width=6, height=9)
```

```{r}
PO_top <- PO_bioag
PO_top_spatial = merge_samples(PO_top, "spatial")
PO_top_spatial_order = tax_glom(PO_top_spatial, "order")

order_tot_bioag <- plot_bar(PO_top_spatial_order, fill="order") + geom_col(stat="identity", position="stack", color = "black") + frame_simple + labs(title="Abundance of macrofauna_foliar at order level ", subtitle="Total abundance of potential bioagressors", x ="Spatial heterogeneity") + scale_y_continuous(labels = scales::comma) + guides(fill=guide_legend(nrow=10)) + scale_x_discrete(limits = order_spatial) + fill_manual_order
order_tot_bioag

PO_top <- PO_auxil
PO_top_spatial = merge_samples(PO_top, "spatial")
PO_top_spatial_order = tax_glom(PO_top_spatial, "order")

order_tot_auxil <- plot_bar(PO_top_spatial_order, fill="order") + geom_col(stat="identity", position="stack", color = "black") + frame_simple + labs(title="Abundance of macrofauna_foliar at order level ", subtitle="Total abundance of potential auxiliaries", x ="Spatial heterogeneity") + scale_y_continuous(labels = scales::comma) + guides(fill=guide_legend(nrow=10)) + scale_x_discrete(limits = order_spatial) + fill_manual_order
order_tot_auxil

abund_tot_order <- plot_grid(order_tot_bioag, order_tot_auxil, ncol=1)
abund_tot_order

#ggsave("analyses/pre-analysis/abund_tot_order_macrofauna_foliar_stand_filterdate_spatial_bioagANDauxil_2024-09-23.png", width=5, height=9)
```

```{r}
#PO_top <- PO_bioag_obs
#PO_top_spatial = merge_samples(PO_top, "spatial")
#PO_top_spatial_family = tax_glom(PO_top_spatial, "family")

#family_tot_bioag <- plot_bar(PO_top_spatial_family, fill="family") + geom_col(stat="identity", position="stack", color = "black") + frame_simple + labs(title="Abundance of macrofauna_foliar at family level ", subtitle="Total abundance of potential bioagressors with observed interactions", x ="Spatial heterogeneity") + scale_y_continuous(labels = scales::comma)  + guides(fill=guide_legend(ncol=1)) + scale_x_discrete(limits = order_spatial) + fill_manual_family
#family_tot_bioag

PO_top <- PO_auxil_obs
PO_top_spatial = merge_samples(PO_top, "spatial")
PO_top_spatial_family = tax_glom(PO_top_spatial, "family")

family_tot_auxil <- plot_bar(PO_top_spatial_family, fill="family") + geom_col(stat="identity", position="stack", color = "black") + frame_simple + labs(title="Abundance of macrofauna_foliar at family level ", subtitle="Total abundance of potential auxiliaries with observed interactions", x ="Spatial heterogeneity") + scale_y_continuous(labels = scales::comma)  + guides(fill=guide_legend(ncol=1)) + scale_x_discrete(limits = order_spatial) + fill_manual_family
family_tot_auxil

#abund_tot_family <- plot_grid(family_tot_bioag, family_tot_auxil, ncol=1)
#abund_tot_family

#ggsave("analyses/pre-analysis/abund_tot_family_macrofauna_foliar_stand_filterdate_spatial_bioagANDauxil_obs_2024-09-23.png", width=5, height=3)
```

```{r}
#list_PO_bioag_obs <- PO_bioag_obs@tax_table
#write.csv(list_PO_bioag_obs, "analyses/pre-analysis/list_macrofauna_foliar_bioag_obs_2024-09-23.csv", row.names = T)

list_PO_auxil_obs <- PO_auxil_obs@tax_table
#write.csv(list_PO_auxil_obs, "analyses/pre-analysis/list_macrofauna_foliar_auxil_obs_2024-09-23.csv", row.names = T)
```

```{r}
V_spatial_bioag <- ggvenn_pq(PO_bioag, fact = "spatial", min_nb_seq = 0, taxonomic_rank = "family", add_nb_samples = F, set_size = 3) + labs(title = "Number of potential bioagressors at family level") # split_by = "date" # label = "count"
V_spatial_bioag

#save manually to ggvenn_macrofauna_foliar_spatial_bioag_family_2024-09-23

V_spatial_auxil <- ggvenn_pq(PO_auxil, fact = "spatial", min_nb_seq = 0, taxonomic_rank = "family", add_nb_samples = F, set_size = 3) + labs(title = "Number of potential auxiliaries at family level") # split_by = "date" # label = "count"
V_spatial_auxil

#save manually to ggvenn_macrofauna_foliar_spatial_auxil_family_2024-09-23
```

## nematodes

### Version raw

```{r}
# otu_table = tidy_data
## select the community
tidy_data_PO_nematodes <- tidy_data_PO %>% filter(community %in% c("nematodes"))

## remove useless columns
tidy_data_PO_nematodes <- tidy_data_PO_nematodes[,c(1,2,13)]

## transform table to taxa_are_rows = TRUE
tidy_data_PO_nematodes <- pivot_wider(tidy_data_PO_nematodes, names_from = code_uhbrd, values_from = abundance)

## replace NA by 0
tidy_data_PO_nematodes[is.na(tidy_data_PO_nematodes)] <- 0

## use "finest_name" as rownames
tidy_data_PO_nematodes <- tidy_data_PO_nematodes %>% tibble::column_to_rownames("finest_name")

## transform tbl_df to matrix table
tidy_data_PO_nematodes <- as.matrix(tidy_data_PO_nematodes)

# tax_table = tax_table # + fun_table
## select the community
tax_table_PO_nematodes <- tax_table %>% filter(community %in% c("nematodes"))

## remove useless columns
tax_table_PO_nematodes <- tax_table_PO_nematodes[,-c(9)]

########## Add fun_table here ########## 
tax_table_PO_nematodes_fun <- left_join(tax_table_PO_nematodes, fun_table_PO, by = c("genus" = "matchName"))

tax_table_PO_nematodes_fun[is.na(tax_table_PO_nematodes_fun)] <- "unassigned"

tax_table_PO_nematodes_fun_inter <- tax_table_PO_nematodes_fun %>%
  rowwise() %>%
  mutate(interaction = if_else(any(c_across(1:8) %in% fun_table_inter_PO$matchName), "observed", "not observed")) %>%
  ungroup()

tax_table_PO_nematodes <- tax_table_PO_nematodes_fun_inter
########## END ########## 

## use "finest_name" as rownames
tax_table_PO_nematodes <- tax_table_PO_nematodes %>% tibble::column_to_rownames("finest_name")

## transform tbl_df to matrix table
tax_table_PO_nematodes <- as.matrix(tax_table_PO_nematodes)

# sam_data = metadata
## select the community
metadata_PO_nematodes <- metadata_PO %>% filter(community %in% c("nematodes"))

## choose the rownames
rownames(metadata_PO_nematodes) <- metadata_PO_nematodes[,11] # code_uhbrd

# create phyloseq object
tidy_data_PO_nematodes = otu_table(tidy_data_PO_nematodes, taxa_are_rows = TRUE)

tax_table_PO_nematodes = tax_table(tax_table_PO_nematodes)

metadata_PO_nematodes = sample_data(metadata_PO_nematodes)
  
PO_nematodes <- phyloseq(tidy_data_PO_nematodes, tax_table_PO_nematodes, metadata_PO_nematodes)

# check PO
PO_nematodes
sample_names(PO_nematodes)
rank_names(PO_nematodes)
sample_variables(PO_nematodes)

#write_rds(PO_nematodes, "data/derived_data/phyloseq_objects/PO_nematodes_raw_2024-09-23.rds")
#PO_nematodes_df <- phyloseq_to_df(PO_nematodes, addtax = T, addtot = T, addmaxrank = F, sorting = "NULL")
#write.csv(PO_nematodes_df, "data/derived_data/phyloseq_objects/PO_nematodes_raw_2024-09-23.csv", row.names = FALSE)
```

### Version stand

```{r}
# standardize abundances to the median sequencing depth
total = median(sample_sums(PO_nematodes))
standf = function(x, t=total) round(t*(x/sum(x)))
PO_nematodes_stand = transform_sample_counts(PO_nematodes, standf)

# check PO
PO_nematodes_stand

#write_rds(PO_nematodes_stand, "data/derived_data/phyloseq_objects/PO_nematodes_stand_2024-09-23.rds")
#PO_nematodes_df <- phyloseq_to_df(PO_nematodes_stand, addtax = T, addtot = T, addmaxrank = F, sorting = "NULL")
#write.table(PO_nematodes_df, "data/derived_data/phyloseq_objects/PO_nematodes_stand_2024-09-23.csv", row.names = FALSE)

propfun = function(x){
  xprop = (x / sum(x))
  return(xprop)
}
PO_nematodes_stand_prop <- transform_sample_counts(PO_nematodes_stand, fun = propfun)

#write_rds(PO_nematodes_stand_prop, "data/derived_data/phyloseq_objects/PO_nematodes_stand_prop_2024-09-23.rds")
#PO_nematodes_df <- phyloseq_to_df(PO_nematodes_stand_prop, addtax = T, addtot = T, addmaxrank = F, sorting = "NULL")
#write.table(PO_nematodes_df, "data/derived_data/phyloseq_objects/PO_nematodes_stand_prop_2024-09-23.csv", row.names = FALSE)
```

### Version norm

```{r}
# normalize
PO_nematodes_norm <- normalize_prop_pq(PO_nematodes, base_log = 2, digits = 0)

# check PO
PO_nematodes_norm

#write_rds(PO_nematodes_norm, "data/derived_data/phyloseq_objects/PO_nematodes_norm_2024-09-23.rds")
#PO_nematodes_df <- phyloseq_to_df(PO_nematodes_norm, addtax = T, addtot = T, addmaxrank = F, sorting = "NULL")
#write.table(PO_nematodes_df, "data/derived_data/phyloseq_objects/PO_nematodes_norm_2024-09-23.csv", row.names = FALSE)

propfun = function(x){
  xprop = (x / sum(x))
  return(xprop)
}
PO_nematodes_norm_prop <- transform_sample_counts(PO_nematodes_norm, fun = propfun)

#write_rds(PO_nematodes_norm_prop, "data/derived_data/phyloseq_objects/PO_nematodes_norm_prop_2024-09-23.rds")
#PO_nematodes_df <- phyloseq_to_df(PO_nematodes_norm_prop, addtax = T, addtot = T, addmaxrank = F, sorting = "NULL")
#write.csv(PO_nematodes_df[,-1], "data/derived_data/phyloseq_objects/PO_nematodes_norm_prop_2024-09-23.csv", row.names = FALSE)
```

### Pre-analyses

```{r}
tidy_data_PO_nematodes_before <- as.data.frame(t(tidy_data_PO_nematodes))
sample_names = rownames(tidy_data_PO_nematodes_before)
curve_before <- rarecurve(tidy_data_PO_nematodes_before, step = 5, label = F)
# save image to rarecurves_nematodes_raw_2024-09-23

tidy_data_PO_nematodes_stand <- otu_table(PO_nematodes_stand)
class(tidy_data_PO_nematodes_stand) <- "matrix"
tidy_data_PO_nematodes_stand <- t(tidy_data_PO_nematodes_stand)
curve_stand <- rarecurve(tidy_data_PO_nematodes_stand, step=5, label=F)
# save image to rarecurves_nematodes_stand_2024-09-23

tidy_data_PO_nematodes_norm <- otu_table(PO_nematodes_norm)
class(tidy_data_PO_nematodes_norm) <- "matrix"
tidy_data_PO_nematodes_norm <- t(tidy_data_PO_nematodes_norm)
curve_stand <- rarecurve(tidy_data_PO_nematodes_norm, step=5, label=F)
# save image to rarecurves_nematodes_norm_2024-09-23
```

```{r}
# Choose the version
PO <- PO_nematodes # raw
PO <- PO_nematodes_stand # this one used first
PO <- PO_nematodes_stand_prop
PO <- PO_nematodes_norm
PO <- PO_nematodes_norm_prop
PO <- PO_bioag
PO <- PO_auxil

# Filter date
PO <- subset_samples(PO, date == "17.04.2023")
```

```{r}
# Order levels
PO@sam_data$date <- factor(PO@sam_data$date, levels = order_date)
PO@sam_data$code_uh <- factor(PO@sam_data$code_uh, levels = order_uh)
PO@sam_data$spatial <- factor(PO@sam_data$spatial, levels = order_spatial)
```

```{r}
alpha_table <- estimate_richness(PO)
#write.table(alpha_table, "analyses/pre-analysis/alpha_nematodes_stand_2024-09-23.csv", sep = ",")

PO_alpha <- merge(PO@sam_data, alpha_table, by = 0)
quickanova <- aov(Observed ~ spatial, PO_alpha)
summary(quickanova)
quickanova <- aov(Shannon ~ spatial, PO_alpha)
summary(quickanova)
quickanova <- aov(InvSimpson ~ spatial, PO_alpha)
summary(quickanova)
```

```{r}
observed <- plot_richness(PO, x="code_uh", shape="block", color="habitat", measures=c("Observed")) +
  geom_boxplot(shape="block") +
  geom_point(position = "jitter") +
  theme_bw() + frame_simple + scale_x_discrete(limits = order_uh) +
  theme(axis.text.x=element_text(angle = -90, hjust = 0)) +
  labs(title="Alpha diversity of nematodes", subtitle="Taxonomy table", x ="land use_habitat", y="Observed richness") + facet_wrap(date ~ .)
observed$layers <- observed$layers[-1]

shannon <- plot_richness(PO, x="code_uh", shape="block", color="habitat", measures=c("Shannon")) +
  geom_boxplot(shape="block") +
  geom_point(position = "jitter") +
  theme_bw() + frame_simple + scale_x_discrete(limits = order_uh) +
  theme(axis.text.x=element_text(angle = -90, hjust = 0)) +
  labs(title="Alpha diversity of nematodes", subtitle="Taxonomy table", x ="land use_habitat", y="Shannon index") + facet_wrap(date ~ .)
shannon$layers <- shannon$layers[-1]

simpson <- plot_richness(PO, x="code_uh", shape="block", color="habitat", measures=c("InvSimpson")) +
  geom_boxplot(shape="block") +
  geom_point(position = "jitter") +
  theme_bw() + frame_simple + scale_x_discrete(limits = order_uh) +
  theme(axis.text.x=element_text(angle = -90, hjust = 0)) +
  labs(title="Alpha diversity of nematodes", subtitle="Taxonomy table", x ="land use_habitat", y="Simpson inverse index") + facet_wrap(date ~ .)
simpson$layers <- simpson$layers[-1]

alpha_graph <- plot_grid(observed, shannon, simpson, ncol=1)
alpha_graph

#ggsave("analyses/pre-analysis/alpha_nematodes_stand_uh_2024-09-23.png", width=5, height=12)
```

```{r}
PO_ord <- ordinate(PO, "PCoA", distance = "bray")
ord <- plot_ordination(PO, PO_ord, color = "code_uh", shape = "block") + theme_bw() + labs(title="Beta diversity of nematodes", subtitle="Taxonomy table")
ord

#ggsave("analyses/pre-analysis/beta_nematodes_plot_ordination_stand_uhb_2024-09-23.png", width=5, height=4)
```

```{r}
#ord_explore(PO)

PO %>%
 tax_transform(rank = "unique", trans = "identity") %>%
 dist_calc(dist = "bray") %>%
 ord_calc(
  method = "PCoA"
 ) %>% 
 ord_plot(
  axes = c(1, 2),
  colour = "date", fill = "date",
  shape = "habitat", alpha = 0.5,
  size = 2
 ) + 
 scale_shape_girafe_filled() +
 ggplot2::stat_ellipse(
  ggplot2::aes(colour = date)
 )

#ggsave("analyses/pre-analysis/beta_nematodes_microViz_stand_hd_2024-09-23.png", width=5, height=4, bg = "white")
```

```{r}
# Filter before to continue here

observed <- plot_richness(PO, x="habitat", shape="block", color="habitat", measures=c("Observed")) +
  geom_boxplot(shape="block") +
  geom_point(position = "jitter") +
  theme_bw() + frame_simple +
  labs(title="Alpha diversity of nematodes", subtitle="Taxonomy table", x =" ", y="Observed richness") + facet_wrap(spatial ~ .)
observed$layers <- observed$layers[-1]

shannon <- plot_richness(PO, x="habitat", shape="block", color="habitat", measures=c("Shannon")) +
  geom_boxplot(shape="block") +
  geom_point(position = "jitter") +
  theme_bw() + frame_simple +
  labs(title="Alpha diversity of nematodes", subtitle="Taxonomy table", x =" ", y="Shannon index") + facet_wrap(spatial ~ .)
shannon$layers <- shannon$layers[-1]

simpson <- plot_richness(PO, x="habitat", shape="block", color="habitat", measures=c("InvSimpson")) +
  geom_boxplot(shape="block") +
  geom_point(position = "jitter") +
  theme_bw() + frame_simple +
  labs(title="Alpha diversity of nematodes", subtitle="Taxonomy table", x =" ", y="Simpson inverse index") + facet_wrap(spatial ~ .)
simpson$layers <- simpson$layers[-1]

alpha_graph <- plot_grid(observed, shannon, simpson, ncol=1)
alpha_graph

#ggsave("analyses/pre-analysis/alpha_nematodes_stand_filterdate_spatial_2024-09-23.png", width=4, height=10)
#ggsave("analyses/pre-analysis/alpha_nematodes_stand_filterdate_spatial_bioag_2024-09-23.png", width=4, height=10)
#ggsave("analyses/pre-analysis/alpha_nematodes_stand_filterdate_spatial_auxil_2024-09-23.png", width=4, height=10)
```

```{r}
PO_ord <- ordinate(PO, "PCoA", distance = "bray")
ord <- plot_ordination(PO, PO_ord, color = "spatial", shape = "block") + theme_bw() + labs(title="Beta diversity of nematodes", subtitle="Taxonomy table")
ord

#ggsave("analyses/pre-analysis/beta_nematodes_plot_ordination_stand_filter_date_spatial_2024-09-23.png", width=5, height=4)
```

```{r}
#ord_explore(PO)

PO %>%
 tax_transform(rank = "unique", trans = "identity") %>%
 dist_calc(dist = "bray") %>%
 ord_calc(
  method = "PCoA"
 ) %>% 
 ord_plot(
  axes = c(1, 2),
  colour = "spatial", fill = "spatial",
  shape = "habitat", alpha = 0.5,
  size = 2
 ) + 
 scale_shape_girafe_filled() +
 ggplot2::stat_ellipse(
  ggplot2::aes(colour = spatial)
 )

#ggsave("analyses/pre-analysis/beta_nematodes_microViz_stand_filterdate_spatial_2024-09-23.png", width=5, height=4, bg = "white")
```

```{r}
adonis_pq(PO, "spatial", na_remove = TRUE)
adonis_pq(PO, "spatial*habitat", na_remove = TRUE)
adonis_pq(PO, "land_use*habitat", na_remove = TRUE)
adonis_pq(PO, "code_uh", na_remove = TRUE)
adonis_pq(PO, "code_uh*block", na_remove = TRUE)
#adonis_pq(PO, "code_uh*replicate", na_remove = TRUE)
```

```{r}
#topOTU = names(sort(taxa_sums(PO), TRUE)[1:100])
#PO_top = prune_taxa(topOTU, PO)
PO_top <- PO # if no pruning
PO_top_uh = merge_samples(PO_top, "code_uh")
PO_top_uh_order = tax_glom(PO_top_uh, "order")
#abund_table <- psmelt(PO_top_uh_order)
#abund_table$order <- as.character(abund_table$order)
#abund_table$order[abund_table$abundance < 0.01] <- "abundunce < 1%" 

order_tot <- plot_bar(PO_top_uh_order, fill="order") + geom_col(stat="identity", position="stack", color = "black") + frame_simple + scale_x_discrete(limits = order_uh) + labs(title="Abundance of nematodes at order level ", subtitle="Total abundance", x ="land use_habitat") + scale_y_continuous(labels = scales::comma) + guides(fill=guide_legend(nrow=12)) #+ fill_manual_order

PO_top_uh_order_relabun = transform_sample_counts(PO_top_uh_order, function(x) x/sum(x) * 100)

order_rel <- plot_bar(PO_top_uh_order_relabun, fill="order") + geom_col(stat="identity", position="stack", color = "black") + frame_simple + scale_x_discrete(limits = order_uh) + labs(title="Abundance of nematodes at order level ", subtitle="Relative abundance", x ="land use_habitat") + scale_y_continuous(labels = scales::comma) + guides(fill=guide_legend(nrow=12)) #+ fill_manual_order

abund_order <- plot_grid(order_tot, order_rel, ncol=1)
abund_order

#ggsave("analyses/pre-analysis/abund_order_nematodes_stand_filterdate_uh_2024-09-23.png", width=5, height=8)
```

```{r}
#topOTU = names(sort(taxa_sums(PO), TRUE)[1:100])
#PO_top = prune_taxa(topOTU, PO)
PO_top <- PO # if no pruning
PO_top_spatial = merge_samples(PO_top, "spatial")
PO_top_spatial_order = tax_glom(PO_top_spatial, "order")
#abund_table <- psmelt(PO_top_spatial_order)
#abund_table$order <- as.character(abund_table$order)
#abund_table$order[abund_table$abundance < 0.01] <- "abundunce < 1%" 

order_tot <- plot_bar(PO_top_spatial_order, fill="order") + geom_col(stat="identity", position="stack", color = "black") + frame_simple + labs(title="Abundance of nematodes at order level ", subtitle="Total abundance", x ="Spatial heterogeneity") + scale_y_continuous(labels = scales::comma) + guides(fill=guide_legend(nrow=12)) + scale_x_discrete(limits = order_spatial) #+ fill_manual_order

PO_top_spatial_order_relabun = transform_sample_counts(PO_top_spatial_order, function(x) x/sum(x) * 100)

order_rel <- plot_bar(PO_top_spatial_order_relabun, fill="order") + geom_col(stat="identity", position="stack", color = "black") + frame_simple + labs(title="Abundance of nematodes at order level ", subtitle="Relative abundance", x ="Spatial heterogeneity") + scale_y_continuous(labels = scales::comma) + guides(fill=guide_legend(nrow=12)) + scale_x_discrete(limits = order_spatial) #+ fill_manual_order

abund_order <- plot_grid(order_tot, order_rel, ncol=1)
abund_order

#ggsave("analyses/pre-analysis/abund_order_nematodes_stand_filterdate_spatial_2024-09-23.png", width=4, height=8)
```

```{r}
#topOTU = names(sort(taxa_sums(PO), TRUE)[1:100])
#PO_top = prune_taxa(topOTU, PO)
PO_top <- PO # if no pruning
PO_top_spatial = merge_samples(PO_top, "spatial")
PO_top_spatial_family = tax_glom(PO_top_spatial, "family")
#abund_table <- psmelt(PO_top_spatial_family)
#abund_table$family <- as.character(abund_table$family)
#abund_table$family[abund_table$abundance < 0.01] <- "abundunce < 1%" 

family_tot <- plot_bar(PO_top_spatial_family, fill="family") + geom_col(stat="identity", position="stack", color = "black") + frame_simple + labs(title="Abundance of nematodes at family level ", subtitle="Total abundance", x ="Spatial heterogeneity") + scale_y_continuous(labels = scales::comma) + guides(fill=guide_legend(nrow=12)) + scale_x_discrete(limits = order_spatial) #+ fill_manual_family

PO_top_spatial_family_relabun = transform_sample_counts(PO_top_spatial_family, function(x) x/sum(x) * 100)

family_rel <- plot_bar(PO_top_spatial_family_relabun, fill="family") + geom_col(stat="identity", position="stack", color = "black") + frame_simple + labs(title="Abundance of nematodes at family level ", subtitle="Relative abundance", x ="Spatial heterogeneity") + scale_y_continuous(labels = scales::comma) + guides(fill=guide_legend(nrow=12)) + scale_x_discrete(limits = order_spatial) #+ fill_manual_family

abund_family <- plot_grid(family_tot, family_rel, ncol=1)
abund_family

ggsave("analyses/pre-analysis/abund_family_nematodes_stand_filterdate_spatial_2024-09-23.png", width=6, height=8)
```

```{r}
# Filter functions
PO_bioag <- subset_taxa(PO, grepl("her|pla", abr_all_guild_level_AND))
PO_bioag_obs <- subset_taxa(PO_bioag, interaction == "observed")

PO_auxil <- subset_taxa(PO, grepl("pre|par", abr_all_guild_level_AND))
PO_auxil_obs <- subset_taxa(PO_auxil, interaction == "observed")
```

```{r}
PO_top <- PO_bioag
PO_top_spatial = merge_samples(PO_top, "spatial")
PO_top_spatial_family = tax_glom(PO_top_spatial, "family")

family_tot_bioag <- plot_bar(PO_top_spatial_family, fill="family") + geom_col(stat="identity", position="stack", color = "black") + frame_simple + labs(title="Abundance of nematodes at family level ", subtitle="Total abundance of potential bioagressors", x ="Spatial heterogeneity") + scale_y_continuous(labels = scales::comma)  + guides(fill=guide_legend(ncol=1)) + scale_x_discrete(limits = order_spatial)
family_tot_bioag

PO_top <- PO_auxil
PO_top_spatial = merge_samples(PO_top, "spatial")
PO_top_spatial_family = tax_glom(PO_top_spatial, "family")

family_tot_auxil <- plot_bar(PO_top_spatial_family, fill="family") + geom_col(stat="identity", position="stack", color = "black") + frame_simple + labs(title="Abundance of nematodes at family level ", subtitle="Total abundance of potential auxiliaries", x ="Spatial heterogeneity") + scale_y_continuous(labels = scales::comma)  + guides(fill=guide_legend(ncol=1)) + scale_x_discrete(limits = order_spatial)
family_tot_auxil

abund_tot_family <- plot_grid(family_tot_bioag, family_tot_auxil, ncol=1)
abund_tot_family

#ggsave("analyses/pre-analysis/abund_tot_family_nematodes_stand_filterdate_spatial_bioagANDauxil_2024-09-23.png", width=5, height=7.5)
```

```{r}
PO_top_obs <- PO_bioag_obs
PO_top_obs_spatial = merge_samples(PO_top_obs, "spatial")
PO_top_obs_spatial_family = tax_glom(PO_top_obs_spatial, "family")

family_tot_bioag_obs <- plot_bar(PO_top_obs_spatial_family, fill="family") + geom_col(stat="identity", position="stack", color = "black") + frame_simple + labs(title="Abundance of nematodes at family level ", subtitle="Total abundance of potential bioagressors with observed interactions", x ="Spatial heterogeneity") + scale_y_continuous(labels = scales::comma)  + guides(fill=guide_legend(ncol=1)) + scale_x_discrete(limits = order_spatial) #+ fill_manual_family
family_tot_bioag_obs

PO_top_obs <- PO_auxil_obs
PO_top_obs_spatial = merge_samples(PO_top_obs, "spatial")
PO_top_obs_spatial_family = tax_glom(PO_top_obs_spatial, "family")

family_tot_auxil_obs <- plot_bar(PO_top_obs_spatial_family, fill="family") + geom_col(stat="identity", position="stack", color = "black") + frame_simple + labs(title="Abundance of nematodes at family level ", subtitle="Total abundance of potential auxiliaries with observed interactions", x ="Spatial heterogeneity") + scale_y_continuous(labels = scales::comma)  + guides(fill=guide_legend(ncol=1)) + scale_x_discrete(limits = order_spatial) #+ fill_manual_family
family_tot_auxil_obs

abund_tot_family_obs <- plot_grid(family_tot_bioag_obs, family_tot_auxil_obs, ncol=1)
abund_tot_family_obs

#ggsave("analyses/pre-analysis/abund_tot_family_nematodes_stand_filterdate_spatial_bioagANDauxil_obs_2024-09-23.png", width=5.2, height=8.2)
```

```{r}
list_PO_bioag_obs <- PO_bioag_obs@tax_table
#write.csv(list_PO_bioag_obs, "analyses/pre-analysis/list_nematodes_bioag_obs_2024-09-23.csv", row.names = T)

list_PO_auxil_obs <- PO_auxil_obs@tax_table
#write.csv(list_PO_auxil_obs, "analyses/pre-analysis/list_nematodes_auxil_obs_2024-09-23.csv", row.names = T)
```

```{r}
V_spatial_bioag <- ggvenn_pq(PO_bioag, fact = "spatial", min_nb_seq = 0, taxonomic_rank = "family", add_nb_samples = F, set_size = 3) + labs(title = "Number of potential bioagressors at family level") # split_by = "date" # label = "count"
V_spatial_bioag

#save manually to ggvenn_nematodes_spatial_bioag_family_2024-09-23

V_spatial_auxil <- ggvenn_pq(PO_auxil, fact = "spatial", min_nb_seq = 0, taxonomic_rank = "family", add_nb_samples = F, set_size = 3) + labs(title = "Number of potential auxiliaries at family level") # split_by = "date" # label = "count"
V_spatial_auxil

#save manually to ggvenn_nematodes_spatial_auxil_family_2024-09-23
```

## micro_arthropodes

### Version raw

```{r}
# otu_table = tidy_data
## select the community
tidy_data_PO_micro_arthropodes <- tidy_data_PO %>% filter(community %in% c("micro_arthropodes"))

## remove useless columns
tidy_data_PO_micro_arthropodes <- tidy_data_PO_micro_arthropodes[,c(1,2,13)]

## transform table to taxa_are_rows = TRUE
tidy_data_PO_micro_arthropodes <- pivot_wider(tidy_data_PO_micro_arthropodes, names_from = code_uhbrd, values_from = abundance)

## replace NA by 0
tidy_data_PO_micro_arthropodes[is.na(tidy_data_PO_micro_arthropodes)] <- 0

## use "finest_name" as rownames
tidy_data_PO_micro_arthropodes <- tidy_data_PO_micro_arthropodes %>% tibble::column_to_rownames("finest_name")

## transform tbl_df to matrix table
tidy_data_PO_micro_arthropodes <- as.matrix(tidy_data_PO_micro_arthropodes)

# tax_table = tax_table # + fun_table
## select the community
tax_table_PO_micro_arthropodes <- tax_table %>% filter(community %in% c("micro_arthropodes"))

## remove useless columns
tax_table_PO_micro_arthropodes <- tax_table_PO_micro_arthropodes[,-c(9)]

########## Add fun_table here ########## 
tax_table_PO_micro_arthropodes_fun <- left_join(tax_table_PO_micro_arthropodes, fun_table_PO, by = c("family" = "matchName"))

tax_table_PO_micro_arthropodes_fun[is.na(tax_table_PO_micro_arthropodes_fun)] <- "unassigned"

tax_table_PO_micro_arthropodes_fun_inter <- tax_table_PO_micro_arthropodes_fun %>%
  rowwise() %>%
  mutate(interaction = if_else(any(c_across(1:8) %in% fun_table_inter_PO$matchName), "observed", "not observed")) %>%
  ungroup()

tax_table_PO_micro_arthropodes <- tax_table_PO_micro_arthropodes_fun_inter
########## END ########## 

## use "finest_name" as rownames
tax_table_PO_micro_arthropodes <- tax_table_PO_micro_arthropodes %>% tibble::column_to_rownames("finest_name")

## transform tbl_df to matrix table
tax_table_PO_micro_arthropodes <- as.matrix(tax_table_PO_micro_arthropodes)

# sam_data = metadata
## select the community
metadata_PO_micro_arthropodes <- metadata_PO %>% filter(community %in% c("micro_arthropodes"))

## choose the rownames
rownames(metadata_PO_micro_arthropodes) <- metadata_PO_micro_arthropodes[,11] # code_uhbrd

# create phyloseq object
tidy_data_PO_micro_arthropodes = otu_table(tidy_data_PO_micro_arthropodes, taxa_are_rows = TRUE)

tax_table_PO_micro_arthropodes = tax_table(tax_table_PO_micro_arthropodes)

metadata_PO_micro_arthropodes = sample_data(metadata_PO_micro_arthropodes)
  
PO_micro_arthropodes <- phyloseq(tidy_data_PO_micro_arthropodes, tax_table_PO_micro_arthropodes, metadata_PO_micro_arthropodes)

# check PO
PO_micro_arthropodes
sample_names(PO_micro_arthropodes)
rank_names(PO_micro_arthropodes)
sample_variables(PO_micro_arthropodes)

#write_rds(PO_micro_arthropodes, "data/derived_data/phyloseq_objects/PO_micro_arthropodes_raw_2024-09-23.rds")
#PO_micro_arthropodes_df <- phyloseq_to_df(PO_micro_arthropodes, addtax = T, addtot = T, addmaxrank = F, sorting = "NULL")
#write.csv(PO_micro_arthropodes_df, "data/derived_data/phyloseq_objects/PO_micro_arthropodes_raw_2024-09-23.csv", row.names = FALSE)
```

### Version stand

```{r}
# standardize abundances to the median sequencing depth
total = median(sample_sums(PO_micro_arthropodes))
standf = function(x, t=total) round(t*(x/sum(x)))
PO_micro_arthropodes_stand = transform_sample_counts(PO_micro_arthropodes, standf)

# check PO
PO_micro_arthropodes_stand

#write_rds(PO_micro_arthropodes_stand, "data/derived_data/phyloseq_objects/PO_micro_arthropodes_stand_2024-09-23.rds")
#PO_micro_arthropodes_df <- phyloseq_to_df(PO_micro_arthropodes_stand, addtax = T, addtot = T, addmaxrank = F, sorting = "NULL")
#write.table(PO_micro_arthropodes_df, "data/derived_data/phyloseq_objects/PO_micro_arthropodes_stand_2024-09-23.csv", row.names = FALSE)

propfun = function(x){
  xprop = (x / sum(x))
  return(xprop)
}
PO_micro_arthropodes_stand_prop <- transform_sample_counts(PO_micro_arthropodes_stand, fun = propfun)

#write_rds(PO_micro_arthropodes_stand_prop, "data/derived_data/phyloseq_objects/PO_micro_arthropodes_stand_prop_2024-09-23.rds")
#PO_micro_arthropodes_df <- phyloseq_to_df(PO_micro_arthropodes_stand_prop, addtax = T, addtot = T, addmaxrank = F, sorting = "NULL")
#write.table(PO_micro_arthropodes_df, "data/derived_data/phyloseq_objects/PO_micro_arthropodes_stand_prop_2024-09-23.csv", row.names = FALSE)
```

### Version norm

```{r}
# normalize
PO_micro_arthropodes_norm <- normalize_prop_pq(PO_micro_arthropodes, base_log = 2, digits = 0)

# check PO
PO_micro_arthropodes_norm

#write_rds(PO_micro_arthropodes_norm, "data/derived_data/phyloseq_objects/PO_micro_arthropodes_norm_2024-09-23.rds")
#PO_micro_arthropodes_df <- phyloseq_to_df(PO_micro_arthropodes_norm, addtax = T, addtot = T, addmaxrank = F, sorting = "NULL")
#write.table(PO_micro_arthropodes_df, "data/derived_data/phyloseq_objects/PO_micro_arthropodes_norm_2024-09-23.csv", row.names = FALSE)

propfun = function(x){
  xprop = (x / sum(x))
  return(xprop)
}
PO_micro_arthropodes_norm_prop <- transform_sample_counts(PO_micro_arthropodes_norm, fun = propfun)

#write_rds(PO_micro_arthropodes_norm_prop, "data/derived_data/phyloseq_objects/PO_micro_arthropodes_norm_prop_2024-09-23.rds")
#PO_micro_arthropodes_df <- phyloseq_to_df(PO_micro_arthropodes_norm_prop, addtax = T, addtot = T, addmaxrank = F, sorting = "NULL")
#write.csv(PO_micro_arthropodes_df[,-1], "data/derived_data/phyloseq_objects/PO_micro_arthropodes_norm_prop_2024-09-23.csv", row.names = FALSE)
```

### Pre-analyses

```{r}
tidy_data_PO_micro_arthropodes_before <- as.data.frame(t(tidy_data_PO_micro_arthropodes))
sample_names = rownames(tidy_data_PO_micro_arthropodes_before)
curve_before <- rarecurve(tidy_data_PO_micro_arthropodes_before, step = 5, label = F)
# save image to rarecurves_micro_arthropodes_raw_2024-09-23

tidy_data_PO_micro_arthropodes_stand <- otu_table(PO_micro_arthropodes_stand)
class(tidy_data_PO_micro_arthropodes_stand) <- "matrix"
tidy_data_PO_micro_arthropodes_stand <- t(tidy_data_PO_micro_arthropodes_stand)
curve_stand <- rarecurve(tidy_data_PO_micro_arthropodes_stand, step=5, label=F)
# save image to rarecurves_micro_arthropodes_stand_2024-09-23

tidy_data_PO_micro_arthropodes_norm <- otu_table(PO_micro_arthropodes_norm)
class(tidy_data_PO_micro_arthropodes_norm) <- "matrix"
tidy_data_PO_micro_arthropodes_norm <- t(tidy_data_PO_micro_arthropodes_norm)
curve_stand <- rarecurve(tidy_data_PO_micro_arthropodes_norm, step=5, label=F)
# save image to rarecurves_micro_arthropodes_norm_2024-09-23
```

```{r}
# Choose the version
PO <- PO_micro_arthropodes # raw
PO <- PO_micro_arthropodes_stand # this one used first
PO <- PO_micro_arthropodes_stand_prop
PO <- PO_micro_arthropodes_norm
PO <- PO_micro_arthropodes_norm_prop
PO <- PO_bioag
PO <- PO_auxil

# Filter date
PO <- subset_samples(PO, date == "17.04.2023")
```

```{r}
# Order levels
PO@sam_data$date <- factor(PO@sam_data$date, levels = order_date)
PO@sam_data$code_uh <- factor(PO@sam_data$code_uh, levels = order_uh)
PO@sam_data$spatial <- factor(PO@sam_data$spatial, levels = order_spatial)
```

```{r}
alpha_table <- estimate_richness(PO)
#write.table(alpha_table, "analyses/pre-analysis/alpha_micro_arthropodes_stand_2024-09-23.csv", sep = ",")

PO_alpha <- merge(PO@sam_data, alpha_table, by = 0)
quickanova <- aov(Observed ~ spatial, PO_alpha)
summary(quickanova)
quickanova <- aov(Shannon ~ spatial, PO_alpha)
summary(quickanova)
quickanova <- aov(InvSimpson ~ spatial, PO_alpha)
summary(quickanova)
```

```{r}
observed <- plot_richness(PO, x="code_uh", shape="block", color="habitat", measures=c("Observed")) +
  geom_boxplot(shape="block") +
  geom_point(position = "jitter") +
  theme_bw() + frame_simple + scale_x_discrete(limits = order_uh) +
  theme(axis.text.x=element_text(angle = -90, hjust = 0)) +
  labs(title="Alpha diversity of micro_arthropodes", subtitle="Taxonomy table", x ="land use_habitat", y="Observed richness") + facet_wrap(date ~ .)
observed$layers <- observed$layers[-1]

shannon <- plot_richness(PO, x="code_uh", shape="block", color="habitat", measures=c("Shannon")) +
  geom_boxplot(shape="block") +
  geom_point(position = "jitter") +
  theme_bw() + frame_simple + scale_x_discrete(limits = order_uh) +
  theme(axis.text.x=element_text(angle = -90, hjust = 0)) +
  labs(title="Alpha diversity of micro_arthropodes", subtitle="Taxonomy table", x ="land use_habitat", y="Shannon index") + facet_wrap(date ~ .)
shannon$layers <- shannon$layers[-1]

simpson <- plot_richness(PO, x="code_uh", shape="block", color="habitat", measures=c("InvSimpson")) +
  geom_boxplot(shape="block") +
  geom_point(position = "jitter") +
  theme_bw() + frame_simple + scale_x_discrete(limits = order_uh) +
  theme(axis.text.x=element_text(angle = -90, hjust = 0)) +
  labs(title="Alpha diversity of micro_arthropodes", subtitle="Taxonomy table", x ="land use_habitat", y="Simpson inverse index") + facet_wrap(date ~ .)
simpson$layers <- simpson$layers[-1]

alpha_graph <- plot_grid(observed, shannon, simpson, ncol=1)
alpha_graph

#ggsave("analyses/pre-analysis/alpha_micro_arthropodes_stand_uh_2024-09-23.png", width=3.5, height=10)
```

```{r}
PO_ord <- ordinate(PO, "PCoA", distance = "bray")
ord <- plot_ordination(PO, PO_ord, color = "code_uh", shape = "block") + theme_bw() + labs(title="Beta diversity of micro_arthropodes", subtitle="Taxonomy table")
ord

#ggsave("analyses/pre-analysis/beta_micro_arthropodes_plot_ordination_stand_uhb_2024-09-23.png", width=5, height=4)
```

```{r}
#ord_explore(PO)

PO %>%
 tax_transform(rank = "unique", trans = "identity") %>%
 dist_calc(dist = "bray") %>%
 ord_calc(
  method = "PCoA"
 ) %>% 
 ord_plot(
  axes = c(1, 2),
  colour = "date", fill = "date",
  shape = "habitat", alpha = 0.5,
  size = 2
 ) + 
 scale_shape_girafe_filled() +
 ggplot2::stat_ellipse(
  ggplot2::aes(colour = date)
 )

#ggsave("analyses/pre-analysis/beta_micro_arthropodes_microViz_stand_hd_2024-09-23.png", width=5, height=4, bg = "white")
```

```{r}
# Filter before to continue here

observed <- plot_richness(PO, x="habitat", shape="block", color="habitat", measures=c("Observed")) +
  geom_boxplot(shape="block") +
  geom_point(position = "jitter") +
  theme_bw() + frame_simple +
  labs(title="Alpha diversity of micro_arthropodes", subtitle="Taxonomy table", x =" ", y="Observed richness") + facet_wrap(spatial ~ .)
observed$layers <- observed$layers[-1]

shannon <- plot_richness(PO, x="habitat", shape="block", color="habitat", measures=c("Shannon")) +
  geom_boxplot(shape="block") +
  geom_point(position = "jitter") +
  theme_bw() + frame_simple +
  labs(title="Alpha diversity of micro_arthropodes", subtitle="Taxonomy table", x =" ", y="Shannon index") + facet_wrap(spatial ~ .)
shannon$layers <- shannon$layers[-1]

simpson <- plot_richness(PO, x="habitat", shape="block", color="habitat", measures=c("InvSimpson")) +
  geom_boxplot(shape="block") +
  geom_point(position = "jitter") +
  theme_bw() + frame_simple +
  labs(title="Alpha diversity of micro_arthropodes", subtitle="Taxonomy table", x =" ", y="Simpson inverse index") + facet_wrap(spatial ~ .)
simpson$layers <- simpson$layers[-1]

alpha_graph <- plot_grid(observed, shannon, simpson, ncol=1)
alpha_graph

#ggsave("analyses/pre-analysis/alpha_micro_arthropodes_stand_filterdate_spatial_2024-09-23.png", width=4, height=10)
#ggsave("analyses/pre-analysis/alpha_micro_arthropodes_stand_filterdate_spatial_bioag_2024-09-23.png", width=4, height=10)
#ggsave("analyses/pre-analysis/alpha_micro_arthropodes_stand_filterdate_spatial_auxil_2024-09-23.png", width=4, height=10)
```

```{r}
PO_ord <- ordinate(PO, "PCoA", distance = "bray")
ord <- plot_ordination(PO, PO_ord, color = "spatial", shape = "block") + theme_bw() + labs(title="Beta diversity of micro_arthropodes", subtitle="Taxonomy table")
ord

#ggsave("analyses/pre-analysis/beta_micro_arthropodes_plot_ordination_stand_filter_date_spatial_2024-09-23.png", width=5, height=4)
```

```{r}
#ord_explore(PO)

PO %>%
 tax_transform(rank = "unique", trans = "identity") %>%
 dist_calc(dist = "bray") %>%
 ord_calc(
  method = "PCoA"
 ) %>% 
 ord_plot(
  axes = c(1, 2),
  colour = "spatial", fill = "spatial",
  shape = "habitat", alpha = 0.5,
  size = 2
 ) + 
 scale_shape_girafe_filled() +
 ggplot2::stat_ellipse(
  ggplot2::aes(colour = spatial)
 )

#ggsave("analyses/pre-analysis/beta_micro_arthropodes_microViz_stand_filterdate_spatial_2024-09-23.png", width=5, height=4, bg = "white")
```

```{r}
adonis_pq(PO, "spatial", na_remove = TRUE)
adonis_pq(PO, "spatial*habitat", na_remove = TRUE)
adonis_pq(PO, "land_use*habitat", na_remove = TRUE)
adonis_pq(PO, "code_uh", na_remove = TRUE)
adonis_pq(PO, "code_uh*block", na_remove = TRUE)
adonis_pq(PO, "code_uh*replicate", na_remove = TRUE)
```

```{r}
#topOTU = names(sort(taxa_sums(PO), TRUE)[1:100])
#PO_top = prune_taxa(topOTU, PO)
PO_top <- PO # if no pruning
PO_top_uh = merge_samples(PO_top, "code_uh")
PO_top_uh_order = tax_glom(PO_top_uh, "order")
#abund_table <- psmelt(PO_top_uh_order)
#abund_table$order <- as.character(abund_table$order)
#abund_table$order[abund_table$abundance < 0.01] <- "abundunce < 1%" 

order_tot <- plot_bar(PO_top_uh_order, fill="order") + geom_col(stat="identity", position="stack", color = "black") + frame_simple + scale_x_discrete(limits = order_uh) + labs(title="Abundance of micro_arthropodes at order level ", subtitle="Total abundance", x ="land use_habitat") + scale_y_continuous(labels = scales::comma) + guides(fill=guide_legend(nrow=11)) #+ fill_manual_order

PO_top_uh_order_relabun = transform_sample_counts(PO_top_uh_order, function(x) x/sum(x) * 100)

order_rel <- plot_bar(PO_top_uh_order_relabun, fill="order") + geom_col(stat="identity", position="stack", color = "black") + frame_simple + scale_x_discrete(limits = order_uh) + labs(title="Abundance of micro_arthropodes at order level ", subtitle="Relative abundance", x ="land use_habitat") + scale_y_continuous(labels = scales::comma) + guides(fill=guide_legend(nrow=11)) #+ fill_manual_order

abund_order <- plot_grid(order_tot, order_rel, ncol=1)
abund_order

#ggsave("analyses/pre-analysis/abund_order_micro_arthropodes_stand_filterdate_uh_2024-09-23.png", width=6, height=9)
```

```{r}
#topOTU = names(sort(taxa_sums(PO), TRUE)[1:100])
#PO_top = prune_taxa(topOTU, PO)
PO_top <- PO # if no pruning
PO_top_spatial = merge_samples(PO_top, "spatial")
PO_top_spatial_order = tax_glom(PO_top_spatial, "order")
#abund_table <- psmelt(PO_top_spatial_order)
#abund_table$order <- as.character(abund_table$order)
#abund_table$order[abund_table$abundance < 0.01] <- "abundunce < 1%" 

order_tot <- plot_bar(PO_top_spatial_order, fill="order") + geom_col(stat="identity", position="stack", color = "black") + frame_simple + labs(title="Abundance of micro_arthropodes at order level ", subtitle="Total abundance", x ="Spatial heterogeneity") + scale_y_continuous(labels = scales::comma) + guides(fill=guide_legend(nrow=11)) + scale_x_discrete(limits = order_spatial) #+ fill_manual_order

PO_top_spatial_order_relabun = transform_sample_counts(PO_top_spatial_order, function(x) x/sum(x) * 100)

order_rel <- plot_bar(PO_top_spatial_order_relabun, fill="order") + geom_col(stat="identity", position="stack", color = "black") + frame_simple + labs(title="Abundance of micro_arthropodes at order level ", subtitle="Relative abundance", x ="Spatial heterogeneity") + scale_y_continuous(labels = scales::comma) + guides(fill=guide_legend(nrow=11)) + scale_x_discrete(limits = order_spatial) #+ fill_manual_order

abund_order <- plot_grid(order_tot, order_rel, ncol=1)
abund_order

#ggsave("analyses/pre-analysis/abund_order_micro_arthropodes_stand_filterdate_spatial_2024-09-23.png", width=5.5, height=9)
```

```{r}
# Filter functions
PO_bioag <- subset_taxa(PO, grepl("her|pla", abr_all_guild_level_AND))
#PO_bioag_obs <- subset_taxa(PO_bioag, interaction == "observed")

PO_auxil <- subset_taxa(PO, grepl("pre|par", abr_all_guild_level_AND))
PO_auxil_obs <- subset_taxa(PO_auxil, interaction == "observed")
```

```{r}
PO_top <- PO_bioag
PO_top_spatial = merge_samples(PO_top, "spatial")
PO_top_spatial_family = tax_glom(PO_top_spatial, "family")

family_tot_bioag <- plot_bar(PO_top_spatial_family, fill="family") + geom_col(stat="identity", position="stack", color = "black") + frame_simple + labs(title="Abundance of micro_arthropodes at family level ", subtitle="Total abundance of potential bioagressors", x ="Spatial heterogeneity") + scale_y_continuous(labels = scales::comma) + guides(fill=guide_legend(ncol=1)) + scale_x_discrete(limits = order_spatial)
family_tot_bioag

PO_top <- PO_auxil
PO_top_spatial = merge_samples(PO_top, "spatial")
PO_top_spatial_family = tax_glom(PO_top_spatial, "family")

family_tot_auxil <- plot_bar(PO_top_spatial_family, fill="family") + geom_col(stat="identity", position="stack", color = "black") + frame_simple + labs(title="Abundance of micro_arthropodes at family level ", subtitle="Total abundance of potential auxiliaries", x ="Spatial heterogeneity") + scale_y_continuous(labels = scales::comma) + guides(fill=guide_legend(ncol=1)) + scale_x_discrete(limits = order_spatial)
family_tot_auxil

abund_tot_family <- plot_grid(family_tot_bioag, family_tot_auxil, ncol=1)
abund_tot_family

#ggsave("analyses/pre-analysis/abund_tot_family_micro_arthropodes_stand_filterdate_spatial_bioagANDauxil_2024-09-23.png", width=5, height=9)
```

```{r}
PO_top <- PO_bioag
PO_top_spatial = merge_samples(PO_top, "spatial")
PO_top_spatial_order = tax_glom(PO_top_spatial, "order")

order_tot_bioag <- plot_bar(PO_top_spatial_order, fill="order") + geom_col(stat="identity", position="stack", color = "black") + frame_simple + labs(title="Abundance of micro_arthropodes at order level ", subtitle="Total abundance of potential bioagressors", x ="Spatial heterogeneity") + scale_y_continuous(labels = scales::comma) + guides(fill=guide_legend(nrow=10)) + scale_x_discrete(limits = order_spatial) + fill_manual_order
order_tot_bioag

PO_top <- PO_auxil
PO_top_spatial = merge_samples(PO_top, "spatial")
PO_top_spatial_order = tax_glom(PO_top_spatial, "order")

order_tot_auxil <- plot_bar(PO_top_spatial_order, fill="order") + geom_col(stat="identity", position="stack", color = "black") + frame_simple + labs(title="Abundance of micro_arthropodes at order level ", subtitle="Total abundance of potential auxiliaries", x ="Spatial heterogeneity") + scale_y_continuous(labels = scales::comma) + guides(fill=guide_legend(nrow=10)) + scale_x_discrete(limits = order_spatial) + fill_manual_order
order_tot_auxil

abund_tot_order <- plot_grid(order_tot_bioag, order_tot_auxil, ncol=1)
abund_tot_order

#ggsave("analyses/pre-analysis/abund_tot_order_micro_arthropodes_stand_filterdate_spatial_bioagANDauxil_2024-09-23.png", width=5, height=9)
```

```{r}
#PO_top <- PO_bioag_obs
#PO_top_spatial = merge_samples(PO_top, "spatial")
#PO_top_spatial_family = tax_glom(PO_top_spatial, "family")

#family_tot_bioag <- plot_bar(PO_top_spatial_family, fill="family") + geom_col(stat="identity", position="stack", color = "black") + frame_simple + labs(title="Abundance of micro_arthropodes at family level ", subtitle="Total abundance of potential bioagressors with observed interactions", x ="Spatial heterogeneity") + scale_y_continuous(labels = scales::comma)  + guides(fill=guide_legend(ncol=1)) + scale_x_discrete(limits = order_spatial) + fill_manual_family
#family_tot_bioag

PO_top <- PO_auxil_obs
PO_top_spatial = merge_samples(PO_top, "spatial")
PO_top_spatial_family = tax_glom(PO_top_spatial, "family")

family_tot_auxil <- plot_bar(PO_top_spatial_family, fill="family") + geom_col(stat="identity", position="stack", color = "black") + frame_simple + labs(title="Abundance of micro_arthropodes at family level ", subtitle="Total abundance of potential auxiliaries with observed interactions", x ="Spatial heterogeneity") + scale_y_continuous(labels = scales::comma)  + guides(fill=guide_legend(ncol=1)) + scale_x_discrete(limits = order_spatial) + fill_manual_family
family_tot_auxil

#abund_tot_family <- plot_grid(family_tot_bioag, family_tot_auxil, ncol=1)
#abund_tot_family

#ggsave("analyses/pre-analysis/abund_tot_family_micro_arthropodes_stand_filterdate_spatial_bioagANDauxil_obs_2024-09-23.png", width=5, height=4)
```

```{r}
#list_PO_bioag_obs <- PO_bioag_obs@tax_table
#write.csv(list_PO_bioag_obs, "analyses/pre-analysis/list_micro_arthropodes_bioag_obs_2024-09-23.csv", row.names = T)

list_PO_auxil_obs <- PO_auxil_obs@tax_table
#write.csv(list_PO_auxil_obs, "analyses/pre-analysis/list_micro_arthropodes_auxil_obs_2024-09-23.csv", row.names = T)
```

```{r}
V_spatial_bioag <- ggvenn_pq(PO_bioag, fact = "spatial", min_nb_seq = 0, taxonomic_rank = "family", add_nb_samples = F, set_size = 3) + labs(title = "Number of potential bioagressors at family level") # split_by = "date" # label = "count"
V_spatial_bioag

#save manually to ggvenn_micro_arthropodes_spatial_bioag_family_2024-09-23

V_spatial_auxil <- ggvenn_pq(PO_auxil, fact = "spatial", min_nb_seq = 0, taxonomic_rank = "family", add_nb_samples = F, set_size = 3) + labs(title = "Number of potential auxiliaries at family level") # split_by = "date" # label = "count"
V_spatial_auxil

#save manually to ggvenn_micro_arthropodes_spatial_auxil_family_2024-09-23
```

## microorganisms

### bacteria

#### Version raw

```{r}
# otu_table = tidy_data
## select the community
tidy_data_PO_bacteria <- tidy_data_PO %>% filter(community %in% c("bacteria"))

## remove useless columns
tidy_data_PO_bacteria <- tidy_data_PO_bacteria[,c(1,2,13)]

## transform table to taxa_are_rows = TRUE
tidy_data_PO_bacteria <- pivot_wider(tidy_data_PO_bacteria, names_from = code_uhbrd, values_from = abundance)

## replace NA by 0
tidy_data_PO_bacteria[is.na(tidy_data_PO_bacteria)] <- 0

## use "finest_name" as rownames
tidy_data_PO_bacteria <- tidy_data_PO_bacteria %>% tibble::column_to_rownames("finest_name")

## transform tbl_df to matrix table
tidy_data_PO_bacteria <- as.matrix(tidy_data_PO_bacteria)

# tax_table = tax_table # + fun_table
## select the community
tax_table_PO_bacteria <- tax_table %>% filter(community %in% c("bacteria"))

## remove useless columns
tax_table_PO_bacteria <- tax_table_PO_bacteria[,-c(9)]

########## Add fun_table here ########## 
tax_table_PO_bacteria_fun <- left_join(tax_table_PO_bacteria, fun_table_PO, by = c("family" = "matchName"))

tax_table_PO_bacteria_fun[is.na(tax_table_PO_bacteria_fun)] <- "unassigned"

tax_table_PO_bacteria_fun_inter <- tax_table_PO_bacteria_fun %>%
  rowwise() %>%
  mutate(interaction = if_else(any(c_across(1:8) %in% fun_table_inter_PO$matchName), "observed", "not observed")) %>%
  ungroup()

tax_table_PO_bacteria <- tax_table_PO_bacteria_fun_inter
########## END ########## 

## use "finest_name" as rownames
tax_table_PO_bacteria <- tax_table_PO_bacteria %>% tibble::column_to_rownames("finest_name")

## transform tbl_df to matrix table
tax_table_PO_bacteria <- as.matrix(tax_table_PO_bacteria)

# sam_data = metadata
## select the community
metadata_PO_bacteria <- metadata_PO %>% filter(community %in% c("bacteria"))

## choose the rownames
rownames(metadata_PO_bacteria) <- metadata_PO_bacteria[,11] # code_uhbrd

# create phyloseq object
tidy_data_PO_bacteria = otu_table(tidy_data_PO_bacteria, taxa_are_rows = TRUE)

tax_table_PO_bacteria = tax_table(tax_table_PO_bacteria)

metadata_PO_bacteria = sample_data(metadata_PO_bacteria)
  
PO_bacteria <- phyloseq(tidy_data_PO_bacteria, tax_table_PO_bacteria, metadata_PO_bacteria)

# check PO
PO_bacteria
sample_names(PO_bacteria)
rank_names(PO_bacteria)
sample_variables(PO_bacteria)

#write_rds(PO_bacteria, "data/derived_data/phyloseq_objects/PO_bacteria_raw_2024-09-23.rds")
#PO_bacteria_df <- phyloseq_to_df(PO_bacteria, addtax = T, addtot = T, addmaxrank = F, sorting = "NULL")
#write.csv(PO_bacteria_df, "data/derived_data/phyloseq_objects/PO_bacteria_raw_2024-09-23.csv", row.names = FALSE)
```

#### Version stand

```{r}
# standardize abundances to the median sequencing depth
total = median(sample_sums(PO_bacteria))
standf = function(x, t=total) round(t*(x/sum(x)))
PO_bacteria_stand = transform_sample_counts(PO_bacteria, standf)

# check PO
PO_bacteria_stand

#write_rds(PO_bacteria_stand, "data/derived_data/phyloseq_objects/PO_bacteria_stand_2024-09-23.rds")
#PO_bacteria_df <- phyloseq_to_df(PO_bacteria_stand, addtax = T, addtot = T, addmaxrank = F, sorting = "NULL")
#write.table(PO_bacteria_df, "data/derived_data/phyloseq_objects/PO_bacteria_stand_2024-09-23.csv", row.names = FALSE)

propfun = function(x){
  xprop = (x / sum(x))
  return(xprop)
}
PO_bacteria_stand_prop <- transform_sample_counts(PO_bacteria_stand, fun = propfun)

#write_rds(PO_bacteria_stand_prop, "data/derived_data/phyloseq_objects/PO_bacteria_stand_prop_2024-09-23.rds")
#PO_bacteria_df <- phyloseq_to_df(PO_bacteria_stand_prop, addtax = T, addtot = T, addmaxrank = F, sorting = "NULL")
#write.table(PO_bacteria_df, "data/derived_data/phyloseq_objects/PO_bacteria_stand_prop_2024-09-23.csv", row.names = FALSE)
```

#### Version norm

```{r}
# normalize
PO_bacteria_norm <- normalize_prop_pq(PO_bacteria, base_log = 2, digits = 0)

# check PO
PO_bacteria_norm

#write_rds(PO_bacteria_norm, "data/derived_data/phyloseq_objects/PO_bacteria_norm_2024-09-23.rds")
#PO_bacteria_df <- phyloseq_to_df(PO_bacteria_norm, addtax = T, addtot = T, addmaxrank = F, sorting = "NULL")
#write.table(PO_bacteria_df, "data/derived_data/phyloseq_objects/PO_bacteria_norm_2024-09-23.csv", row.names = FALSE)

propfun = function(x){
  xprop = (x / sum(x))
  return(xprop)
}
PO_bacteria_norm_prop <- transform_sample_counts(PO_bacteria_norm, fun = propfun)

#write_rds(PO_bacteria_norm_prop, "data/derived_data/phyloseq_objects/PO_bacteria_norm_prop_2024-09-23.rds")
#PO_bacteria_df <- phyloseq_to_df(PO_bacteria_norm_prop, addtax = T, addtot = T, addmaxrank = F, sorting = "NULL")
#write.csv(PO_bacteria_df[,-1], "data/derived_data/phyloseq_objects/PO_bacteria_norm_prop_2024-09-23.csv", row.names = FALSE)
```

#### Pre-analyses

```{r}
tidy_data_PO_bacteria_before <- as.data.frame(t(tidy_data_PO_bacteria))
sample_names = rownames(tidy_data_PO_bacteria_before)
curve_before <- rarecurve(tidy_data_PO_bacteria_before, step = 50, label = F)
# save image to rarecurves_bacteria_raw_2024-09-23

tidy_data_PO_bacteria_stand <- otu_table(PO_bacteria_stand)
class(tidy_data_PO_bacteria_stand) <- "matrix"
tidy_data_PO_bacteria_stand <- t(tidy_data_PO_bacteria_stand)
curve_stand <- rarecurve(tidy_data_PO_bacteria_stand, step=50, label=F)
# save image to rarecurves_bacteria_stand_2024-09-23

tidy_data_PO_bacteria_norm <- otu_table(PO_bacteria_norm)
class(tidy_data_PO_bacteria_norm) <- "matrix"
tidy_data_PO_bacteria_norm <- t(tidy_data_PO_bacteria_norm)
curve_stand <- rarecurve(tidy_data_PO_bacteria_norm, step=50, label=F)
# save image to rarecurves_bacteria_norm_2024-09-23
```

### fungi

#### Version raw

```{r}
# otu_table = tidy_data
## select the community
tidy_data_PO_fungi <- tidy_data_PO %>% filter(community %in% c("fungi"))

## remove useless columns
tidy_data_PO_fungi <- tidy_data_PO_fungi[,c(1,2,13)]

## transform table to taxa_are_rows = TRUE
tidy_data_PO_fungi <- pivot_wider(tidy_data_PO_fungi, names_from = code_uhbrd, values_from = abundance)

## replace NA by 0
tidy_data_PO_fungi[is.na(tidy_data_PO_fungi)] <- 0

## use "finest_name" as rownames
tidy_data_PO_fungi <- tidy_data_PO_fungi %>% tibble::column_to_rownames("finest_name")

## transform tbl_df to matrix table
tidy_data_PO_fungi <- as.matrix(tidy_data_PO_fungi)

# tax_table = tax_table # + fun_table
## select the community
tax_table_PO_fungi <- tax_table %>% filter(community %in% c("fungi"))

## remove useless columns
tax_table_PO_fungi <- tax_table_PO_fungi[,-c(9)]

########## Add fun_table here ########## 
tax_table_PO_fungi_fun <- left_join(tax_table_PO_fungi, fun_table_PO, by = c("family" = "matchName"))

tax_table_PO_fungi_fun[is.na(tax_table_PO_fungi_fun)] <- "unassigned"

tax_table_PO_fungi_fun_inter <- tax_table_PO_fungi_fun %>%
  rowwise() %>%
  mutate(interaction = if_else(any(c_across(1:8) %in% fun_table_inter_PO$matchName), "observed", "not observed")) %>%
  ungroup()

tax_table_PO_fungi <- tax_table_PO_fungi_fun_inter
########## END ########## 

## use "finest_name" as rownames
tax_table_PO_fungi <- tax_table_PO_fungi %>% tibble::column_to_rownames("finest_name")

## transform tbl_df to matrix table
tax_table_PO_fungi <- as.matrix(tax_table_PO_fungi)

# sam_data = metadata
## select the community
metadata_PO_fungi <- metadata_PO %>% filter(community %in% c("fungi"))

## choose the rownames
rownames(metadata_PO_fungi) <- metadata_PO_fungi[,11] # code_uhbrd

# create phyloseq object
tidy_data_PO_fungi = otu_table(tidy_data_PO_fungi, taxa_are_rows = TRUE)

tax_table_PO_fungi = tax_table(tax_table_PO_fungi)

metadata_PO_fungi = sample_data(metadata_PO_fungi)
  
PO_fungi <- phyloseq(tidy_data_PO_fungi, tax_table_PO_fungi, metadata_PO_fungi)

# check PO
PO_fungi
sample_names(PO_fungi)
rank_names(PO_fungi)
sample_variables(PO_fungi)

#write_rds(PO_fungi, "data/derived_data/phyloseq_objects/PO_fungi_raw_2024-09-23.rds")
#PO_fungi_df <- phyloseq_to_df(PO_fungi, addtax = T, addtot = T, addmaxrank = F, sorting = "NULL")
#write.csv(PO_fungi_df, "data/derived_data/phyloseq_objects/PO_fungi_raw_2024-09-23.csv", row.names = FALSE)
```

#### Version stand

```{r}
# standardize abundances to the median sequencing depth
total = median(sample_sums(PO_fungi))
standf = function(x, t=total) round(t*(x/sum(x)))
PO_fungi_stand = transform_sample_counts(PO_fungi, standf)

# check PO
PO_fungi_stand

#write_rds(PO_fungi_stand, "data/derived_data/phyloseq_objects/PO_fungi_stand_2024-09-23.rds")
#PO_fungi_df <- phyloseq_to_df(PO_fungi_stand, addtax = T, addtot = T, addmaxrank = F, sorting = "NULL")
#write.table(PO_fungi_df, "data/derived_data/phyloseq_objects/PO_fungi_stand_2024-09-23.csv", row.names = FALSE)

propfun = function(x){
  xprop = (x / sum(x))
  return(xprop)
}
PO_fungi_stand_prop <- transform_sample_counts(PO_fungi_stand, fun = propfun)

#write_rds(PO_fungi_stand_prop, "data/derived_data/phyloseq_objects/PO_fungi_stand_prop_2024-09-23.rds")
#PO_fungi_df <- phyloseq_to_df(PO_fungi_stand_prop, addtax = T, addtot = T, addmaxrank = F, sorting = "NULL")
#write.table(PO_fungi_df, "data/derived_data/phyloseq_objects/PO_fungi_stand_prop_2024-09-23.csv", row.names = FALSE)
```

#### Version norm

```{r}
# normalize
PO_fungi_norm <- normalize_prop_pq(PO_fungi, base_log = 2, digits = 0)

# check PO
PO_fungi_norm

#write_rds(PO_fungi_norm, "data/derived_data/phyloseq_objects/PO_fungi_norm_2024-09-23.rds")
#PO_fungi_df <- phyloseq_to_df(PO_fungi_norm, addtax = T, addtot = T, addmaxrank = F, sorting = "NULL")
#write.table(PO_fungi_df, "data/derived_data/phyloseq_objects/PO_fungi_norm_2024-09-23.csv", row.names = FALSE)

propfun = function(x){
  xprop = (x / sum(x))
  return(xprop)
}
PO_fungi_norm_prop <- transform_sample_counts(PO_fungi_norm, fun = propfun)

#write_rds(PO_fungi_norm_prop, "data/derived_data/phyloseq_objects/PO_fungi_norm_prop_2024-09-23.rds")
#PO_fungi_df <- phyloseq_to_df(PO_fungi_norm_prop, addtax = T, addtot = T, addmaxrank = F, sorting = "NULL")
#write.csv(PO_fungi_df[,-1], "data/derived_data/phyloseq_objects/PO_fungi_norm_prop_2024-09-23.csv", row.names = FALSE)
```

#### Pre-analyses

```{r}
tidy_data_PO_fungi_before <- as.data.frame(t(tidy_data_PO_fungi))
sample_names = rownames(tidy_data_PO_fungi_before)
curve_before <- rarecurve(tidy_data_PO_fungi_before, step = 50, label = F)
# save image to rarecurves_fungi_raw_2024-09-23

tidy_data_PO_fungi_stand <- otu_table(PO_fungi_stand)
class(tidy_data_PO_fungi_stand) <- "matrix"
tidy_data_PO_fungi_stand <- t(tidy_data_PO_fungi_stand)
curve_stand <- rarecurve(tidy_data_PO_fungi_stand, step=50, label=F)
# save image to rarecurves_fungi_stand_2024-09-23

tidy_data_PO_fungi_norm <- otu_table(PO_fungi_norm)
class(tidy_data_PO_fungi_norm) <- "matrix"
tidy_data_PO_fungi_norm <- t(tidy_data_PO_fungi_norm)
curve_stand <- rarecurve(tidy_data_PO_fungi_norm, step=50, label=F)
# save image to rarecurves_fungi_norm_2024-09-23
```

### protists

#### Version raw

```{r}
# otu_table = tidy_data
## select the community
tidy_data_PO_protists <- tidy_data_PO %>% filter(community %in% c("protists"))

## remove useless columns
tidy_data_PO_protists <- tidy_data_PO_protists[,c(1,2,13)]

## transform table to taxa_are_rows = TRUE
tidy_data_PO_protists <- pivot_wider(tidy_data_PO_protists, names_from = code_uhbrd, values_from = abundance)

## replace NA by 0
tidy_data_PO_protists[is.na(tidy_data_PO_protists)] <- 0

## use "finest_name" as rownames
tidy_data_PO_protists <- tidy_data_PO_protists %>% tibble::column_to_rownames("finest_name")

## transform tbl_df to matrix table
tidy_data_PO_protists <- as.matrix(tidy_data_PO_protists)

# tax_table = tax_table # + fun_table
## select the community
tax_table_PO_protists <- tax_table %>% filter(community %in% c("protists"))

## remove useless columns
tax_table_PO_protists <- tax_table_PO_protists[,-c(9)]

########## Add fun_table here ########## 
tax_table_PO_protists_fun <- left_join(tax_table_PO_protists, fun_table_PO, by = c("family" = "matchName"))

tax_table_PO_protists_fun[is.na(tax_table_PO_protists_fun)] <- "unassigned"

tax_table_PO_protists_fun_inter <- tax_table_PO_protists_fun %>%
  rowwise() %>%
  mutate(interaction = if_else(any(c_across(1:8) %in% fun_table_inter_PO$matchName), "observed", "not observed")) %>%
  ungroup()

tax_table_PO_protists <- tax_table_PO_protists_fun_inter
########## END ########## 

## use "finest_name" as rownames
tax_table_PO_protists <- tax_table_PO_protists %>% tibble::column_to_rownames("finest_name")

## transform tbl_df to matrix table
tax_table_PO_protists <- as.matrix(tax_table_PO_protists)

# sam_data = metadata
## select the community
metadata_PO_protists <- metadata_PO %>% filter(community %in% c("protists"))

## choose the rownames
rownames(metadata_PO_protists) <- metadata_PO_protists[,11] # code_uhbrd

# create phyloseq object
tidy_data_PO_protists = otu_table(tidy_data_PO_protists, taxa_are_rows = TRUE)

tax_table_PO_protists = tax_table(tax_table_PO_protists)

metadata_PO_protists = sample_data(metadata_PO_protists)
  
PO_protists <- phyloseq(tidy_data_PO_protists, tax_table_PO_protists, metadata_PO_protists)

# check PO
PO_protists
sample_names(PO_protists)
rank_names(PO_protists)
sample_variables(PO_protists)

#write_rds(PO_protists, "data/derived_data/phyloseq_objects/PO_protists_raw_2024-09-23.rds")
#PO_protists_df <- phyloseq_to_df(PO_protists, addtax = T, addtot = T, addmaxrank = F, sorting = "NULL")
#write.csv(PO_protists_df, "data/derived_data/phyloseq_objects/PO_protists_raw_2024-09-23.csv", row.names = FALSE)
```

#### Version stand

```{r}
# standardize abundances to the median sequencing depth
total = median(sample_sums(PO_protists))
standf = function(x, t=total) round(t*(x/sum(x)))
PO_protists_stand = transform_sample_counts(PO_protists, standf)

# check PO
PO_protists_stand

#write_rds(PO_protists_stand, "data/derived_data/phyloseq_objects/PO_protists_stand_2024-09-23.rds")
#PO_protists_df <- phyloseq_to_df(PO_protists_stand, addtax = T, addtot = T, addmaxrank = F, sorting = "NULL")
#write.table(PO_protists_df, "data/derived_data/phyloseq_objects/PO_protists_stand_2024-09-23.csv", row.names = FALSE)

propfun = function(x){
  xprop = (x / sum(x))
  return(xprop)
}
PO_protists_stand_prop <- transform_sample_counts(PO_protists_stand, fun = propfun)

#write_rds(PO_protists_stand_prop, "data/derived_data/phyloseq_objects/PO_protists_stand_prop_2024-09-23.rds")
#PO_protists_df <- phyloseq_to_df(PO_protists_stand_prop, addtax = T, addtot = T, addmaxrank = F, sorting = "NULL")
#write.table(PO_protists_df, "data/derived_data/phyloseq_objects/PO_protists_stand_prop_2024-09-23.csv", row.names = FALSE)
```

#### Version norm

```{r}
# normalize
PO_protists_norm <- normalize_prop_pq(PO_protists, base_log = 2, digits = 0)

# check PO
PO_protists_norm

#write_rds(PO_protists_norm, "data/derived_data/phyloseq_objects/PO_protists_norm_2024-09-23.rds")
#PO_protists_df <- phyloseq_to_df(PO_protists_norm, addtax = T, addtot = T, addmaxrank = F, sorting = "NULL")
#write.table(PO_protists_df, "data/derived_data/phyloseq_objects/PO_protists_norm_2024-09-23.csv", row.names = FALSE)

propfun = function(x){
  xprop = (x / sum(x))
  return(xprop)
}
PO_protists_norm_prop <- transform_sample_counts(PO_protists_norm, fun = propfun)

#write_rds(PO_protists_norm_prop, "data/derived_data/phyloseq_objects/PO_protists_norm_prop_2024-09-23.rds")
#PO_protists_df <- phyloseq_to_df(PO_protists_norm_prop, addtax = T, addtot = T, addmaxrank = F, sorting = "NULL")
#write.csv(PO_protists_df[,-1], "data/derived_data/phyloseq_objects/PO_protists_norm_prop_2024-09-23.csv", row.names = FALSE)
```

#### Pre-analyses

```{r}
tidy_data_PO_protists_before <- as.data.frame(t(tidy_data_PO_protists))
sample_names = rownames(tidy_data_PO_protists_before)
curve_before <- rarecurve(tidy_data_PO_protists_before, step = 50, label = F)
# save image to rarecurves_protists_raw_2024-09-23

tidy_data_PO_protists_stand <- otu_table(PO_protists_stand)
class(tidy_data_PO_protists_stand) <- "matrix"
tidy_data_PO_protists_stand <- t(tidy_data_PO_protists_stand)
curve_stand <- rarecurve(tidy_data_PO_protists_stand, step=50, label=F)
# save image to rarecurves_protists_stand_2024-09-23

tidy_data_PO_protists_norm <- otu_table(PO_protists_norm)
class(tidy_data_PO_protists_norm) <- "matrix"
tidy_data_PO_protists_norm <- t(tidy_data_PO_protists_norm)
curve_stand <- rarecurve(tidy_data_PO_protists_norm, step=50, label=F)
# save image to rarecurves_protists_norm_2024-09-23
```

### all

```{r}
PO_microorganisms_raw <- merge_phyloseq(PO_bacteria, PO_fungi, PO_protists)
PO_microorganisms_raw

#write_rds(PO_microorganisms_raw, "data/derived_data/phyloseq_objects/PO_microorganisms_raw_2024-09-23.rds")
#PO_microorganisms_df <-phyloseq_to_df(PO_microorganisms_raw, addtax = T, addtot = T, addmaxrank = F, sorting = "NULL")
#write.table(PO_microorganisms_df, "data/derived_data/phyloseq_objects/PO_microorganisms_raw_2024-09-23.csv", sep = ",")
```

```{r}
PO_microorganisms_stand <- merge_phyloseq(PO_bacteria_stand, PO_fungi_stand, PO_protists_stand)
PO_microorganisms_stand

#write_rds(PO_microorganisms_stand, "data/derived_data/phyloseq_objects/PO_microorganisms_stand_2024-09-23.rds")
#PO_microorganisms_df <-phyloseq_to_df(PO_microorganisms_stand, addtax = T, addtot = T, addmaxrank = F, sorting = "NULL")
#write.table(PO_microorganisms_df, "data/derived_data/phyloseq_objects/PO_microorganisms_stand_2024-09-23.csv", sep = ",")

PO_microorganisms_stand_prop <- merge_phyloseq(PO_bacteria_stand_prop, PO_fungi_stand_prop, PO_protists_stand_prop)
PO_microorganisms_stand_prop

#write_rds(PO_microorganisms_stand, "data/derived_data/phyloseq_objects/PO_microorganisms_stand_prop_2024-09-23.rds")
#PO_microorganisms_df <-phyloseq_to_df(PO_microorganisms_stand_prop, addtax = T, addtot = T, addmaxrank = F, sorting = "NULL")
#write.table(PO_microorganisms_df, "data/derived_data/phyloseq_objects/PO_microorganisms_stand_prop_2024-09-23.csv", sep = ",")
```

```{r}
PO_microorganisms_norm <- merge_phyloseq(PO_bacteria_norm, PO_fungi_norm, PO_protists_norm)
PO_microorganisms_norm

#write_rds(PO_microorganisms_norm, "data/derived_data/phyloseq_objects/PO_microorganisms_norm_2024-09-23.rds")
#PO_microorganisms_df <-phyloseq_to_df(PO_microorganisms_norm, addtax = T, addtot = T, addmaxrank = F, sorting = "NULL")
#write.table(PO_microorganisms_df, "data/derived_data/phyloseq_objects/PO_microorganisms_norm_2024-09-23.csv", sep = ",")

PO_microorganisms_norm_prop <- merge_phyloseq(PO_bacteria_norm_prop, PO_fungi_norm_prop, PO_protists_norm_prop)
PO_microorganisms_norm_prop

#write_rds(PO_microorganisms_norm_prop, "data/derived_data/phyloseq_objects/PO_microorganisms_norm_prop_2024-09-23.rds")
#PO_microorganisms_df <-phyloseq_to_df(PO_microorganisms_norm_prop, addtax = T, addtot = T, addmaxrank = F, sorting = "NULL")
#write.table(PO_microorganisms_df, "data/derived_data/phyloseq_objects/PO_microorganisms_norm_prop_2024-09-23.csv", sep = ",")
```

#### Pre-analyses

```{r}
tidy_data_PO_microorganisms_before <- otu_table(PO_microorganisms_raw)
class(tidy_data_PO_microorganisms_before) <- "matrix"
tidy_data_PO_microorganisms_before <- t(tidy_data_PO_microorganisms_before)
curve_before <- rarecurve(tidy_data_PO_microorganisms_before, step = 50, label = F)
# save image to rarecurves_microorganisms_raw_2024-09-23

tidy_data_PO_microorganisms_stand <- otu_table(PO_microorganisms_stand)
class(tidy_data_PO_microorganisms_stand) <- "matrix"
tidy_data_PO_microorganisms_stand <- t(tidy_data_PO_microorganisms_stand)
curve_stand <- rarecurve(tidy_data_PO_microorganisms_stand, step=50, label=F)
# save image to rarecurves_microorganisms_stand_2024-09-23

tidy_data_PO_microorganisms_norm <- otu_table(PO_microorganisms_norm)
class(tidy_data_PO_microorganisms_norm) <- "matrix"
tidy_data_PO_microorganisms_norm <- t(tidy_data_PO_microorganisms_norm)
curve_stand <- rarecurve(tidy_data_PO_microorganisms_norm, step=50, label=F)
# save image to rarecurves_microorganisms_norm_2024-09-23
```

```{r}
# Choose the version
PO <- PO_microorganisms # raw
PO <- PO_microorganisms_stand # this one used first
PO <- PO_microorganisms_stand_prop # this one if several communities/methods (microorganisms and all) in plot_bar
PO <- PO_microorganisms_norm
PO <- PO_microorganisms_norm_prop
PO <- PO_bioag
PO <- PO_auxil

# Filter date
PO <- subset_samples(PO, date == "17.04.2023")
```

```{r}
# Order levels
PO@sam_data$date <- factor(PO@sam_data$date, levels = order_date)
PO@sam_data$code_uh <- factor(PO@sam_data$code_uh, levels = order_uh)
PO@sam_data$spatial <- factor(PO@sam_data$spatial, levels = order_spatial)
```

```{r}
alpha_table <- estimate_richness(PO) # works only with integer
#write.table(alpha_table, "analyses/pre-analysis/alpha_microorganisms_stand_2024-09-23.csv", sep = ",")

PO_alpha <- merge(PO@sam_data, alpha_table, by = 0)
quickanova <- aov(Observed ~ spatial, PO_alpha)
summary(quickanova)
quickanova <- aov(Shannon ~ spatial, PO_alpha)
summary(quickanova)
quickanova <- aov(InvSimpson ~ spatial, PO_alpha)
summary(quickanova)
```

```{r}
observed <- plot_richness(PO, x="code_uh", shape="block", color="habitat", measures=c("Observed")) +
  geom_boxplot(shape="block") +
  geom_point(position = "jitter") +
  theme_bw() + frame_simple + scale_x_discrete(limits = order_uh) +
  theme(axis.text.x=element_text(angle = -90, hjust = 0)) +
  labs(title="Alpha diversity of microorganisms", subtitle="Taxonomy table", x ="land use_habitat", y="Observed richness") + facet_wrap(date ~ .)
observed$layers <- observed$layers[-1]

shannon <- plot_richness(PO, x="code_uh", shape="block", color="habitat", measures=c("Shannon")) +
  geom_boxplot(shape="block") +
  geom_point(position = "jitter") +
  theme_bw() + frame_simple + scale_x_discrete(limits = order_uh) +
  theme(axis.text.x=element_text(angle = -90, hjust = 0)) +
  labs(title="Alpha diversity of microorganisms", subtitle="Taxonomy table", x ="land use_habitat", y="Shannon index") + facet_wrap(date ~ .)
shannon$layers <- shannon$layers[-1]

simpson <- plot_richness(PO, x="code_uh", shape="block", color="habitat", measures=c("InvSimpson")) +
  geom_boxplot(shape="block") +
  geom_point(position = "jitter") +
  theme_bw() + frame_simple + scale_x_discrete(limits = order_uh) +
  theme(axis.text.x=element_text(angle = -90, hjust = 0)) +
  labs(title="Alpha diversity of microorganisms", subtitle="Taxonomy table", x ="land use_habitat", y="Simpson inverse index") + facet_wrap(date ~ .)
simpson$layers <- simpson$layers[-1]

alpha_graph <- plot_grid(observed, shannon, simpson, ncol=1)
alpha_graph

#ggsave("analyses/pre-analysis/alpha_microorganisms_stand_uh_2024-09-23.png", width=3.5, height=12)
```

```{r}
PO_ord <- ordinate(PO, "PCoA", distance = "bray")
ord <- plot_ordination(PO, PO_ord, color = "code_uh", shape = "block") + theme_bw() + labs(title="Beta diversity of microorganisms", subtitle="Taxonomy table")
ord

#ggsave("analyses/pre-analysis/beta_microorganisms_plot_ordination_stand_uhb_2024-09-23.png", width=5, height=4)
```

```{r}
#ord_explore(PO)

PO %>%
 tax_transform(rank = "unique", trans = "identity") %>%
 dist_calc(dist = "bray") %>%
 ord_calc(
  method = "PCoA"
 ) %>% 
 ord_plot(
  axes = c(1, 2),
  colour = "date", fill = "date",
  shape = "habitat", alpha = 0.5,
  size = 2
 ) + 
 scale_shape_girafe_filled() +
 ggplot2::stat_ellipse(
  ggplot2::aes(colour = date)
 )

#ggsave("analyses/pre-analysis/beta_microorganisms_microViz_stand_hd_2024-09-23.png", width=5, height=4, bg = "white")
```

```{r}
# Filter before to continue here

observed <- plot_richness(PO, x="habitat", shape="block", color="habitat", measures=c("Observed")) +
  geom_boxplot(shape="block") +
  geom_point(position = "jitter") +
  theme_bw() + frame_simple +
  labs(title="Alpha diversity of microorganisms", subtitle="Taxonomy table", x =" ", y="Observed richness") + facet_wrap(spatial ~ .)
observed$layers <- observed$layers[-1]

shannon <- plot_richness(PO, x="habitat", shape="block", color="habitat", measures=c("Shannon")) +
  geom_boxplot(shape="block") +
  geom_point(position = "jitter") +
  theme_bw() + frame_simple +
  labs(title="Alpha diversity of microorganisms", subtitle="Taxonomy table", x =" ", y="Shannon index") + facet_wrap(spatial ~ .)
shannon$layers <- shannon$layers[-1]

simpson <- plot_richness(PO, x="habitat", shape="block", color="habitat", measures=c("InvSimpson")) +
  geom_boxplot(shape="block") +
  geom_point(position = "jitter") +
  theme_bw() + frame_simple +
  labs(title="Alpha diversity of microorganisms", subtitle="Taxonomy table", x =" ", y="Simpson inverse index") + facet_wrap(spatial ~ .)
simpson$layers <- simpson$layers[-1]

alpha_graph <- plot_grid(observed, shannon, simpson, ncol=1)
alpha_graph

#ggsave("analyses/pre-analysis/alpha_microorganisms_stand_filterdate_spatial_2024-09-23.png", width=4, height=10)
#ggsave("analyses/pre-analysis/alpha_microorganisms_stand_filterdate_spatial_bioag_2024-09-23.png", width=4, height=10)
#ggsave("analyses/pre-analysis/alpha_microorganisms_stand_filterdate_spatial_auxil_2024-09-23.png", width=4, height=10)
```

```{r}
PO_ord <- ordinate(PO, "PCoA", distance = "bray")
ord <- plot_ordination(PO, PO_ord, color = "spatial", shape = "block") + theme_bw() + labs(title="Beta diversity of microorganisms", subtitle="Taxonomy table")
ord

#ggsave("analyses/pre-analysis/beta_microorganisms_plot_ordination_stand_spatial_2024-09-23.png", width=5, height=4)
```

```{r}
#ord_explore(PO)

PO %>%
 tax_transform(rank = "unique", trans = "identity") %>%
 dist_calc(dist = "bray") %>%
 ord_calc(
  method = "PCoA"
 ) %>% 
 ord_plot(
  axes = c(1, 2),
  colour = "spatial", fill = "spatial",
  shape = "habitat", alpha = 0.5,
  size = 2
 ) + 
 scale_shape_girafe_filled() +
 ggplot2::stat_ellipse(
  ggplot2::aes(colour = spatial)
 )

#ggsave("analyses/pre-analysis/beta_microorganisms_microViz_stand_spatial_2024-09-23.png", width=5, height=4, bg = "white")
```

```{r}
adonis_pq(PO, "spatial", na_remove = TRUE)
adonis_pq(PO, "spatial*habitat", na_remove = TRUE)
adonis_pq(PO, "land_use*habitat", na_remove = TRUE)
adonis_pq(PO, "code_uh", na_remove = TRUE)
adonis_pq(PO, "code_uh*block", na_remove = TRUE)
adonis_pq(PO, "code_uh*replicate", na_remove = TRUE)
adonis_pq(PO, "block", na_remove = TRUE)
```

```{r}
# Chose PO_prop

#topOTU = names(sort(taxa_sums(PO), TRUE)[1:100])
#PO_top = prune_taxa(topOTU, PO)
PO_top <- PO # if no pruning
PO_top_uh = merge_samples(PO_top, "code_uh")
PO_top_uh_phylum = tax_glom(PO_top_uh, "phylum")
#abund_table <- psmelt(PO_top_uh_phylum)
#abund_table$phylum <- as.character(abund_table$phylum)
#abund_table$phylum[abund_table$abundance < 0.01] <- "abundunce < 1%"

PO_top_uh_phylum_relabun = transform_sample_counts(PO_top_uh_phylum, function(x) x/sum(x) * 100)

phylum_rel <- plot_bar(PO_top_uh_phylum_relabun, fill="phylum") + geom_col(stat="identity", position="stack", color = "black") + frame_simple + scale_x_discrete(limits = order_uh) + labs(title="Abundance of microorganisms at phylum level ", subtitle="Total abundance", x ="land use_habitat") + scale_y_continuous(labels = scales::comma) + guides(fill=guide_legend(ncol=3))
phylum_rel

#ggsave("analyses/pre-analysis/abund_phylum_microorganisms_stand_prop_uh_2024-09-23.png", width=7, height=6)
```

```{r}
# Chose PO_prop

#topOTU = names(sort(taxa_sums(PO), TRUE)[1:100])
#PO_top = prune_taxa(topOTU, PO)
PO_top <- PO # if no pruning
PO_top_spatial = merge_samples(PO_top, "spatial")
PO_top_spatial_phylum = tax_glom(PO_top_spatial, "phylum")
#abund_table <- psmelt(PO_top_spatial_phylum)
#abund_table$phylum <- as.character(abund_table$phylum)
#abund_table$phylum[abund_table$abundance < 0.01] <- "abundunce < 1%"

PO_top_spatial_phylum_relabun = transform_sample_counts(PO_top_spatial_phylum, function(x) x/sum(x) * 100)

phylum_rel <- plot_bar(PO_top_spatial_phylum_relabun, fill="phylum") + geom_col(stat="identity", position="stack", color = "black") + frame_simple + scale_x_discrete(limits = order_spatial) + labs(title="Abundance of microorganisms at phylum level ", subtitle="Total abundance", x ="land use_habitat") + scale_y_continuous(labels = scales::comma) + guides(fill=guide_legend(ncol=3))
phylum_rel

#ggsave("analyses/pre-analysis/abund_phylum_microorganisms_stand_prop_spatial_2024-09-23.png", width=6.5, height=6)
```


```{r}
# Filter functions
PO_bioag <- subset_taxa(PO, grepl("her|pla", abr_all_guild_level_AND))
PO_bioag_obs <- subset_taxa(PO_bioag, interaction == "observed")

PO_auxil <- subset_taxa(PO, grepl("pre|par", abr_all_guild_level_AND))
PO_auxil_obs <- subset_taxa(PO_auxil, interaction == "observed")
```

```{r}
PO_top <- PO_bioag
PO_top_spatial = merge_samples(PO_top, "spatial")
PO_top_spatial_family = tax_glom(PO_top_spatial, "family")

family_tot_bioag <- plot_bar(PO_top_spatial_family, fill="family") + geom_col(stat="identity", position="stack", color = "black") + frame_simple + labs(title="Abundance of microorganisms at family level ", subtitle="Total abundance of potential bioagressors", x ="Spatial heterogeneity") + scale_y_continuous(labels = scales::comma)  + guides(fill=guide_legend(ncol=3))
family_tot_bioag

PO_top <- PO_auxil
PO_top_spatial = merge_samples(PO_top, "spatial")
PO_top_spatial_family = tax_glom(PO_top_spatial, "family")

family_tot_auxil <- plot_bar(PO_top_spatial_family, fill="family") + geom_col(stat="identity", position="stack", color = "black") + frame_simple + labs(title="Abundance of microorganisms at family level ", subtitle="Total abundance of potential auxiliaries", x ="Spatial heterogeneity") + scale_y_continuous(labels = scales::comma)  + guides(fill=guide_legend(ncol=3))
family_tot_auxil

abund_tot_family <- plot_grid(family_tot_bioag, family_tot_auxil, ncol=1)
abund_tot_family

#ggsave("analyses/pre-analysis/abund_tot_family_microorganisms_stand_prop_spatial_bioagANDauxil_2024-09-23.png", width=6, height=9)
```

```{r}
PO_top <- PO_bioag_obs
PO_top_spatial = merge_samples(PO_top, "spatial")
PO_top_spatial_family = tax_glom(PO_top_spatial, "family")

family_tot_bioag <- plot_bar(PO_top_spatial_family, fill="family") + geom_col(stat="identity", position="stack", color = "black") + frame_simple + labs(title="Abundance of microorganisms at family level ", subtitle="Total abundance of potential bioagressors with observed interactions", x ="Spatial heterogeneity") + scale_y_continuous(labels = scales::comma)  + guides(fill=guide_legend(ncol=1)) + scale_x_discrete(limits = order_spatial) #+ fill_manual_family
family_tot_bioag

PO_top <- PO_auxil_obs
PO_top_spatial = merge_samples(PO_top, "spatial")
PO_top_spatial_family = tax_glom(PO_top_spatial, "family")

family_tot_auxil <- plot_bar(PO_top_spatial_family, fill="family") + geom_col(stat="identity", position="stack", color = "black") + frame_simple + labs(title="Abundance of microorganisms at family level ", subtitle="Total abundance of potential auxiliaries with observed interactions", x ="Spatial heterogeneity") + scale_y_continuous(labels = scales::comma)  + guides(fill=guide_legend(ncol=1)) + scale_x_discrete(limits = order_spatial) #+ fill_manual_family
family_tot_auxil

abund_tot_family <- plot_grid(family_tot_bioag, family_tot_auxil, ncol=1)
abund_tot_family

#ggsave("analyses/pre-analysis/abund_tot_family_microorganisms_stand_spatial_bioagANDauxil_obs_2024-09-23.png", width=5, height=9)
```

```{r}
list_PO_bioag_obs <- PO_bioag_obs@tax_table
#write.csv(list_PO_bioag_obs, "analyses/pre-analysis/list_microorganisms_bioag_obs_2024-09-23.csv", row.names = T)

list_PO_auxil_obs <- PO_auxil_obs@tax_table
#write.csv(list_PO_auxil_obs, "analyses/pre-analysis/list_microorganisms_auxil_obs_2024-09-23.csv", row.names = T)
```

```{r}
V_spatial_bioag <- ggvenn_pq(PO_bioag, fact = "spatial", min_nb_seq = 0, taxonomic_rank = "family", add_nb_samples = F, set_size = 3) + labs(title = "Number of potential bioagressors at family level") # split_by = "date" # label = "count"
V_spatial_bioag

#save manually to ggvenn_microorganisms_spatial_bioag_family_2024-09-23

V_spatial_auxil <- ggvenn_pq(PO_auxil, fact = "spatial", min_nb_seq = 0, taxonomic_rank = "family", add_nb_samples = F, set_size = 3) + labs(title = "Number of potential auxiliaries at family level") # split_by = "date" # label = "count"
V_spatial_auxil

#save manually to ggvenn_microorganisms_spatial_auxil_family_2024-09-23
```


## all

### Version stand

```{r}
PO_macrofauna_surface_stand <- read_rds("data/derived_data/phyloseq_objects/PO_macrofauna_surface_stand_2024-09-04.rds")
PO_macrofauna_aerial_stand <- read_rds("data/derived_data/phyloseq_objects/PO_macrofauna_aerial_stand_2024-09-04.rds")
PO_macrofauna_foliar_stand <- read_rds("data/derived_data/phyloseq_objects/PO_macrofauna_foliar_stand_2024-09-04.rds")
PO_nematodes_stand <- read_rds("data/derived_data/phyloseq_objects/PO_nematodes_stand_2024-09-04.rds")
PO_micro_arthropodes_stand <- read_rds("data/derived_data/phyloseq_objects/PO_micro_arthropodes_stand_2024-09-04.rds")
PO_microorganisms_stand <- read_rds("data/derived_data/phyloseq_objects/PO_microorganisms_stand_2024-09-04.rds")

PO_all_stand <- merge_phyloseq(PO_macrofauna_surface_stand, PO_macrofauna_aerial_stand, PO_macrofauna_foliar_stand, PO_nematodes_stand, PO_micro_arthropodes_stand, PO_microorganisms_stand)

check <- verify_pq(PO_all_stand)
PO_all_stand

summary_plot_pq(PO_all_stand, add_info = T)

#save manually to summary_plot_all_stand_2024-09-04

#write_rds(PO_all_stand, "data/derived_data/phyloseq_objects/PO_all_stand_2024-09-04.rds")
#PO_all_df <-phyloseq_to_df(PO_all_stand, addtax = T, addtot = T, addmaxrank = F, sorting = "NULL")
#write.table(PO_all_df, "data/derived_data/phyloseq_objects/PO_all_stand_2024-09-04.csv", sep = ",")
```

### Version norm

```{r}
PO_macrofauna_surface_norm <- read_rds("data/derived_data/phyloseq_objects/PO_macrofauna_surface_norm_2024-09-04.rds")
PO_macrofauna_aerial_norm <- read_rds("data/derived_data/phyloseq_objects/PO_macrofauna_aerial_norm_2024-09-04.rds")
PO_macrofauna_foliar_norm <- read_rds("data/derived_data/phyloseq_objects/PO_macrofauna_foliar_norm_2024-09-04.rds")
PO_nematodes_norm <- read_rds("data/derived_data/phyloseq_objects/PO_nematodes_norm_2024-09-04.rds")
PO_micro_arthropodes_norm <- read_rds("data/derived_data/phyloseq_objects/PO_micro_arthropodes_norm_2024-09-04.rds")
PO_microorganisms_norm <- read_rds("data/derived_data/phyloseq_objects/PO_microorganisms_norm_2024-09-04.rds")

PO_all_norm <- merge_phyloseq(PO_macrofauna_surface_norm, PO_macrofauna_aerial_norm, PO_macrofauna_foliar_norm, PO_nematodes_norm, PO_micro_arthropodes_norm, PO_microorganisms_norm)

check <- verify_pq(PO_all_norm)
PO_all_norm

summary_plot_pq(PO_all_norm, add_info = T)

#save manually to summary_plot_all_norm_2024-09-04

#write_rds(PO_all_norm, "data/derived_data/phyloseq_objects/PO_all_norm_2024-09-04.rds")
#PO_all_df <-phyloseq_to_df(PO_all_norm, addtax = T, addtot = T, addmaxrank = F, sorting = "NULL")
#write.table(PO_all_df, "data/derived_data/phyloseq_objects/PO_all_norm_2024-09-04.csv", sep = ",")
```

### Version norm_prop

```{r}
PO_macrofauna_surface_norm_prop <- read_rds("data/derived_data/phyloseq_objects/PO_macrofauna_surface_norm_prop_2024-09-04.rds")
PO_macrofauna_aerial_norm_prop <- read_rds("data/derived_data/phyloseq_objects/PO_macrofauna_aerial_norm_prop_2024-09-04.rds")
PO_macrofauna_foliar_norm_prop <- read_rds("data/derived_data/phyloseq_objects/PO_macrofauna_foliar_norm_prop_2024-09-04.rds")
PO_nematodes_norm_prop <- read_rds("data/derived_data/phyloseq_objects/PO_nematodes_norm_prop_2024-09-04.rds")
PO_micro_arthropodes_norm_prop <- read_rds("data/derived_data/phyloseq_objects/PO_micro_arthropodes_norm_prop_2024-09-04.rds")
PO_microorganisms_norm_prop <- read_rds("data/derived_data/phyloseq_objects/PO_microorganisms_norm_prop_2024-09-04.rds")

PO_all_norm_prop <- merge_phyloseq(PO_macrofauna_surface_norm_prop, PO_macrofauna_aerial_norm_prop, PO_macrofauna_foliar_norm_prop, PO_nematodes_norm_prop, PO_micro_arthropodes_norm_prop, PO_microorganisms_norm_prop)

check <- verify_pq(PO_all_norm_prop)
PO_all_norm_prop

summary_plot_pq(PO_all_norm_prop, add_info = T)

#save manually to summary_plot_all_norm_prop_2024-09-04

#write_rds(PO_all_norm_prop, "data/derived_data/phyloseq_objects/PO_all_norm_prop_2024-09-04.rds")
#PO_all_df <-phyloseq_to_df(PO_all_norm_prop, addtax = T, addtot = T, addmaxrank = F, sorting = "NULL")
#write.table(PO_all_df, "data/derived_data/phyloseq_objects/PO_all_norm_prop_2024-09-04.csv", sep = ",")
```

```{r}
tidy_data_PO_all_stand <- otu_table(PO_all_stand)
class(tidy_data_PO_all_stand) <- "matrix"
tidy_data_PO_all_stand <- t(tidy_data_PO_all_stand)
curve_stand <- rarecurve(tidy_data_PO_all_stand, step=50, label=F)
#save manually to rarecurve_all_stand_2024-09-04

tidy_data_PO_all_norm <- otu_table(PO_all_norm)
class(tidy_data_PO_all_norm) <- "matrix"
tidy_data_PO_all_norm <- t(tidy_data_PO_all_norm)
curve_stand <- rarecurve(tidy_data_PO_all_norm, step=50, label=F)
#save manually to rarecurve_all_norm_prop_2024-09-04
```

### Subset

```{r}
PO_all_stand_sub <- subset_samples(PO_all_stand, date == "17.04.2023")
PO_all_norm_sub <- subset_samples(PO_all_norm, date == "17.04.2023")
```

### Pre-analyses

```{r}
alpha_table <- estimate_richness(PO_all_stand_sub)
#write.table(alpha_table, "analyses/pre-analysis/alpha_all_2024-09-04.csv", sep = ",")

PO_all_stand_sub@sam_data$date <- factor(PO_all_stand_sub@sam_data$date, levels = order_date)
```

```{r}
observed <- plot_richness(PO_all_stand_sub, x="code_uh", shape="block", color="habitat", measures=c("Observed")) +
  geom_boxplot(shape="block") +
  geom_point(position = "jitter") +
  theme_bw() + frame_simple + scale_x_discrete(limits = order_uh) +
  theme(axis.text.x=element_text(angle = -90, hjust = 0)) +
  labs(title="Alpha diversity of all", subtitle="Taxonomy table", x ="land use_habitat", y="Observed richness") #+ facet_wrap(date ~ .)
observed$layers <- observed$layers[-1]

shannon <- plot_richness(PO_all_stand_sub, x="code_uh", shape="block", color="habitat", measures=c("Shannon")) +
  geom_boxplot(shape="block") +
  geom_point(position = "jitter") +
  theme_bw() + frame_simple + scale_x_discrete(limits = order_uh) +
  theme(axis.text.x=element_text(angle = -90, hjust = 0)) +
  labs(title="Alpha diversity of all", subtitle="Taxonomy table", x ="land use_habitat", y="Shannon index") #+ facet_wrap(date ~ .)
shannon$layers <- shannon$layers[-1]

simpson <- plot_richness(PO_all_stand_sub, x="code_uh", shape="block", color="habitat", measures=c("InvSimpson")) +
  geom_boxplot(shape="block") +
  geom_point(position = "jitter") +
  theme_bw() + frame_simple + scale_x_discrete(limits = order_uh) +
  theme(axis.text.x=element_text(angle = -90, hjust = 0)) +
  labs(title="Alpha diversity of all", subtitle="Taxonomy table", x ="land use_habitat", y="Simpson inverse index") #+ facet_wrap(date ~ .)
simpson$layers <- simpson$layers[-1]

alpha_graph <- plot_grid(observed, shannon, simpson, ncol=1)

##ggsave("analyses/pre-analysis/alpha_all_2024-09-04.png", width=4, height=10)
```

```{r}
H <- hill_tuckey_pq(PO_all_stand_sub, "code_uh")
H

#ggsave("analyses/pre-analysis/alpha_all_H_2024-09-04.png", width=6, height=4)
```

```{r}
PO_ord <- ordinate(PO_all_stand_sub, "PCoA", distance = "bray")
ord <- plot_ordination(PO_all_stand_sub, PO_ord, color = "code_uh", shape = "block") + theme_bw() + labs(title="Beta diversity of all", subtitle="Taxonomy table")
ord

#ggsave("analyses/pre-analysis/beta_all_plot_ordination_2024-09-04.png", width=5, height=4)
```

```{r}
#ord_explore(PO_all_stand_sub)

PO_all_stand_sub %>%
 tax_transform(rank = "unique", trans = "identity") %>%
 dist_calc(dist = "bray") %>%
 ord_calc(
  method = "PCoA"
 ) %>% 
 ord_plot(
  axes = c(1, 2),
  colour = "code_uh", fill = "code_uh",
  shape = "habitat", alpha = 0.5,
  size = 2
 ) + 
 scale_shape_girafe_filled() +
 ggplot2::stat_ellipse(
  ggplot2::aes(colour = code_uh)
 )

#ggsave("analyses/pre-analysis/beta_all_microViz_2024-09-04_V1.png", width=5, height=4, bg = "white")

PO_all_stand_sub %>%
 tax_transform(rank = "unique", trans = "identity") %>%
 dist_calc(dist = "bray") %>%
 ord_calc(
  method = "PCoA"
 ) %>% 
 ord_plot(
  axes = c(1, 2),
  colour = "replicate", fill = "replicate",
  shape = "habitat", alpha = 0.5,
  size = 2
 ) + 
 scale_shape_girafe_filled() +
 ggplot2::stat_ellipse(
  ggplot2::aes(colour = replicate)
 )

#ggsave("analyses/pre-analysis/beta_all_microViz_2024-09-04_V2.png", width=5, height=4, bg = "white")
```

```{r}
adonis_pq(PO_all_stand_sub, "land_use*habitat", na_remove = TRUE)
adonis_pq(PO_all_stand_sub, "code_uh", na_remove = TRUE)
adonis_pq(PO_all_stand_sub, "code_uh*block", na_remove = TRUE)
adonis_pq(PO_all_stand_sub, "code_uh*replicate", na_remove = TRUE)

PO_all_stand_sub_clr <- microbiome::transform(PO_all_stand_sub, "clr")
ps_dist_matrix <- phyloseq::distance(PO_all_stand_sub_clr, method ="bray")
vegan::adonis2(ps_dist_matrix ~ phyloseq::sample_data(PO_all_stand_sub)$code_uh)

pw_adonis <- pairwise.adonis(ps_dist_matrix, phyloseq::sample_data(PO_all_stand_sub)$code_uh)

write.table(pw_adonis, "analyses/pre-analysis/pw_adonis_all_2024-09-04.csv", sep = ",")
```

```{r}
upset_pq(PO_all_stand_sub, fact = "code_uh", taxa_fill = "kingdom")

#save manually to upset_all_kingdom_2024-09-04
```

```{r}
#topOTU = names(sort(taxa_sums(PO_all_stand_sub), TRUE)[1:100])
#PO_all_stand_sub_top = prune_taxa(topOTU, PO_all_stand_sub)
PO_all_stand_sub_top <- PO_all_stand_sub # if no pruning
PO_all_stand_sub_top_uh = merge_samples(PO_all_stand_sub_top, "code_uh")
PO_all_stand_sub_top_uh_kingdom = tax_glom(PO_all_stand_sub_top_uh, "kingdom")
#abund_table <- psmelt(PO_all_stand_sub_top_uh_kingdom)
#abund_table$kingdom <- as.character(abund_table$kingdom)
#abund_table$kingdom[abund_table$abundance < 0.01] <- "abundunce < 1%" 

kingdom_tot <- plot_bar(PO_all_stand_sub_top_uh_kingdom, fill="kingdom") + geom_col(stat="identity", position="stack", color = "black") + frame_simple + scale_x_discrete(limits = order_uh) + labs(title="Abundance of all at kingdom level ", subtitle="Total abundance", x ="land use_habitat") + scale_y_continuous(labels = scales::comma)  + guides(fill=guide_legend(ncol=1))

PO_all_stand_sub_top_uh_kingdom_relabun = transform_sample_counts(PO_all_stand_sub_top_uh_kingdom, function(x) x/sum(x) * 100)

kingdom_rel <- plot_bar(PO_all_stand_sub_top_uh_kingdom_relabun, fill="kingdom") + geom_col(stat="identity", position="stack", color = "black") + frame_simple + scale_x_discrete(limits = order_uh) + labs(title="Abundance of all at kingdom level ", subtitle="Relative abundance", x ="land use_habitat") + scale_y_continuous(labels = scales::comma) + guides(fill=guide_legend(ncol=1))

abund_kingdom <- plot_grid(kingdom_tot, kingdom_rel, ncol=1)

#ggsave("analyses/pre-analysis/abund_kingdom_all_2024-09-04.png", width=4, height=8)
```

```{r}
#topOTU = names(sort(taxa_sums(PO_all_stand_sub), TRUE)[1:100])
#PO_all_stand_sub_top = prune_taxa(topOTU, PO_all_stand_sub)
PO_all_stand_sub_top <- PO_all_stand_sub # if no pruning
PO_all_stand_sub_top_uh = merge_samples(PO_all_stand_sub_top, "code_uh")
PO_all_stand_sub_top_uh_phylum = tax_glom(PO_all_stand_sub_top_uh, "phylum")
#abund_table <- psmelt(PO_all_stand_sub_top_uh_phylum)
#abund_table$phylum <- as.character(abund_table$phylum)
#abund_table$phylum[abund_table$abundance < 0.01] <- "abundunce < 1%" 

phylum_tot <- plot_bar(PO_all_stand_sub_top_uh_phylum, fill="phylum") + geom_col(stat="identity", position="stack", color = "black") + frame_simple + scale_x_discrete(limits = order_uh) + labs(title="Abundance of all at phylum level ", subtitle="Total abundance", x ="land use_habitat") + scale_y_continuous(labels = scales::comma)  + guides(fill=guide_legend(ncol=2))

PO_all_stand_sub_top_uh_phylum_relabun = transform_sample_counts(PO_all_stand_sub_top_uh_phylum, function(x) x/sum(x) * 100)

phylum_rel <- plot_bar(PO_all_stand_sub_top_uh_phylum_relabun, fill="phylum") + geom_col(stat="identity", position="stack", color = "black") + frame_simple + scale_x_discrete(limits = order_uh) + labs(title="Abundance of all at phylum level ", subtitle="Relative abundance", x ="land use_habitat") + scale_y_continuous(labels = scales::comma) + guides(fill=guide_legend(ncol=2))

abund_phylum <- plot_grid(phylum_tot, phylum_rel, ncol=1)

#ggsave("analyses/pre-analysis/abund_phylum_all_2024-09-04.png", width=6, height=10)
```

```{r}
V_uh <- ggvenn_pq(PO_all_stand_sub, fact = "code_uh", min_nb_seq = 0, taxonomic_rank = "family", add_nb_samples = F, set_size = 3, label = "count") + labs(title = "family") # split_by = "date"
V_uh

#save manually to ggvenn_all_uh_family_2024-09-04

V_h <- ggvenn_pq(PO_all_stand_sub, fact = "habitat", min_nb_seq = 0, taxonomic_rank = "family", add_nb_samples = F, set_size = 3, label = "count") + labs(title = "family") # split_by = "date"
V_h

#save manually to ggvenn_all_h_family_2024-09-04
```

```{r}
plot_edgeR_pq(PO_all_stand_sub, c("habitat", "T", "C"), color_tax = "kingdom", taxolev = "phylum")

#ggsave("analyses/pre-analysis/edgeR_all_2024-09-04.png", width=6, height=10)
```

```{r}
if (requireNamespace("mia")) {
  PO_all_stand_sub@tax_table <- phyloseq::tax_table(cbind(
    PO_all_stand_sub@tax_table,
    "taxon" = taxa_names(PO_all_stand_sub)
  ))

  res_time <- ancombc_pq(
    PO_all_stand_sub,
    fact = "habitat",
    levels_fact = c("C", "T"),
    tax_level = "taxon",
    verbose = TRUE
  )

  plot_ancombc_pq(PO_all_stand_sub, res_time,
    filter_passed = FALSE,
    tax_label = "genus", tax_col = "order"
  )
  plot_ancombc_pq(PO_all_stand_sub, res_time, tax_col = "genus")
  plot_ancombc_pq(PO_all_stand_sub, res_time,
    filter_passed = FALSE,
    filter_diff = FALSE, tax_col = "family", add_label = FALSE
  )
}
```

## Faprotax

```{r}
# Alternatively, you can use the functional assignations from Faprotax without tax_table

## samples_df = metadata
metadata <- metadata_mo_1 # cf 0. to import the initial metadata_mo_0 and 1. microorganisms to prepare metadata_mo_1
metadata$code_uh <- paste0(metadata$land_use, "_", metadata$habitat)
metadata <- metadata %>% tibble::column_to_rownames("code_uhbr")
metadata <- metadata[,-1]

## otu_mat = tidy_data
tidy_data_faprotax <- read_tsv("FAPROTAX_1.2.10/func_table3.tsv")
tidy_data_faprotax <- tidy_data_faprotax %>% tibble::column_to_rownames("group")
tidy_data_faprotax <- as.matrix(tidy_data_faprotax)

# Create phyloseq object
OTU = otu_table(tidy_data_faprotax, taxa_are_rows = TRUE)
samples = sample_data(metadata)
PO_Faprotax <- phyloseq(OTU, samples)

total = median(sample_sums(PO_Faprotax))
standf = function(x, t=total) round(t*(x/sum(x)))
PO_Faprotax = transform_sample_counts(PO_Faprotax, standf)

PO_Faprotax
sample_names(PO_Faprotax)
sample_variables(PO_Faprotax)
```

```{r}
OTU_curve = as.data.frame(t(OTU))
sample_names = rownames(OTU_curve)
otu.rarecurve = rarecurve(OTU_curve, step = 1000, label = T)
```

```{r}
alpha <- plot_richness(PO_Faprotax, x="code_uh", measures=c("Observed", "Shannon", "Simpson")) + geom_boxplot() + geom_point(size=1, alpha=0.7) + theme_bw() + frame_simple + scale_x_discrete(limits = order_uh) + theme(title =element_text(size=8, face='bold'), axis.text.x=element_text(angle = -90, hjust = 0)) + labs(title="Alpha diversity of bacteria", subtitle="Functional table (Faprotax)", x ="land use_habitat")
alpha

#ggsave("analyses/pre-analysis/alpha_Faprotax_2024-08-23.png", width=6, height=4)
```

```{r}
PO_ord <- ordinate(PO_Faprotax, "PCoA", distance = "bray")
ord <- plot_ordination(PO_Faprotax, PO_ord, color = "code_uh", shape = "block") + theme_bw() + labs(title="Beta diversity of bacteria", subtitle="Functional table (Faprotax)")
ord

#ggsave("analyses/pre-analysis/ord_Faprotax_2024-08-23.png", width=6, height=5)
```

## Gratin

```{r}
# Alternatively, you can use the functional assignations from Gratin
#fun_table <- read.csv()
```

Ref normalisation
https://adrientaudiere.github.io/MiscMetabar/reference/normalize_prop_pq.html
https://datadryad.org/stash/dataset/doi:10.5061/dryad.tn8qs35
https://scienceparkstudygroup.github.io/microbiome-lesson/05-data-filtering-and-normalisation/index.html