---
title: "Create tidy_data, tax_table and fun_table"
output: html_document
date: "2024-02-15"
---

# 0. Load packages and import data

```{r warning=FALSE}
library(readr)
library(dplyr)
library(tidyr)
library(readxl)
library(janitor)
library(stringr)
library(rebus)
library(taxize)
library(rgbif)
library(GratinNavigatoR)
```

suffix "_0" = raw data (as imported)

```{r message=FALSE, warning=FALSE}
macrofauna_surface_0 <- read_delim("data/1. barber_taxo2023-09-26.csv", show_col_types = FALSE)[,-1]

macrofauna_aerial_0 <- read_csv("data/1. pantrap_taxo2023-09-08.csv", show_col_types = FALSE)[,-1]

macrofauna_foliar_0 <- read_csv("data/clean_data_robinier_2023-09-16.csv", show_col_types =  FALSE)[,-1]

micro_arthropodes_all_0 <- read_excel("data/NH_Tri_mesofaune - INRAE MONTPELLER_AM_MH.xlsx")[,-c(11,12)]
micro_arthropodes_col_0 <- read_excel("data/BDD Collemboles - INRAE MONTPELLIER.xlsx")

nematodes_0 <- read_excel("data/raw data march april 2023.xlsx")

bacteria_0 <- read_delim("data/OTU_Table_Bacterie.tsv")[,c(0:72)]

fungi_0 <- read_delim("data/OTU_Table_Champignon.tsv")[,c(0:72)]

protists_0 <- read_delim("data/OTU_Table_Cercozoa.tsv")[,c(0:72)]

metadata_mo_0 <- read_delim("data/mapping_brut.csv")
```

# 1. Tidy data

Files generated in this section:
prefix "tidy_"
suffix "_1" = homogenized data (at the end of the chunck)
suffix "_2" = data with minimal information (almost ready to export)
suffix "_ALL" = all data combined (all communities)

Minimal information we want in the final tidy data files: abundance, name (of individual), code_uhbr
code_uhbr = short code including usage, habitat, block and replicate abbreviations
Additional information: date, method, community

## macrofauna_surface

```{r}
tidy_macrofauna_surface_1 <- macrofauna_surface_0

# rename columns
tidy_macrofauna_surface_1 <- tidy_macrofauna_surface_1 %>%  rename(block = bloc, replicate = rep)

# add column method, date and community
tidy_macrofauna_surface_1 <- tidy_macrofauna_surface_1 %>% mutate(method = "barber_trap")
tidy_macrofauna_surface_1 <- tidy_macrofauna_surface_1 %>% mutate(community = "macrofauna_surface")
tidy_macrofauna_surface_1 <- tidy_macrofauna_surface_1 %>%
  mutate(date = case_when(
      grepl(number_range(229, 288), id_sample) ~ "03/2023",
      grepl(number_range(349, 408), id_sample) ~ "04/2023",
      grepl(number_range(679, 738), id_sample) ~ "05/2023"))
#id_sample ANI-23-0229 to ANI-23-0288 = id_campaign 189 = date 03/23 (14th)
#id_sample ANI-23-0349 to ANI-23-0408 = id_campaign 190 = date 04/23 (17th)
#id_sample ANI-23-0679 to ANI-23-0738 = id_campaign 191 = date 05/23 (2nd)

# homogeneise code
tidy_macrofauna_surface_1$usage[tidy_macrofauna_surface_1$usage == 'C'] <- 'MC'
tidy_macrofauna_surface_1$usage[tidy_macrofauna_surface_1$usage == 'F'] <- 'TP'

tidy_macrofauna_surface_1$block[tidy_macrofauna_surface_1$block == '1'] <- 'B1'
tidy_macrofauna_surface_1$block[tidy_macrofauna_surface_1$block == '2'] <- 'B2'
tidy_macrofauna_surface_1$block[tidy_macrofauna_surface_1$block == '3'] <- 'B3'

tidy_macrofauna_surface_1$habitat[tidy_macrofauna_surface_1$habitat == 'Cult' | tidy_macrofauna_surface_1$habitat == 'cult' ] <- 'C'
tidy_macrofauna_surface_1$habitat[tidy_macrofauna_surface_1$habitat == 'Forest' | tidy_macrofauna_surface_1$habitat == 'LSA' ] <- 'T'

tidy_macrofauna_surface_1$replicate[tidy_macrofauna_surface_1$replicate == '1'] <- 'R1'
tidy_macrofauna_surface_1$replicate[tidy_macrofauna_surface_1$replicate == '2'] <- 'R2'
tidy_macrofauna_surface_1$replicate[tidy_macrofauna_surface_1$replicate == '3'] <- 'R3'
tidy_macrofauna_surface_1$replicate[tidy_macrofauna_surface_1$replicate == '4'] <- 'R4'
tidy_macrofauna_surface_1$replicate[tidy_macrofauna_surface_1$replicate == '5'] <- 'R5'

# create column code_uhbr
tidy_macrofauna_surface_1$code_uhbr <- paste(tidy_macrofauna_surface_1$usage, tidy_macrofauna_surface_1$habitat, tidy_macrofauna_surface_1$block, tidy_macrofauna_surface_1$replicate, sep = "_")
```

## macrofauna_aerial

```{r }
tidy_macrofauna_aerial_1 <- macrofauna_aerial_0

# rename columns
tidy_macrofauna_aerial_1 <- tidy_macrofauna_aerial_1 %>%  rename(block = bloc, replicate = rep)

# add column method, date and community
tidy_macrofauna_aerial_1 <- tidy_macrofauna_aerial_1 %>% mutate(method = "pan_trap")
tidy_macrofauna_aerial_1 <- tidy_macrofauna_aerial_1 %>%
  mutate(date = case_when(
    grepl('03/23', month) ~ "03/2023",
    grepl('04/23', month) ~ "04/2023")) # samples collected in May (id_campaign 191 and id_sample from ANI-23-0739 to ANI-23-0798 are not in this table because they are not identified yet)
tidy_macrofauna_aerial_1 <- tidy_macrofauna_aerial_1 %>% mutate(community = "macrofauna_aerial")

# homogeneise code
tidy_macrofauna_aerial_1$usage[tidy_macrofauna_aerial_1$usage == 'C'] <- 'MC'
tidy_macrofauna_aerial_1$usage[tidy_macrofauna_aerial_1$usage == 'F'] <- 'TP'

tidy_macrofauna_aerial_1$block[tidy_macrofauna_aerial_1$block == '1'] <- 'B1'
tidy_macrofauna_aerial_1$block[tidy_macrofauna_aerial_1$block == '2'] <- 'B2'
tidy_macrofauna_aerial_1$block[tidy_macrofauna_aerial_1$block == '3'] <- 'B3'

tidy_macrofauna_aerial_1$habitat[tidy_macrofauna_aerial_1$habitat == 'Cult' | tidy_macrofauna_aerial_1$habitat == 'cult' ] <- 'C'
tidy_macrofauna_aerial_1$habitat[tidy_macrofauna_aerial_1$habitat == 'Forest' | tidy_macrofauna_aerial_1$habitat == 'LSA' ] <- 'T'

tidy_macrofauna_aerial_1$replicate[tidy_macrofauna_aerial_1$replicate == '1'] <- 'R1'
tidy_macrofauna_aerial_1$replicate[tidy_macrofauna_aerial_1$replicate == '2'] <- 'R2'
tidy_macrofauna_aerial_1$replicate[tidy_macrofauna_aerial_1$replicate == '3'] <- 'R3'
tidy_macrofauna_aerial_1$replicate[tidy_macrofauna_aerial_1$replicate == '4'] <- 'R4'
tidy_macrofauna_aerial_1$replicate[tidy_macrofauna_aerial_1$replicate == '5'] <- 'R5'

# create column code_uhbr
tidy_macrofauna_aerial_1$code_uhbr <- paste(tidy_macrofauna_aerial_1$usage, tidy_macrofauna_aerial_1$habitat, tidy_macrofauna_aerial_1$block, tidy_macrofauna_aerial_1$replicate, sep = "_")
```

## macrofauna_foliar

```{r}
tidy_macrofauna_foliar_1 <- macrofauna_foliar_0

# modify column method (two organs were collected for macrofauna_foliar)
tidy_macrofauna_foliar_1$method[macrofauna_foliar_0$method == 'rameau'] <- 'twig'
tidy_macrofauna_foliar_1$method[tidy_macrofauna_foliar_1$method == 'dissection inflo'] <- 'inflorescence'

# add column habitat, date and community
tidy_macrofauna_foliar_1 <- tidy_macrofauna_foliar_1 %>% mutate(habitat = "T")
tidy_macrofauna_foliar_1 <- tidy_macrofauna_foliar_1 %>% mutate(community = "macrofauna_foliar")
tidy_macrofauna_foliar_1 <- tidy_macrofauna_foliar_1 %>%
  mutate(date = case_when(
      grepl(number_range(409, 498), id_sample) ~ "04/2023",
      grepl(number_range(499, 588), id_sample) ~ "05/2023",
      grepl(number_range(589, 678), id_sample) ~ "06/2023"))
#id_sample ANI-23-0409 to ANI-23-0498 = id_campaign 190 = date 04/23 (17th)
#id_sample ANI-23-0499 to ANI-23-0588 = id_campaign 191 = date 05/23 (2nd)
#id_sample ANI-23-0589 to ANI-23-0678 = id_campaign 192 = date 06/23 (25th May)
#samples collected on leaf pods in June (id_campaign 192 and id_sample from ANI-23-0799 to ANI-23-0887) are not in this table

# homogeneise code
tidy_macrofauna_foliar_1 <- tidy_macrofauna_foliar_1 %>% 
   separate(col = usage_bloc , into = c("usage","block","rang","arbre"), sep = "_")

tidy_macrofauna_foliar_1 <- tidy_macrofauna_foliar_1 %>% mutate(across('rang', str_replace, 'arbre', ''))

# OPTION 1 (keeping rang and arbre info in code_uhbr)
#macrofauna_foliar_1 <- macrofauna_foliar_1 %>% mutate(rang = gsub("rang", "R", rang)) 
#macrofauna_foliar_1 <- macrofauna_foliar_1 %>% mutate(arbre = gsub("arbre", "", arbre))
#macrofauna_foliar_1 <- macrofauna_foliar_1 %>% mutate(replicate = paste(rang, arbre ,sep="."))

# OPTION 2 (simplified: rang number becomes replicate number)
tidy_macrofauna_foliar_1 <- tidy_macrofauna_foliar_1 %>% mutate(replicate = gsub("rang", "R", rang)) 

tidy_macrofauna_foliar_1$block[tidy_macrofauna_foliar_1$block == '1'] <- 'B1'
tidy_macrofauna_foliar_1$block[tidy_macrofauna_foliar_1$block == '2'] <- 'B2'
tidy_macrofauna_foliar_1$block[tidy_macrofauna_foliar_1$block == '3'] <- 'B3'

# create column code_uhbr
tidy_macrofauna_foliar_1$code_uhbr <- paste(tidy_macrofauna_foliar_1$usage, tidy_macrofauna_foliar_1$habitat, tidy_macrofauna_foliar_1$block, tidy_macrofauna_foliar_1$replicate, sep = "_")
```

## nematodes

### march

```{r}
nematodes_m_0 <- nematodes_0[15:27,2:26]
nematodes_m_0 <- nematodes_m_0 %>% row_to_names(row_number = 1)
nematodes_m_0 <- nematodes_m_0 %>% mutate_at(-c(1,2), as.numeric) %>% mutate_if(is.numeric, round, digits=3)

# add columns replicate, date, method and community
nematodes_m_0 <- nematodes_m_0 %>% mutate(replicate = "R1")
nematodes_m_0 <- nematodes_m_0 %>% mutate(date = "03/2023")
nematodes_m_0 <- nematodes_m_0 %>% mutate(method = "baermann")
nematodes_m_0 <- nematodes_m_0 %>% mutate(community = "nematodes")

# homogeneise code
nematodes_m_0 <- nematodes_m_0 %>%
  mutate(
    usage = case_when(
      Taxa == "Tree plantation" ~ "TP",
      Taxa == "Annual crop" ~ "MC",
      Taxa == "Under tree in AF" |  Taxa == "Under annual crop in AF" ~ "AF"),
    habitat = case_when(
      Taxa == "Tree plantation" | Taxa == "Under tree in AF" ~ "T",
      Taxa == "Annual crop" | Taxa == "Under annual crop in AF" ~ "C"),
    block = paste0("B", parse_number(Block))
    )

# create column code_uhbr
nematodes_m_0$code_uhbr <- paste(nematodes_m_0$usage, nematodes_m_0$habitat, nematodes_m_0$block, nematodes_m_0$replicate, sep="_")
```

### april

```{r}
nematodes_a_0 <- nematodes_0[1:13,2:32]
nematodes_a_0 <- nematodes_a_0 %>% row_to_names(row_number = 1)
nematodes_a_0 <- nematodes_a_0 %>% mutate_at(-c(1,2), as.numeric) %>% mutate_if(is.numeric, round, digits=3)

# add columns replicate, date, method and community
nematodes_a_0 <- nematodes_a_0 %>% mutate(replicate = "R1")
nematodes_a_0 <- nematodes_a_0 %>% mutate(date = "04/2023")
nematodes_a_0 <- nematodes_a_0 %>% mutate(method = "baermann")
nematodes_a_0 <- nematodes_a_0 %>% mutate(community = "nematodes")

# homogeneise code
nematodes_a_0 <- nematodes_a_0 %>%
  mutate(
    usage = case_when(
      Taxa == "Tree plantation" ~ "TP",
      Taxa == "Annual crop" ~ "MC",
      Taxa == "Under tree in AF" |  Taxa == "Under annual crop in AF" ~ "AF"),
    habitat = case_when(
      Taxa == "Tree plantation" | Taxa == "Under tree in AF" ~ "T",
      Taxa == "Annual crop" | Taxa == "Under annual crop in AF" ~ "C"),
    block = paste0("B", parse_number(Block))
    )

# create column code_uhbr
nematodes_a_0$code_uhbr <- paste(nematodes_a_0$usage, nematodes_a_0$habitat, nematodes_a_0$block, nematodes_a_0$replicate, sep="_")
```

### all

```{r}
# bind and pivot data
tidy_nematodes_1 <- bind_rows(nematodes_a_0[,-c(1,2)], nematodes_m_0[,-c(1,2)])
tidy_nematodes_1 <- tidy_nematodes_1 %>% replace(is.na(.), 0)
tidy_nematodes_1 <- tidy_nematodes_1 %>% select(code_uhbr, usage, habitat, block, replicate, date, method, community, everything())
tidy_nematodes_1 <- pivot_longer(tidy_nematodes_1, cols = c(9:38), names_to = "taxa", values_to = "abundance")
tidy_nematodes_1 <- tidy_nematodes_1 %>% rename(name=taxa)
tidy_nematodes_1$abundance <- lapply(tidy_nematodes_1$abundance, round)
```

## micro-arthropodes

```{r}
micro_arthropodes_all_1 <- micro_arthropodes_all_0

# delete 3 last rows (they are not samples)
micro_arthropodes_all_1 <- micro_arthropodes_all_1 %>% slice(-c(346:348))

# remplace NA by 0 for abundance
micro_arthropodes_all_1 <- micro_arthropodes_all_1 %>%
  mutate(Nbr_identifiés = ifelse(is.na(Nbr_identifiés), 0, Nbr_identifiés),
         Nbr_ind_tri = ifelse(is.na(Nbr_ind_tri), 0, Nbr_ind_tri))

# add column Nbr_individus_final = Nbr_ind_tri - Nbr_identifiés
## Nbr_ind_tri is the total number of individuals
## Nbr_identifiés is the number of collembola more finely identified
## Nbr_individus_final is needed because not all collembola have been identified
micro_arthropodes_all_1$Nbr_ind_tri <- as.numeric(micro_arthropodes_all_1$Nbr_ind_tri)
micro_arthropodes_all_1$Nbr_identifiés <- as.numeric(micro_arthropodes_all_1$Nbr_identifiés) 

micro_arthropodes_all_1 <- transform(micro_arthropodes_all_1, Nbr_individus_final = Nbr_ind_tri - Nbr_identifiés)

# add collembola identified
micro_arthropodes_all_1 <- micro_arthropodes_all_1 %>% 
  rename(Sites=Num_prélevement)

tidy_micro_arthropodes_1 <- bind_rows(micro_arthropodes_all_1, micro_arthropodes_col_0)

# add a unique column abundance
tidy_micro_arthropodes_1 <- tidy_micro_arthropodes_1 %>% mutate(abundance = coalesce(Nbr_individus_final, Nbr_individus))
tidy_micro_arthropodes_1 <- tidy_micro_arthropodes_1 %>% mutate(name = coalesce(Noms_espèces, best_ID))

# add columns date, method and community
tidy_micro_arthropodes_1 <- tidy_micro_arthropodes_1 %>% mutate(date = "04/2023")
tidy_micro_arthropodes_1 <- tidy_micro_arthropodes_1 %>% mutate(method = "macfayden")
tidy_micro_arthropodes_1 <- tidy_micro_arthropodes_1 %>% mutate(community = "micro_arthropodes")

# homogeneise code
tidy_micro_arthropodes_1 <- tidy_micro_arthropodes_1 %>%
 mutate(usage = case_when(
    grepl('C[[:digit:]]_', Sites) ~ "MC",
    grepl('AF[[:digit:]]_', Sites) ~ "AF",
    grepl('F[[:digit:]]_', Sites) ~ "TP"),
        habitat = case_when(
    grepl('C[[:digit:]]_|AF[[:digit:]]_C', Sites) ~ "C",
    grepl('F[[:digit:]]_|AF[[:digit:]]_LSA', Sites) ~ "T"),
        block = case_when(
    grepl('1_', Sites) ~ "B1",
    grepl('2_', Sites) ~ "B2",
    grepl('3_', Sites) ~ "B3"),
        replicate = case_when(
    grepl('_C1|_LSA1', Sites) ~ "R1",
    grepl('_C2|_LSA2', Sites) ~ "R2",
    grepl('_C3|_LSA3', Sites) ~ "R3",
    grepl('_C4|_LSA4', Sites) ~ "R4",
    grepl('_C5|_LSA5', Sites) ~ "R5"))

# create column code_uhbr
tidy_micro_arthropodes_1$code_uhbr <- paste(tidy_micro_arthropodes_1$usage, tidy_micro_arthropodes_1$habitat, tidy_micro_arthropodes_1$block, tidy_micro_arthropodes_1$replicate, sep="_")
```

## micro-organisms

```{r}
metadata_mo_1 <- metadata_mo_0

# rename columns
metadata_mo_1 <- metadata_mo_1[c(1:60),] %>%  rename(block = Bloc, replicate = REP, habitat = HABITAT)

# add columns date and method
metadata_mo_1 <- metadata_mo_1 %>% mutate(date = "04/2023")
metadata_mo_1 <- metadata_mo_1 %>% mutate(method = "metabarcoding")

# homogeneise code
metadata_mo_1$usage[metadata_mo_1$usage == 'C'] <- 'MC'
metadata_mo_1$usage[metadata_mo_1$usage == 'F'] <- 'TP'

metadata_mo_1$block <- paste0("B", parse_number(metadata_mo_1$block))

metadata_mo_1$habitat[metadata_mo_1$habitat == 'Tree_line' | metadata_mo_1$habitat == 'Tree_plantation' ] <- 'T'
metadata_mo_1$habitat[metadata_mo_1$habitat == 'Crop' ] <- 'C'

metadata_mo_1$replicate <- paste0("R", parse_number(metadata_mo_1$replicate))

# create column code_uhbr
metadata_mo_1$code_uhbr <- paste (metadata_mo_1$usage, metadata_mo_1$habitat, metadata_mo_1$block, metadata_mo_1$replicate, sep = "_")
```

### bacteria

```{r}
tidy_bacteria_1 <- bacteria_0

tidy_bacteria_1 <- pivot_longer(tidy_bacteria_1, cols = c(13:72), names_to =  "Reference_ADNid", values_to = "abundance", names_repair = "minimal")

tidy_bacteria_1$Reference_ADNid <- str_replace(tidy_bacteria_1$Reference_ADNid, '_.+_.+', "")

tidy_bacteria_1 <- left_join(tidy_bacteria_1, metadata_mo_1, by = "Reference_ADNid")

tidy_bacteria_1 <- tidy_bacteria_1 %>% mutate(community = "bacteria")
```

### fungi

```{r}
tidy_fungi_1 <- fungi_0

tidy_fungi_1 <- pivot_longer(tidy_fungi_1, cols = c(13:72), names_to =  "Reference_ADNid", values_to = "abundance", names_repair = "minimal")

tidy_fungi_1$Reference_ADNid <- str_replace(tidy_fungi_1$Reference_ADNid, '_.+_.+', "")

tidy_fungi_1 <- left_join(tidy_fungi_1, metadata_mo_1, by = "Reference_ADNid")

tidy_fungi_1 <- tidy_fungi_1 %>% mutate(community = "fungi")
```

### protists

```{r}
tidy_protists_1 <- protists_0

tidy_protists_1 <- pivot_longer(tidy_protists_1, cols = c(13:72), names_to =  "Reference_ADNid", values_to = "abundance", names_repair = "minimal")

tidy_protists_1$Reference_ADNid <- str_replace(tidy_protists_1$Reference_ADNid, '_.+_.+', "")

tidy_protists_1 <- left_join(tidy_protists_1, metadata_mo_1, by = "Reference_ADNid")

tidy_protists_1 <- tidy_protists_1 %>% mutate(community = "protists")
```

At this step, the files with cleaned data (suffix _1) are:
  macrofauna_surface
  macrofauna_aerial
  macrofauna_foliar
  micro_arthropodes
  nematodes
  bacteria
  fungi
  protists

# Export tidy_data

```{r}
# Select minimal data to export

tidy_macrofauna_surface_2 <- tidy_macrofauna_surface_1 %>% 
  select(code_uhbr, name, abundance, date, method, community)
# stade and mass can be used for traits
# fullName, genusName, familyName, orderName, className, phylumName will be regenerated in homogenized format during the tax_table section
write.csv(tidy_macrofauna_surface_2, "final_files/tidy_data/tidy_macrofauna_surface.csv")

tidy_macrofauna_aerial_2 <- tidy_macrofauna_aerial_1 %>% 
  select(code_uhbr, name, abundance, date, method, community)
# INat can be used for ID and stade can be used for traits
# fullName, genusName, familyName, orderName, className, phylumName will be regenerated in homogenized format during the tax_table section
write.csv(tidy_macrofauna_aerial_2, "final_files/tidy_data/tidy_macrofauna_aerial.csv")

tidy_macrofauna_foliar_2 <- tidy_macrofauna_foliar_1 %>% 
  select(code_uhbr, name, abundance, date, method, community)
# Valid.Name (what is it?)
# fullName, genusName, familyName, orderName, className, phylumName will be regenerated in homogenized format during the tax_table section
write.csv(tidy_macrofauna_foliar_2, "final_files/tidy_data/tidy_macrofauna_foliar.csv")

tidy_nematodes_2 <- tidy_nematodes_1 %>% select(code_uhbr, name, abundance, date, method, community)
tidy_nematodes_2$abundance = as.numeric(tidy_nematodes_2$abundance)
write.csv(tidy_nematodes_2, "final_files/tidy_data/tidy_nematodes.csv")

tidy_micro_arthropodes_2 <- tidy_micro_arthropodes_1 %>% select(code_uhbr, name, abundance, date, method, community)
# stade can be used for traits
# higher.level and lower.level will be regenerated in another tax_table
write.csv(tidy_micro_arthropodes_2, "final_files/tidy_data/tidy_micro_arthropodes.csv")

tidy_bacteria_2 <- tidy_bacteria_1 %>% 
  select(code_uhbr, seed_id, abundance, date, method, community)
tidy_bacteria_2 <- tidy_bacteria_2 %>%  rename(name = seed_id)
write.csv(tidy_bacteria_2, "final_files/tidy_data/tidy_bacteria.csv")

tidy_fungi_2 <- tidy_fungi_1 %>% 
  select(code_uhbr, seed_id, abundance, date, method, community)
tidy_fungi_2 <- tidy_fungi_2 %>%  rename(name = seed_id)
write.csv(tidy_fungi_2, "final_files/tidy_data/tidy_fungi.csv")

tidy_protists_2 <- tidy_protists_1 %>% 
  select(code_uhbr, seed_id, abundance, date, method, community)
tidy_protists_2 <- tidy_protists_2 %>%  rename(name = seed_id)
write.csv(tidy_protists_2, "final_files/tidy_data/tidy_protists.csv")
```

```{r}
tidy_ALL <- bind_rows(tidy_macrofauna_surface_2, tidy_macrofauna_aerial_2, tidy_macrofauna_foliar_2, tidy_nematodes_2, tidy_micro_arthropodes_2, tidy_bacteria_2, tidy_fungi_2, tidy_protists_2) # bind all communities in one file

tidy_ALL <- aggregate(tidy_ALL$abundance, by=list(code_uhbr=tidy_ALL$code_uhbr, name=tidy_ALL$name, date=tidy_ALL$date, method=tidy_ALL$method, community=tidy_ALL$community), FUN=sum) %>% rename(abundance = x) # aggregate abundances by uhbr, name, date, method and community

tidy_ALL <- filter(tidy_ALL, abundance != 0) # remove rows with abundance = 0 (especially needed for amplicon barcoding since each sample has a row for all barcode/assigned taxa name)

write.csv(tidy_ALL, "final_files/tidy_data/tidy_data_update21-05-24.csv")
saveRDS(tidy_ALL, "final_files/tidy_data/tidy_data_update21-05-24.rds")
```

# 2. Taxonomy table

Files generated in this section:
prefix "tax_"
suffix "_1" = file to use for taxize input (query)
suffix "_2" = raw taxize output (ITIS and NCBI results)
suffix "_3" = cleaned taxonomy
suffix "_ALL" = all data combined (all communities)

To go faster while generating the taxonomy, use a NCBI entrez key.
First, generate a key cf https://ncbiinsights.ncbi.nlm.nih.gov/2017/11/02/new-api-keys-for-the-e-utilities/
Then, set your environment by doing once:
usethis::edit_r_environ()
ENTREZ_KEY='youractualkey' # To write in the environment

NB: Even with the key, error (400) appears but can be solved by using a loop.

## macrofauna_surface

```{r}
tax_macrofauna_surface_1 <- tidy_macrofauna_surface_2 %>% distinct(name)
temp <- gnr_resolve(tax_macrofauna_surface_1$name, preferred_data_sources = c(1,3,4,11,12), best_match_only = T) # 1 = CLC, 3 = ITIS, 4 = NCBI, 11 = GBIF, 12 = EOL (https://resolver.globalnames.org/data_sources)
diff <- setdiff(tax_macrofauna_surface_1$name, temp$user_supplied_name) 
print(diff)
# Manual cleaning
## Pezottetix not found because wrong spelling
tax_macrofauna_surface_1$name[tax_macrofauna_surface_1$name == "Pezottetix"] <- "Pezotettix"
# Run again from temp... and check diff = 0 now

write.csv(tax_macrofauna_surface_1, "intermediate_files/taxize/macrofauna_surface_taxizeinput.csv")
```

```{r, echo=FALSE}
tax_macrofauna_surface_2 = NULL
for (i in 1: nrow(tax_macrofauna_surface_1)) {
classes_i <- try(tax_name(sci = tax_macrofauna_surface_1$name[i], get = c("kingdom", "phylum", "class", "order", "family", "genus", "species"), db = "both", ask = FALSE))
if (class(classes_i)=="try-error") {
Sys.sleep(10)
classes_i <- try(tax_name(sci = tax_macrofauna_surface_1$name[i], get = c("kingdom", "phylum", "class", "order", "family", "genus", "species"), db = "both", ask = FALSE))}
tax_macrofauna_surface_2 <- rbind(tax_macrofauna_surface_2, classes_i)
}

tax_macrofauna_surface_2 <- tax_macrofauna_surface_2[- grep("Error", tax_macrofauna_surface_2$query),] # remove the rows with errors
write.csv(tax_macrofauna_surface_2, "intermediate_files/taxize/macrofauna_surface_taxizeoutput.csv")
```

```{r, echo=FALSE}
taxnotfoundboth <- tax_macrofauna_surface_2[!complete.cases(tax_macrofauna_surface_2$kingdom),]
taxnotfoundonedb <- taxnotfoundboth %>% group_by(query) %>% filter(n() == 1) # These are the taxa found in only one database (not found in the database in this object = found in the other database in macrofauna_foliar_2). Therefore, if we filter NCBI (by default) results in macrofauna_foliar_2, we must also keep ITIS results for the taxa with no NCBI results in this objects. These taxa to keep are saved in the following vector :
keepitis <- c(subset(taxnotfoundonedb, db == 'ncbi')$query)
taxnotfound_ncbianditis <- subset(taxnotfoundboth,duplicated(query)) # These are the taxa not found in any database. We will export the list, fill it manually and reimport it.
write.csv(taxnotfound_ncbianditis[,-1], "intermediate_files/taxize/macrofauna_surface_taxnotfound_ncbianditis.csv") # Fill cells with info found on internet, save it to x_mod and import it
taxfound_manually <- read.csv("intermediate_files/taxize/macrofauna_surface_taxnotfound_ncbianditis_mod.csv", sep = ";")
taxfound_manually <- taxfound_manually[complete.cases(taxfound_manually$kingdom),-c(9,10)]

# Keep NCBI results in macrofauna_surface_tax and add ITIS results and manual taxonomy
taxfound_ncbi <- tax_macrofauna_surface_2[tax_macrofauna_surface_2$db == 'ncbi' & complete.cases(tax_macrofauna_surface_2$kingdom), -1]
taxfound_itisonly <- tax_macrofauna_surface_2[tax_macrofauna_surface_2$db == 'itis' & tax_macrofauna_surface_2$query %in% keepitis, ]
tax_macrofauna_surface_3 <- bind_rows(taxfound_ncbi, taxfound_itisonly[,-1], taxfound_manually) %>% mutate(community = "macrofauna_surface")

tax_macrofauna_surface_3[tax_macrofauna_surface_3 == "Animalia"] <- "Metazoa" # Replace Animalia (ITIS) par Metazoa (NCBI)
tax_macrofauna_surface_3 <- tax_macrofauna_surface_3 %>% replace(is.na(.), "unassigned")

write.csv(tax_macrofauna_surface_3, "final_files/tax_table/tax_macrofauna_surface.csv")
```

## macrofauna_aerial

```{r}
tax_macrofauna_aerial_1 <- tidy_macrofauna_aerial_2 %>% distinct(name)
temp <- gnr_resolve(tax_macrofauna_aerial_1$name, preferred_data_sources = c(1,3,4,11,12), best_match_only = T) # 1 = CLC, 3 = ITIS, 4 = NCBI, 11 = GBIF, 12 = EOL (https://resolver.globalnames.org/data_sources)
diff <- setdiff(tax_macrofauna_aerial_1$name, temp$user_supplied_name) 
print(diff)
# Manual cleaning
## Belytini not found because its a tribu of Belytinae
tax_macrofauna_aerial_1$name[tax_macrofauna_aerial_1$name == "Belytini"] <- "Belytinae"
# Run again from temp... and check diff = 0 now

write.csv(tax_macrofauna_aerial_1, "intermediate_files/taxize/tax_macrofauna_aerial_taxizeinput.csv")
```

```{r, echo=FALSE}
tax_macrofauna_aerial_2 = NULL
for (i in 1: nrow(tax_macrofauna_aerial_1)) {
classes_i <- try(tax_name(sci = tax_macrofauna_aerial_1$name[i], get = c("kingdom", "phylum", "class", "order", "family", "genus", "species"), db = "both", ask = FALSE))
if (class(classes_i)=="try-error") {
Sys.sleep(10)
classes_i <- try(tax_name(sci = tax_macrofauna_aerial_1$name[i], get = c("kingdom", "phylum", "class", "order", "family", "genus", "species"), db = "both", ask = FALSE))}
tax_macrofauna_aerial_2 <- rbind(tax_macrofauna_aerial_2, classes_i)
}

write.csv(tax_macrofauna_aerial_2, "intermediate_files/taxize/macrofauna_aerial_taxize_output.csv")
```

```{r, echo=FALSE}
taxnotfoundboth <- tax_macrofauna_aerial_2[!complete.cases(tax_macrofauna_aerial_2$kingdom),]
taxnotfoundonedb <- taxnotfoundboth %>% group_by(query) %>% filter(n() ==1) # These are the taxa found in only one database (not found in the database in this object = found in the other database in macrofauna_foliar_2). Therefore, if we filter NCBI (by default) results in macrofauna_foliar_2, we must also keep ITIS results for the taxa with no NCBI results in this objects. These taxa to keep are saved in the following vector :
keepitis <- c(subset(taxnotfoundonedb, db == 'ncbi')$query)
taxnotfound_ncbianditis <- subset(taxnotfoundboth,duplicated(query)) # These are the taxa not found in any database. We will export the list, fill it manually and reimport it.
write.csv(taxnotfound_ncbianditis[,-1], "intermediate_files/taxize/macrofauna_aerial_taxnotfound_ncbianditis.csv") # Fill cells with info found on internet, save it to x_mod and import it
taxfound_manually <- read.csv("intermediate_files/taxize/macrofauna_aerial_taxnotfound_ncbianditis_mod.csv", sep = ";")
taxfound_manually <- taxfound_manually[complete.cases(taxfound_manually$kingdom),-c(9,10)]

# Keep NCBI results in macrofauna_aerial_tax and add ITIS results and manual taxonomy
taxfound_ncbi <- tax_macrofauna_aerial_2[tax_macrofauna_aerial_2$db == 'ncbi' & complete.cases(tax_macrofauna_aerial_2$kingdom), -1]
taxfound_itisonly <- tax_macrofauna_aerial_2[tax_macrofauna_aerial_2$db == 'itis' & tax_macrofauna_aerial_2$query %in% keepitis, ]
tax_macrofauna_aerial_3 <- bind_rows(taxfound_ncbi, taxfound_itisonly[,-1], taxfound_manually) %>% mutate(community = "macrofauna_aerial")

tax_macrofauna_aerial_3[tax_macrofauna_aerial_3 == "Animalia"] <- "Metazoa" # Replace Animalia (ITIS) par Metazoa (NCBI)
tax_macrofauna_aerial_3 <- tax_macrofauna_aerial_3 %>% replace(is.na(.), "unassigned")

write.csv(tax_macrofauna_aerial_3, "final_files/tax_table/tax_macrofauna_aerial.csv")
```

## macrofauna_foliar

```{r}
tax_macrofauna_foliar_1 <- tidy_macrofauna_foliar_2 %>% distinct(name)
temp <- gnr_resolve(tax_macrofauna_foliar_1$name, preferred_data_sources = c(1,3,4,11,12), best_match_only = T) # 1 = CLC, 3 = ITIS, 4 = NCBI, 11 = GBIF, 12 = EOL (https://resolver.globalnames.org/data_sources)
diff <- setdiff(tax_macrofauna_foliar_1$name, temp$user_supplied_name) 
print(diff)
# Manual cleaning
## Phonognathidae not found because wrong spelling
tax_macrofauna_foliar_1$name[tax_macrofauna_foliar_1$name == "Phonognathidae"] <- "Phonognathinae"
## vide should be removed
tax_macrofauna_foliar_1 <- tax_macrofauna_foliar_1[!(tax_macrofauna_foliar_1$name %in% "vide"),]
# Run again from temp... and check diff = 0 now

write.csv(tax_macrofauna_foliar_1, "intermediate_files/taxize/tax_macrofauna_foliar_taxizeinput.csv")
```

```{r, echo=FALSE}
tax_macrofauna_foliar_2 = NULL
for (i in 1: nrow(tax_macrofauna_foliar_1)) {
classes_i <- try(tax_name(sci = tax_macrofauna_foliar_1$name[i], get = c("kingdom", "phylum", "class", "order", "family", "genus", "species"), db = "both", ask = FALSE))
if (class(classes_i)=="try-error") {
Sys.sleep(10)
classes_i <- try(tax_name(sci = tax_macrofauna_foliar_1$name[i], get = c("kingdom", "phylum", "class", "order", "family", "genus", "species"), db = "both", ask = FALSE))}
tax_macrofauna_foliar_2 <- rbind(tax_macrofauna_foliar_2, classes_i)
}

tax_macrofauna_foliar_2 <- tax_macrofauna_foliar_2[- grep("Error", tax_macrofauna_foliar_2$query),] # remove the rows with errors
write.csv(tax_macrofauna_foliar_2, "intermediate_files/taxize/macrofauna_foliar_taxizeoutput.csv")
```

```{r, echo=FALSE}
taxnotfoundboth <- tax_macrofauna_foliar_2[!complete.cases(tax_macrofauna_foliar_2$kingdom),]
taxnotfoundonedb <- taxnotfoundboth %>% group_by(query) %>% filter(n() ==1) # These are the taxa found in only one database (not found in the database in this object = found in the other database in macrofauna_foliar_2). Therefore, if we filter NCBI (by default) results in macrofauna_foliar_2, we must also keep ITIS results for the taxa with no NCBI results in this objects. These taxa to keep are saved in the following vector :
keepitis <- c(subset(taxnotfoundonedb, db == 'ncbi')$query)
taxnotfound_ncbianditis <- subset(taxnotfoundboth,duplicated(query)) # These are the taxa not found in any database. We will export the list, fill it manually and reimport it.
write.csv(taxnotfound_ncbianditis[,-1], "intermediate_files/taxize/macrofauna_foliar_taxnotfound_ncbianditis.csv") # Fill cells with info found on internet, save it to x_mod and import it
taxfound_manually <- read.csv("intermediate_files/taxize/macrofauna_foliar_taxnotfound_ncbianditis_mod.csv", sep = ";")
taxfound_manually <- taxfound_manually[complete.cases(taxfound_manually$kingdom),-c(9,10)]

# Keep NCBI results in macrofauna_foliar_tax and add ITIS results and manual taxonomy
taxfound_ncbi <- tax_macrofauna_foliar_2[tax_macrofauna_foliar_2$db == 'ncbi' & complete.cases(tax_macrofauna_foliar_2$kingdom), -1]
taxfound_itisonly <- tax_macrofauna_foliar_2[tax_macrofauna_foliar_2$db == 'itis' & tax_macrofauna_foliar_2$query %in% keepitis, ]
tax_macrofauna_foliar_3 <- bind_rows(taxfound_ncbi, taxfound_itisonly[,-1], taxfound_manually) %>% mutate(community = "macrofauna_foliar")

tax_macrofauna_foliar_3[tax_macrofauna_foliar_3 == "Animalia"] <- "Metazoa" # Replace Animalia (ITIS) par Metazoa (NCBI)
tax_macrofauna_foliar_3 <- tax_macrofauna_foliar_3 %>% replace(is.na(.), "unassigned")

write.csv(tax_macrofauna_foliar_3, "final_files/tax_table/tax_macrofauna_foliar.csv")
```                   

## nematodes
                    
```{r}
tax_nematodes_1 <- tidy_nematodes_2 %>% distinct(name)
temp <- gnr_resolve(tax_nematodes_1$name, preferred_data_sources = c(1,3,4,11,12), best_match_only = T) # 1 = CLC, 3 = ITIS, 4 = NCBI, 11 = GBIF, 12 = EOL (https://resolver.globalnames.org/data_sources)
diff <- setdiff(tax_nematodes_1$name, temp$user_supplied_name) 
print(diff)
# diff = 0 so no need for manual cleaning

write.csv(tax_nematodes_1, "intermediate_files/taxize/tax_nematodes_taxizeinput.csv")
```

```{r, echo=FALSE}
tax_nematodes_2 = NULL
for (i in 1: nrow(tax_nematodes_1)) {
classes_i <- try(tax_name(sci = tax_nematodes_1$name[i], get = c("kingdom", "phylum", "class", "order", "family", "genus", "species"), db = "both", ask = FALSE))
if (class(classes_i)=="try-error") {
Sys.sleep(10)
classes_i <- try(tax_name(sci = tax_nematodes_1$name[i], get = c("kingdom", "phylum", "class", "order", "family", "genus", "species"), db = "both", ask = FALSE))}
tax_nematodes_2 <- rbind(tax_nematodes_2, classes_i)
}

write.csv(tax_nematodes_2, "intermediate_files/taxize/nematodes_taxizeoutput.csv")
```

```{r, echo=FALSE}
taxnotfoundboth <- tax_nematodes_2[!complete.cases(tax_nematodes_2$kingdom),] 
taxnotfound_ncbianditis <- subset(taxnotfoundboth,duplicated(query)) # All taxa are found with NCBI or ITIS.

# Keep NCBI results in nematodes_tax and add ITIS results and manual taxonomy
tax_nematodes_3 <- tax_nematodes_2[tax_nematodes_2$db == 'ncbi' & complete.cases(tax_nematodes_2$kingdom), -1] %>% mutate(community = "nematodes")

tax_nematodes_3[tax_nematodes_3 == "Animalia"] <- "Metazoa" # Replace Animalia (ITIS) par Metazoa (NCBI)
tax_nematodes_3 <- tax_nematodes_3 %>% replace(is.na(.), "unassigned")

write.csv(tax_nematodes_3, "final_files/tax_table/tax_nematodes.csv")
```

## micro-arthropodes

```{r}
tax_micro_arthropodes_1 <- tidy_micro_arthropodes_2 %>% distinct(name)
temp <- gnr_resolve(tax_micro_arthropodes_1$name, preferred_data_sources = c(1,3,4,11,12), best_match_only = T) # 1 = CLC, 3 = ITIS, 4 = NCBI, 11 = GBIF, 12 = EOL (https://resolver.globalnames.org/data_sources)
diff <- setdiff(tax_micro_arthropodes_1$name, temp$user_supplied_name) 
print(diff)
# Manual cleaning
## NA should be removed
tax_micro_arthropodes_1 <- tax_micro_arthropodes_1[!(tax_micro_arthropodes_1$name %in% "NA"),]
tax_micro_arthropodes_1 <- data.frame(tax_micro_arthropodes_1)
tax_micro_arthropodes_1 <- tax_micro_arthropodes_1 %>%  rename(name = tax_micro_arthropodes_1)
# diff = 0 now

write.csv(tax_micro_arthropodes_1, "intermediate_files/taxize/tax_micro_arthropodes_taxizeinput.csv")
```

```{r, echo=FALSE}
tax_micro_arthropodes_2 = NULL
for (i in 1: nrow(tax_micro_arthropodes_1)) {
classes_i <- try(tax_name(sci = tax_micro_arthropodes_1$name[i], get = c("kingdom", "phylum", "class", "order", "family", "genus", "species"), db = "both", ask = FALSE))
if (class(classes_i)=="try-error") {
Sys.sleep(10)
classes_i <- try(tax_name(sci = tax_micro_arthropodes_1$name[i], get = c("kingdom", "phylum", "class", "order", "family", "genus", "species"), db = "both", ask = FALSE))}
tax_micro_arthropodes_2 <- rbind(tax_micro_arthropodes_2, classes_i)
}

write.csv(tax_micro_arthropodes_2, "intermediate_files/taxize/micro_arthropodes_taxizeoutput.csv")
```

```{r, echo=FALSE}
taxnotfoundboth <- tax_micro_arthropodes_2[!complete.cases(tax_micro_arthropodes_2$kingdom),]
taxnotfoundonedb <- taxnotfoundboth %>% group_by(query) %>% filter(n() ==1) # These are the taxa found in only one database (not found in the database in this object = found in the other database in micro_arthropodes_2). Therefore, if we filter NCBI (by default) results in micro_arthropodes_2, we must also keep ITIS results for the taxa with no NCBI results in this objects. These taxa to keep are saved in the following vector :
keepitis <- c(subset(taxnotfoundonedb, db == 'ncbi')$query)
taxnotfound_ncbianditis <- subset(taxnotfoundboth,duplicated(query)) # These are the taxa not found in any database. We will export the list, fill it manually and reimport it.
write.csv(taxnotfound_ncbianditis[,-1], "intermediate_files/taxize/micro_arthropodes_taxnotfound_ncbianditis.csv") # Fill cells with info found on internet, save it to x_mod and import it
taxfound_manually <- read.csv("intermediate_files/taxize/micro_arthropodes_taxnotfound_ncbianditis_mod.csv", sep = ";")
taxfound_manually <- taxfound_manually[complete.cases(taxfound_manually$kingdom),-c(9,10)]

# Keep NCBI results in micro_arthropodes_tax and add ITIS results and manual taxonomy
taxfound_ncbi <- tax_micro_arthropodes_2[tax_micro_arthropodes_2$db == 'ncbi' & complete.cases(tax_micro_arthropodes_2$kingdom), -1]
taxfound_itisonly <- tax_micro_arthropodes_2[tax_micro_arthropodes_2$db == 'itis' & tax_micro_arthropodes_2$query %in% keepitis, ]
tax_micro_arthropodes_3 <- bind_rows(taxfound_ncbi, taxfound_itisonly[,-1], taxfound_manually) %>% mutate(community = "micro_arthropodes")

tax_micro_arthropodes_3[tax_micro_arthropodes_3 == "Animalia"] <- "Metazoa" # Replace Animalia (ITIS) par Metazoa (NCBI)
tax_micro_arthropodes_3 <- tax_micro_arthropodes_3 %>% replace(is.na(.), "unassigned")

write.csv(tax_micro_arthropodes_3, "final_files/tax_table/tax_micro_arthropodes.csv")
```

## micro-organisms

### bacteria

```{r}
tax_bacteria_1 <- tidy_bacteria_1 %>% distinct(seed_id, blast_taxonomy)
tax_bacteria_1 <- tax_bacteria_1 %>% rename(query = seed_id)
tax_bacteria_1 <- tax_bacteria_1 %>% rename(name = blast_taxonomy)

tax_bacteria_2 <- tax_bacteria_1 %>% 
   separate(col = name, into = c("kingdom","phylum","class","order", "family", "genus", "species"), sep = ";")
tax_bacteria_2 <- tax_bacteria_2 %>% mutate(community = "bacteria")

tax_bacteria_3 <- tax_bacteria_3 %>% replace(is.na(.), "unassigned")
tax_bacteria_3[tax_bacteria_3 == "no data"] <- "unassigned"
tax_bacteria_3 <- tax_bacteria_3 %>% mutate(across(c(everything()), ~ sub("^unknown.*|Multi-affiliation", "unassigned", .)))

write.csv(tax_bacteria_3, "final_files/tax_table/tax_bacteria.csv")
```

### fungi

```{r}
tax_fungi_1 <- tidy_fungi_1 %>% distinct(seed_id, blast_taxonomy)
tax_fungi_1 <- tax_fungi_1 %>%  rename(query = seed_id)
tax_fungi_1 <- tax_fungi_1 %>%  rename(name = blast_taxonomy)

tax_fungi_2 <- tax_fungi_1 %>% 
   separate(col = name, into = c("kingdom","phylum","class","order", "family", "genus", "species"), sep = ";")
tax_fungi_2[] <- lapply(tax_fungi_2, gsub, pattern='.__', replacement='')
tax_fungi_2 <- tax_fungi_2 %>% mutate(community = "fungi")

tax_fungi_3 <- tax_fungi_2 %>% replace(is.na(.), "unassigned")
tax_fungi_3[tax_fungi_3 == "no data"] <- "unassigned"
tax_fungi_3 <- tax_fungi_3 %>% mutate(across(c(everything()), ~ sub("Multi-affiliation|unidentified", "unassigned", .)))

write.csv(tax_fungi_3, "final_files/tax_table/tax_fungi.csv")
```

### protists

```{r}
tax_protists_1 <- tidy_protists_1 %>% distinct(seed_id, blast_taxonomy)
tax_protists_1 <- tax_protists_1 %>%  rename(query = seed_id)
tax_protists_1 <- tax_protists_1 %>%  rename(name = blast_taxonomy)

tax_protists_2 <- tax_protists_1 %>% 
   separate(col = name, into = c("kingdom","phylum","class","order", "family", "genus", "species"), sep = ";")
tax_protists_2 <- tax_protists_2 %>% mutate(community = "protists")

tax_protists_3 <- tax_protists_2 %>% replace(is.na(.), "unassigned")
tax_protists_3[tax_protists_3 == "no data"] <- "unassigned"
tax_protists_3 <- tax_protists_3 %>% mutate(across(c(everything()), ~ sub("^unknown.*|Multi-affiliation|metagenome", "unassigned", .)))

write.csv(tax_protists_3, "final_files/tax_table/tax_protists.csv")
```

# Export tax_table

```{r}
tax_ALL <- bind_rows(tax_macrofauna_surface_3, tax_macrofauna_aerial_3, tax_macrofauna_foliar_3, tax_nematodes_3, tax_micro_arthropodes_3, tax_bacteria_3, tax_fungi_3, tax_protists_3) # bind all communities in one file
write.csv(tax_ALL, "final_files/tax_table/tax_table_update21-05-24.csv")
```

# 3. Functional table

```{r}
endpoint_url <- "http://129.88.204.79:7200/repositories/gratin"

tax_table <- read.csv("final_files/tax_table/tax_table_update21-05-24.csv")[,-1]
macrofauna_surface_tax <- subset(tax_table, community == 'macrofauna_surface')
macrofauna_aerial_tax <- subset(tax_table, community == 'macrofauna_aerial')
macrofauna_foliar_tax <- subset(tax_table, community == 'macrofauna_foliar')
nematodes_tax <- subset(tax_table, community == 'nematodes')
micro_arthropodes_tax <- subset(tax_table, community == 'micro_arthropodes')
bacteria_tax <- subset(tax_table, community == 'bacteria')
fungi_tax <- subset(tax_table, community == 'fungi')
protists_tax <- subset(tax_table, community == 'protists')
```

## macrofauna_surface

```{r}
macrofauna_surface_guilds_genus <- get.guilds(sciName = unique(macrofauna_surface_tax$genus), endpoint = endpoint_url)
write.csv(apply(macrofauna_surface_guilds_genus,2,as.character), "final_files/fun_table/genus/fun_macrofauna_surface_guilds_genus.csv")

macrofauna_surface_diets_genus <- get.diets(sciName = unique(macrofauna_surface_tax$genus), endpoint = endpoint_url)
write.csv(apply(macrofauna_surface_diets_genus,2,as.character), "final_files/fun_table/genus/fun_macrofauna_surface_diets_genus.csv")

macrofauna_surface_groups_genus <- get.trophic.groups(sciName = unique(macrofauna_surface_tax$genus), endpoint = endpoint_url)
write.csv(apply(macrofauna_surface_groups_genus,2,as.character), "final_files/fun_table/genus/fun_macrofauna_surface_groups_genus.csv")

macrofauna_surface_interactions_genus <- get.interactions(sciName = unique(macrofauna_surface_tax$genus), endpoint = endpoint_url)
write.csv(apply(macrofauna_surface_interactions_genus,2,as.character), "final_files/fun_table/genus/fun_macrofauna_surface_interactions_genus.csv")
```

## macrofauna_aerial

```{r}
macrofauna_aerial_guilds_genus <- get.guilds(sciName = unique(macrofauna_aerial_tax$genus), endpoint = endpoint_url)
write.csv(apply(macrofauna_aerial_guilds_genus,2,as.character), "final_files/fun_table/genus/fun_macrofauna_aerial_guilds_genus.csv")

macrofauna_aerial_diets_genus <- get.diets(sciName = unique(macrofauna_aerial_tax$genus), endpoint = endpoint_url)
write.csv(apply(macrofauna_aerial_diets_genus,2,as.character), "final_files/fun_table/genus/fun_macrofauna_aerial_diets_genus.csv")

macrofauna_aerial_groups_genus <- get.trophic.groups(sciName = unique(macrofauna_aerial_tax$genus), endpoint = endpoint_url)
write.csv(apply(macrofauna_aerial_groups_genus,2,as.character), "final_files/fun_table/genus/fun_macrofauna_aerial_groups_genus.csv")

macrofauna_aerial_interactions_genus <- get.interactions(sciName = unique(macrofauna_aerial_tax$genus), endpoint = endpoint_url)
write.csv(apply(macrofauna_aerial_interactions_genus,2,as.character), "final_files/fun_table/genus/fun_macrofauna_aerial_interactions_genus.csv")
```

## macrofauna_foliar

```{r}
macrofauna_foliar_guilds_genus <- get.guilds(sciName = unique(macrofauna_foliar_tax$genus), endpoint = endpoint_url)
write.csv(apply(macrofauna_foliar_guilds_genus,2,as.character), "final_files/fun_table/genus/fun_macrofauna_foliar_guilds_genus.csv")

macrofauna_foliar_diets_genus <- get.diets(sciName = unique(macrofauna_foliar_tax$genus), endpoint = endpoint_url)
write.csv(apply(macrofauna_foliar_diets_genus,2,as.character), "final_files/fun_table/genus/fun_macrofauna_foliar_diets_genus.csv")

macrofauna_foliar_groups_genus <- get.trophic.groups(sciName = unique(macrofauna_foliar_tax$genus), endpoint = endpoint_url)
write.csv(apply(macrofauna_foliar_groups_genus,2,as.character), "final_files/fun_table/genus/fun_macrofauna_foliar_groups_genus.csv")

macrofauna_foliar_interactions_genus <- get.interactions(sciName = unique(macrofauna_foliar_tax$genus), endpoint = endpoint_url)
write.csv(apply(macrofauna_foliar_interactions_genus,2,as.character), "final_files/fun_table/genus/fun_macrofauna_foliar_interactions_genus.csv")
```

## nematodes

```{r}
nematodes_guilds_genus <- get.guilds(sciName = unique(nematodes_tax$genus), endpoint = endpoint_url)
write.csv(apply(nematodes_guilds_genus,2,as.character), "final_files/fun_table/genus/fun_nematodes_guilds_genus.csv")

nematodes_diets_genus <- get.diets(sciName = unique(nematodes_tax$genus), endpoint = endpoint_url)
write.csv(apply(nematodes_diets_genus,2,as.character), "final_files/fun_table/genus/fun_nematodes_diets_genus.csv")

nematodes_groups_genus <- get.trophic.groups(sciName = unique(nematodes_tax$genus), endpoint = endpoint_url)
write.csv(apply(nematodes_groups_genus,2,as.character), "final_files/fun_table/genus/fun_nematodes_groups_genus.csv")

nematodes_interactions_genus <- get.interactions(sciName = unique(nematodes_tax$genus), endpoint = endpoint_url)
write.csv(apply(nematodes_interactions_genus,2,as.character), "final_files/fun_table/genus/fun_nematodes_interactions_genus.csv")
```

## micro-arthropodes

```{r}
micro_arthropodes_guilds_genus <- get.guilds(sciName = unique(micro_arthropodes_tax$genus), endpoint = endpoint_url)
write.csv(apply(micro_arthropodes_guilds_genus,2,as.character), "final_files/fun_table/genus/fun_micro_arthropodes_guilds_genus.csv")

micro_arthropodes_diets_genus <- get.diets(sciName = unique(micro_arthropodes_tax$genus), endpoint = endpoint_url)
write.csv(apply(micro_arthropodes_diets_genus,2,as.character), "final_files/fun_table/genus/fun_micro_arthropodes_diets_genus.csv")

micro_arthropodes_groups_genus <- get.trophic.groups(sciName = unique(micro_arthropodes_tax$genus), endpoint = endpoint_url)
write.csv(apply(micro_arthropodes_groups_genus,2,as.character), "final_files/fun_table/genus/fun_micro_arthropodes_groups_genus.csv")

micro_arthropodes_interactions_genus <- get.interactions(sciName = unique(micro_arthropodes_tax$genus), endpoint = endpoint_url)
write.csv(apply(micro_arthropodes_interactions_genus,2,as.character), "final_files/fun_table/genus/fun_micro_arthropodes_interactions_genus.csv")
```

## micro-organisms

### bacteria

```{r}
bacteria_guilds_genus <- get.guilds(sciName = unique(bacteria_tax$genus), endpoint = endpoint_url)
write.csv(apply(bacteria_guilds_genus,2,as.character), "final_files/fun_table/genus/fun_bacteria_guilds_genus.csv")

bacteria_diets_genus <- get.diets(sciName = unique(bacteria_tax$genus), endpoint = endpoint_url)
write.csv(apply(bacteria_diets_genus,2,as.character), "final_files/fun_table/genus/fun_bacteria_diets_genus.csv")

bacteria_groups_genus <- get.trophic.groups(sciName = unique(bacteria_tax$genus), endpoint = endpoint_url)
write.csv(apply(bacteria_groups_genus,2,as.character), "final_files/fun_table/genus/fun_bacteria_groups_genus.csv")

bacteria_interactions_genus <- get.interactions(sciName = unique(bacteria_tax$genus), endpoint = endpoint_url)
write.csv(apply(bacteria_interactions_genus,2,as.character), "final_files/fun_table/genus/fun_bacteria_interactions_genus.csv")
```

### fungi

```{r}
fungi_guilds_genus <- get.guilds(sciName = unique(fungi_tax$genus), endpoint = endpoint_url)
write.csv(apply(fungi_guilds_genus,2,as.character), "final_files/fun_table/genus/fun_fungi_guilds_genus.csv")

fungi_diets_genus <- get.diets(sciName = unique(fungi_tax$genus), endpoint = endpoint_url)
write.csv(apply(fungi_diets_genus,2,as.character), "final_files/fun_table/genus/fun_fungi_diets_genus.csv")

fungi_groups_genus <- get.trophic.groups(sciName = unique(fungi_tax$genus), endpoint = endpoint_url)
write.csv(apply(fungi_groups_genus,2,as.character), "final_files/fun_table/genus/fun_fungi_groups_genus.csv")

fungi_interactions_genus <- get.interactions(sciName = unique(fungi_tax$genus), endpoint = endpoint_url)
write.csv(apply(fungi_interactions_genus,2,as.character), "final_files/fun_table/genus/fun_fungi_interactions_genus.csv")
```

### protists

```{r}
protists_guilds_genus <- get.guilds(sciName = unique(protists_tax$genus), endpoint = endpoint_url)
write.csv(apply(protists_guilds_genus,2,as.character), "final_files/fun_table/genus/fun_protists_guilds_genus.csv")

protists_diets_genus <- get.diets(sciName = unique(protists_tax$genus), endpoint = endpoint_url)
write.csv(apply(protists_diets_genus,2,as.character), "final_files/fun_table/genus/fun_protists_diets_genus.csv")

protists_groups_genus <- get.trophic.groups(sciName = unique(protists_tax$genus), endpoint = endpoint_url)
write.csv(apply(protists_groups_genus,2,as.character), "final_files/fun_table/genus/fun_protists_groups_genus.csv")

protists_interactions_genus <- get.interactions(sciName = unique(protists_tax$genus), endpoint = endpoint_url)
write.csv(apply(protists_interactions_genus,2,as.character), "final_files/fun_table/genus/fun_protists_interactions_genus.csv")
```